{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "# General imports \n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import os\n",
    "import re\n",
    "import plotly.express as px\n",
    "from tqdm import tqdm\n",
    "import tempfile\n",
    "import csv\n",
    "import requests as r\n",
    "from Bio import SeqIO\n",
    "from io import StringIO\n",
    "import matplotlib.pyplot as plt\n",
    "import seaborn as sns\n",
    "import scipy\n",
    "\n",
    "\n",
    "# Import structuremap functions\n",
    "import structuremap.utils\n",
    "structuremap.utils.set_logger()\n",
    "from structuremap.processing import download_alphafold_cif, download_alphafold_pae, format_alphafold_data, annotate_accessibility, get_smooth_score, annotate_proteins_with_idr_pattern, get_extended_flexible_pattern, get_proximity_pvals, perform_enrichment_analysis, perform_enrichment_analysis_per_protein, evaluate_ptm_colocalization, extract_motifs_in_proteome\n",
    "from structuremap.plotting import plot_enrichment, plot_ptm_colocalization"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Set Parameters of Analysis"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "analysis_threshold = 20 # number of amino acids either side to analyze\n",
    "\n",
    "modifications = [\"649.3660\", \"655.3735\"] # which modifications we are looking for, as regex strings\n",
    "heavy_modification = \"655.3735\" \n",
    "light_modification = \"649.3660\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load Dataset - Labeled Methionines"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/ritwiksrinivas/Desktop/Projects/methionine-analysis/datasets\n"
     ]
    }
   ],
   "source": [
    "# path for csv output data\n",
    "datasets_path_str = \"../datasets/\"\n",
    "datasets_path = os.path.abspath(datasets_path_str)\n",
    "print(datasets_path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide Sequence</th>\n",
       "      <th>Modified Peptide</th>\n",
       "      <th>Light Modified Peptide</th>\n",
       "      <th>Heavy Modified Peptide</th>\n",
       "      <th>1 Log2 Ratio HL</th>\n",
       "      <th>10 Log2 Ratio HL</th>\n",
       "      <th>11 Log2 Ratio HL</th>\n",
       "      <th>12 Log2 Ratio HL</th>\n",
       "      <th>2 Log2 Ratio HL</th>\n",
       "      <th>3 Log2 Ratio HL</th>\n",
       "      <th>...</th>\n",
       "      <th>Protein</th>\n",
       "      <th>Protein ID</th>\n",
       "      <th>Entry Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Protein Description</th>\n",
       "      <th>p-value</th>\n",
       "      <th>neglogp</th>\n",
       "      <th>Log2HL avg</th>\n",
       "      <th>Unnamed: 24</th>\n",
       "      <th>Protein ID.1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVM[649.3660]IR</td>\n",
       "      <td>AADTIGYPVM[655.3735]IR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.269016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|Q8C196|CPSM_MOUSE</td>\n",
       "      <td>Q8C196</td>\n",
       "      <td>CPSM_MOUSE</td>\n",
       "      <td>Cps1</td>\n",
       "      <td>Carbamoyl-phosphate synthase [ammonia], mitoch...</td>\n",
       "      <td>4.058191e-03</td>\n",
       "      <td>2.391668</td>\n",
       "      <td>3.289988</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q8C196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAM[649.3660]QTLDMGR</td>\n",
       "      <td>IAM[655.3735]QTLDMGR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.783695</td>\n",
       "      <td>3.114945</td>\n",
       "      <td>2.697822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|Q07417|ACADS_MOUSE</td>\n",
       "      <td>Q07417</td>\n",
       "      <td>ACADS_MOUSE</td>\n",
       "      <td>Acads</td>\n",
       "      <td>Short-chain specific acyl-CoA dehydrogenase, m...</td>\n",
       "      <td>9.647931e-06</td>\n",
       "      <td>5.015566</td>\n",
       "      <td>2.789195</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q07417</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIM[649.3660]EK</td>\n",
       "      <td>FVGAVDPIM[655.3735]EK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.383482</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|Q91YI0|ARLY_MOUSE</td>\n",
       "      <td>Q91YI0</td>\n",
       "      <td>ARLY_MOUSE</td>\n",
       "      <td>Asl</td>\n",
       "      <td>Argininosuccinate lyase</td>\n",
       "      <td>4.283587e-02</td>\n",
       "      <td>1.368192</td>\n",
       "      <td>2.555706</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q91YI0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[649.3660]PINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[655.3735]PINGPFKPDHYRY</td>\n",
       "      <td>2.394458</td>\n",
       "      <td>2.380664</td>\n",
       "      <td>2.682897</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.435014</td>\n",
       "      <td>2.412394</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|P50247|SAHH_MOUSE</td>\n",
       "      <td>P50247</td>\n",
       "      <td>SAHH_MOUSE</td>\n",
       "      <td>Ahcy</td>\n",
       "      <td>Adenosylhomocysteinase</td>\n",
       "      <td>8.561078e-10</td>\n",
       "      <td>9.067472</td>\n",
       "      <td>2.538998</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P50247</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPM[649.3660]NLPHR</td>\n",
       "      <td>FADVIPM[655.3735]NLPHR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.580647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.446614</td>\n",
       "      <td>2.450718</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|P33267|CP2F2_MOUSE</td>\n",
       "      <td>P33267</td>\n",
       "      <td>CP2F2_MOUSE</td>\n",
       "      <td>Cyp2f2</td>\n",
       "      <td>Cytochrome P450 2F2</td>\n",
       "      <td>2.032232e-04</td>\n",
       "      <td>3.692027</td>\n",
       "      <td>2.389057</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P33267</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[649.3660]VTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[655.3735]VTLDYTVQVPGTGR</td>\n",
       "      <td>-1.822067</td>\n",
       "      <td>-1.687501</td>\n",
       "      <td>-1.706146</td>\n",
       "      <td>-1.647231</td>\n",
       "      <td>-1.793966</td>\n",
       "      <td>-1.407199</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|Q9QXF8|GNMT_MOUSE</td>\n",
       "      <td>Q9QXF8</td>\n",
       "      <td>GNMT_MOUSE</td>\n",
       "      <td>Gnmt</td>\n",
       "      <td>Glycine N-methyltransferase</td>\n",
       "      <td>1.880219e-08</td>\n",
       "      <td>7.725792</td>\n",
       "      <td>-1.857135</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q9QXF8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDM[649.3660]K</td>\n",
       "      <td>KEQESEVDM[655.3735]K</td>\n",
       "      <td>-1.759694</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.821773</td>\n",
       "      <td>-1.960428</td>\n",
       "      <td>-1.781592</td>\n",
       "      <td>-1.988603</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|Q8K3J1|NDUS8_MOUSE</td>\n",
       "      <td>Q8K3J1</td>\n",
       "      <td>NDUS8_MOUSE</td>\n",
       "      <td>Ndufs8</td>\n",
       "      <td>NADH dehydrogenase [ubiquinone] iron-sulfur pr...</td>\n",
       "      <td>7.034064e-14</td>\n",
       "      <td>13.152794</td>\n",
       "      <td>-1.871741</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q8K3J1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVM[649.3660]LAVDAVIAELK</td>\n",
       "      <td>RGVM[655.3735]LAVDAVIAELK</td>\n",
       "      <td>-1.812349</td>\n",
       "      <td>-1.980371</td>\n",
       "      <td>-2.307386</td>\n",
       "      <td>-2.492858</td>\n",
       "      <td>-1.405837</td>\n",
       "      <td>-2.094407</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|P63038|CH60_MOUSE</td>\n",
       "      <td>P63038</td>\n",
       "      <td>CH60_MOUSE</td>\n",
       "      <td>Hspd1</td>\n",
       "      <td>60 kDa heat shock protein, mitochondrial</td>\n",
       "      <td>7.289026e-11</td>\n",
       "      <td>10.137330</td>\n",
       "      <td>-1.950460</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P63038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>M[649.3660]QLLEIITTDK</td>\n",
       "      <td>M[655.3735]QLLEIITTDK</td>\n",
       "      <td>-2.286954</td>\n",
       "      <td>-2.700322</td>\n",
       "      <td>-1.941018</td>\n",
       "      <td>-1.860306</td>\n",
       "      <td>-2.069488</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|Q8BMS1|ECHA_MOUSE</td>\n",
       "      <td>Q8BMS1</td>\n",
       "      <td>ECHA_MOUSE</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondrial</td>\n",
       "      <td>5.582467e-09</td>\n",
       "      <td>8.253174</td>\n",
       "      <td>-2.270072</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q8BMS1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAM[649.3660]R</td>\n",
       "      <td>DHFEEAM[655.3735]R</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.891986</td>\n",
       "      <td>-2.129614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|Q01853|TERA_MOUSE</td>\n",
       "      <td>Q01853</td>\n",
       "      <td>TERA_MOUSE</td>\n",
       "      <td>Vcp</td>\n",
       "      <td>Transitional endoplasmic reticulum ATPase</td>\n",
       "      <td>6.484059e-04</td>\n",
       "      <td>3.188153</td>\n",
       "      <td>-2.507726</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q01853</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>203 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Peptide Sequence      Modified Peptide  \\\n",
       "0            AADTIGYPVMIR          AADTIGYPVMIR   \n",
       "1              IAMQTLDMGR            IAMQTLDMGR   \n",
       "2             FVGAVDPIMEK           FVGAVDPIMEK   \n",
       "3    QAQYLGMPINGPFKPDHYRY  QAQYLGMPINGPFKPDHYRY   \n",
       "4            FADVIPMNLPHR          FADVIPMNLPHR   \n",
       "..                    ...                   ...   \n",
       "198     AHMVTLDYTVQVPGTGR     AHMVTLDYTVQVPGTGR   \n",
       "199            KEQESEVDMK            KEQESEVDMK   \n",
       "200       RGVMLAVDAVIAELK       RGVMLAVDAVIAELK   \n",
       "201           MQLLEIITTDK           MQLLEIITTDK   \n",
       "202              DHFEEAMR              DHFEEAMR   \n",
       "\n",
       "             Light Modified Peptide          Heavy Modified Peptide  \\\n",
       "0            AADTIGYPVM[649.3660]IR          AADTIGYPVM[655.3735]IR   \n",
       "1              IAM[649.3660]QTLDMGR            IAM[655.3735]QTLDMGR   \n",
       "2             FVGAVDPIM[649.3660]EK           FVGAVDPIM[655.3735]EK   \n",
       "3    QAQYLGM[649.3660]PINGPFKPDHYRY  QAQYLGM[655.3735]PINGPFKPDHYRY   \n",
       "4            FADVIPM[649.3660]NLPHR          FADVIPM[655.3735]NLPHR   \n",
       "..                              ...                             ...   \n",
       "198     AHM[649.3660]VTLDYTVQVPGTGR     AHM[655.3735]VTLDYTVQVPGTGR   \n",
       "199            KEQESEVDM[649.3660]K            KEQESEVDM[655.3735]K   \n",
       "200       RGVM[649.3660]LAVDAVIAELK       RGVM[655.3735]LAVDAVIAELK   \n",
       "201           M[649.3660]QLLEIITTDK           M[655.3735]QLLEIITTDK   \n",
       "202              DHFEEAM[649.3660]R              DHFEEAM[655.3735]R   \n",
       "\n",
       "     1 Log2 Ratio HL  10 Log2 Ratio HL  11 Log2 Ratio HL  12 Log2 Ratio HL  \\\n",
       "0                NaN               NaN          3.269016               NaN   \n",
       "1                NaN          2.783695          3.114945          2.697822   \n",
       "2                NaN               NaN               NaN               NaN   \n",
       "3           2.394458          2.380664          2.682897               NaN   \n",
       "4                NaN               NaN          2.580647               NaN   \n",
       "..               ...               ...               ...               ...   \n",
       "198        -1.822067         -1.687501         -1.706146         -1.647231   \n",
       "199        -1.759694               NaN         -1.821773         -1.960428   \n",
       "200        -1.812349         -1.980371         -2.307386         -2.492858   \n",
       "201        -2.286954         -2.700322         -1.941018         -1.860306   \n",
       "202              NaN         -2.891986         -2.129614               NaN   \n",
       "\n",
       "     2 Log2 Ratio HL  3 Log2 Ratio HL  ...                Protein  Protein ID  \\\n",
       "0                NaN              NaN  ...   sp|Q8C196|CPSM_MOUSE      Q8C196   \n",
       "1                NaN              NaN  ...  sp|Q07417|ACADS_MOUSE      Q07417   \n",
       "2                NaN         2.383482  ...   sp|Q91YI0|ARLY_MOUSE      Q91YI0   \n",
       "3           2.435014         2.412394  ...   sp|P50247|SAHH_MOUSE      P50247   \n",
       "4           2.446614         2.450718  ...  sp|P33267|CP2F2_MOUSE      P33267   \n",
       "..               ...              ...  ...                    ...         ...   \n",
       "198        -1.793966        -1.407199  ...   sp|Q9QXF8|GNMT_MOUSE      Q9QXF8   \n",
       "199        -1.781592        -1.988603  ...  sp|Q8K3J1|NDUS8_MOUSE      Q8K3J1   \n",
       "200        -1.405837        -2.094407  ...   sp|P63038|CH60_MOUSE      P63038   \n",
       "201        -2.069488              NaN  ...   sp|Q8BMS1|ECHA_MOUSE      Q8BMS1   \n",
       "202              NaN              NaN  ...   sp|Q01853|TERA_MOUSE      Q01853   \n",
       "\n",
       "      Entry Name    Gene                                Protein Description  \\\n",
       "0     CPSM_MOUSE    Cps1  Carbamoyl-phosphate synthase [ammonia], mitoch...   \n",
       "1    ACADS_MOUSE   Acads  Short-chain specific acyl-CoA dehydrogenase, m...   \n",
       "2     ARLY_MOUSE     Asl                            Argininosuccinate lyase   \n",
       "3     SAHH_MOUSE    Ahcy                             Adenosylhomocysteinase   \n",
       "4    CP2F2_MOUSE  Cyp2f2                                Cytochrome P450 2F2   \n",
       "..           ...     ...                                                ...   \n",
       "198   GNMT_MOUSE    Gnmt                        Glycine N-methyltransferase   \n",
       "199  NDUS8_MOUSE  Ndufs8  NADH dehydrogenase [ubiquinone] iron-sulfur pr...   \n",
       "200   CH60_MOUSE   Hspd1           60 kDa heat shock protein, mitochondrial   \n",
       "201   ECHA_MOUSE   Hadha  Trifunctional enzyme subunit alpha, mitochondrial   \n",
       "202   TERA_MOUSE     Vcp          Transitional endoplasmic reticulum ATPase   \n",
       "\n",
       "          p-value    neglogp Log2HL avg Unnamed: 24 Protein ID.1  \n",
       "0    4.058191e-03   2.391668   3.289988         NaN       Q8C196  \n",
       "1    9.647931e-06   5.015566   2.789195         NaN       Q07417  \n",
       "2    4.283587e-02   1.368192   2.555706         NaN       Q91YI0  \n",
       "3    8.561078e-10   9.067472   2.538998         NaN       P50247  \n",
       "4    2.032232e-04   3.692027   2.389057         NaN       P33267  \n",
       "..            ...        ...        ...         ...          ...  \n",
       "198  1.880219e-08   7.725792  -1.857135         NaN       Q9QXF8  \n",
       "199  7.034064e-14  13.152794  -1.871741         NaN       Q8K3J1  \n",
       "200  7.289026e-11  10.137330  -1.950460         NaN       P63038  \n",
       "201  5.582467e-09   8.253174  -2.270072         NaN       Q8BMS1  \n",
       "202  6.484059e-04   3.188153  -2.507726         NaN       Q01853  \n",
       "\n",
       "[203 rows x 26 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_loc = os.path.join(datasets_path, \"RvsS_DataSet.xlsx\")\n",
    "peptides = pd.read_excel(data_loc, sheet_name=\"Sheet2\")\n",
    "peptides"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide Sequence</th>\n",
       "      <th>Modified Peptide</th>\n",
       "      <th>Light Modified Peptide</th>\n",
       "      <th>Heavy Modified Peptide</th>\n",
       "      <th>1 Log2 Ratio HL</th>\n",
       "      <th>10 Log2 Ratio HL</th>\n",
       "      <th>11 Log2 Ratio HL</th>\n",
       "      <th>12 Log2 Ratio HL</th>\n",
       "      <th>2 Log2 Ratio HL</th>\n",
       "      <th>3 Log2 Ratio HL</th>\n",
       "      <th>...</th>\n",
       "      <th>8 Log2 Ratio HL</th>\n",
       "      <th>9 Log2 Ratio HL</th>\n",
       "      <th>Protein</th>\n",
       "      <th>Protein ID</th>\n",
       "      <th>Entry Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Protein Description</th>\n",
       "      <th>p-value</th>\n",
       "      <th>neglogp</th>\n",
       "      <th>Log2HL avg</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVM[649.3660]IR</td>\n",
       "      <td>AADTIGYPVM[655.3735]IR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.269016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sp|Q8C196|CPSM_MOUSE</td>\n",
       "      <td>Q8C196</td>\n",
       "      <td>CPSM_MOUSE</td>\n",
       "      <td>Cps1</td>\n",
       "      <td>Carbamoyl-phosphate synthase [ammonia], mitoch...</td>\n",
       "      <td>4.058191e-03</td>\n",
       "      <td>2.391668</td>\n",
       "      <td>3.289988</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAM[649.3660]QTLDMGR</td>\n",
       "      <td>IAM[655.3735]QTLDMGR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.783695</td>\n",
       "      <td>3.114945</td>\n",
       "      <td>2.697822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sp|Q07417|ACADS_MOUSE</td>\n",
       "      <td>Q07417</td>\n",
       "      <td>ACADS_MOUSE</td>\n",
       "      <td>Acads</td>\n",
       "      <td>Short-chain specific acyl-CoA dehydrogenase, m...</td>\n",
       "      <td>9.647931e-06</td>\n",
       "      <td>5.015566</td>\n",
       "      <td>2.789195</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIM[649.3660]EK</td>\n",
       "      <td>FVGAVDPIM[655.3735]EK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.383482</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sp|Q91YI0|ARLY_MOUSE</td>\n",
       "      <td>Q91YI0</td>\n",
       "      <td>ARLY_MOUSE</td>\n",
       "      <td>Asl</td>\n",
       "      <td>Argininosuccinate lyase</td>\n",
       "      <td>4.283587e-02</td>\n",
       "      <td>1.368192</td>\n",
       "      <td>2.555706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[649.3660]PINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[655.3735]PINGPFKPDHYRY</td>\n",
       "      <td>2.394458</td>\n",
       "      <td>2.380664</td>\n",
       "      <td>2.682897</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.435014</td>\n",
       "      <td>2.412394</td>\n",
       "      <td>...</td>\n",
       "      <td>2.839492</td>\n",
       "      <td>2.607501</td>\n",
       "      <td>sp|P50247|SAHH_MOUSE</td>\n",
       "      <td>P50247</td>\n",
       "      <td>SAHH_MOUSE</td>\n",
       "      <td>Ahcy</td>\n",
       "      <td>Adenosylhomocysteinase</td>\n",
       "      <td>8.561078e-10</td>\n",
       "      <td>9.067472</td>\n",
       "      <td>2.538998</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPM[649.3660]NLPHR</td>\n",
       "      <td>FADVIPM[655.3735]NLPHR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.580647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.446614</td>\n",
       "      <td>2.450718</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sp|P33267|CP2F2_MOUSE</td>\n",
       "      <td>P33267</td>\n",
       "      <td>CP2F2_MOUSE</td>\n",
       "      <td>Cyp2f2</td>\n",
       "      <td>Cytochrome P450 2F2</td>\n",
       "      <td>2.032232e-04</td>\n",
       "      <td>3.692027</td>\n",
       "      <td>2.389057</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[649.3660]VTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[655.3735]VTLDYTVQVPGTGR</td>\n",
       "      <td>-1.822067</td>\n",
       "      <td>-1.687501</td>\n",
       "      <td>-1.706146</td>\n",
       "      <td>-1.647231</td>\n",
       "      <td>-1.793966</td>\n",
       "      <td>-1.407199</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.613004</td>\n",
       "      <td>-1.760150</td>\n",
       "      <td>sp|Q9QXF8|GNMT_MOUSE</td>\n",
       "      <td>Q9QXF8</td>\n",
       "      <td>GNMT_MOUSE</td>\n",
       "      <td>Gnmt</td>\n",
       "      <td>Glycine N-methyltransferase</td>\n",
       "      <td>1.880219e-08</td>\n",
       "      <td>7.725792</td>\n",
       "      <td>-1.857135</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDM[649.3660]K</td>\n",
       "      <td>KEQESEVDM[655.3735]K</td>\n",
       "      <td>-1.759694</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.821773</td>\n",
       "      <td>-1.960428</td>\n",
       "      <td>-1.781592</td>\n",
       "      <td>-1.988603</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.891596</td>\n",
       "      <td>-1.881011</td>\n",
       "      <td>sp|Q8K3J1|NDUS8_MOUSE</td>\n",
       "      <td>Q8K3J1</td>\n",
       "      <td>NDUS8_MOUSE</td>\n",
       "      <td>Ndufs8</td>\n",
       "      <td>NADH dehydrogenase [ubiquinone] iron-sulfur pr...</td>\n",
       "      <td>7.034064e-14</td>\n",
       "      <td>13.152794</td>\n",
       "      <td>-1.871741</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVM[649.3660]LAVDAVIAELK</td>\n",
       "      <td>RGVM[655.3735]LAVDAVIAELK</td>\n",
       "      <td>-1.812349</td>\n",
       "      <td>-1.980371</td>\n",
       "      <td>-2.307386</td>\n",
       "      <td>-2.492858</td>\n",
       "      <td>-1.405837</td>\n",
       "      <td>-2.094407</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.829820</td>\n",
       "      <td>-1.935236</td>\n",
       "      <td>sp|P63038|CH60_MOUSE</td>\n",
       "      <td>P63038</td>\n",
       "      <td>CH60_MOUSE</td>\n",
       "      <td>Hspd1</td>\n",
       "      <td>60 kDa heat shock protein, mitochondrial</td>\n",
       "      <td>7.289026e-11</td>\n",
       "      <td>10.137330</td>\n",
       "      <td>-1.950460</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>M[649.3660]QLLEIITTDK</td>\n",
       "      <td>M[655.3735]QLLEIITTDK</td>\n",
       "      <td>-2.286954</td>\n",
       "      <td>-2.700322</td>\n",
       "      <td>-1.941018</td>\n",
       "      <td>-1.860306</td>\n",
       "      <td>-2.069488</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.326194</td>\n",
       "      <td>-2.205174</td>\n",
       "      <td>sp|Q8BMS1|ECHA_MOUSE</td>\n",
       "      <td>Q8BMS1</td>\n",
       "      <td>ECHA_MOUSE</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondrial</td>\n",
       "      <td>5.582467e-09</td>\n",
       "      <td>8.253174</td>\n",
       "      <td>-2.270072</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAM[649.3660]R</td>\n",
       "      <td>DHFEEAM[655.3735]R</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.891986</td>\n",
       "      <td>-2.129614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.351995</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sp|Q01853|TERA_MOUSE</td>\n",
       "      <td>Q01853</td>\n",
       "      <td>TERA_MOUSE</td>\n",
       "      <td>Vcp</td>\n",
       "      <td>Transitional endoplasmic reticulum ATPase</td>\n",
       "      <td>6.484059e-04</td>\n",
       "      <td>3.188153</td>\n",
       "      <td>-2.507726</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>203 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Peptide Sequence      Modified Peptide  \\\n",
       "0            AADTIGYPVMIR          AADTIGYPVMIR   \n",
       "1              IAMQTLDMGR            IAMQTLDMGR   \n",
       "2             FVGAVDPIMEK           FVGAVDPIMEK   \n",
       "3    QAQYLGMPINGPFKPDHYRY  QAQYLGMPINGPFKPDHYRY   \n",
       "4            FADVIPMNLPHR          FADVIPMNLPHR   \n",
       "..                    ...                   ...   \n",
       "198     AHMVTLDYTVQVPGTGR     AHMVTLDYTVQVPGTGR   \n",
       "199            KEQESEVDMK            KEQESEVDMK   \n",
       "200       RGVMLAVDAVIAELK       RGVMLAVDAVIAELK   \n",
       "201           MQLLEIITTDK           MQLLEIITTDK   \n",
       "202              DHFEEAMR              DHFEEAMR   \n",
       "\n",
       "             Light Modified Peptide          Heavy Modified Peptide  \\\n",
       "0            AADTIGYPVM[649.3660]IR          AADTIGYPVM[655.3735]IR   \n",
       "1              IAM[649.3660]QTLDMGR            IAM[655.3735]QTLDMGR   \n",
       "2             FVGAVDPIM[649.3660]EK           FVGAVDPIM[655.3735]EK   \n",
       "3    QAQYLGM[649.3660]PINGPFKPDHYRY  QAQYLGM[655.3735]PINGPFKPDHYRY   \n",
       "4            FADVIPM[649.3660]NLPHR          FADVIPM[655.3735]NLPHR   \n",
       "..                              ...                             ...   \n",
       "198     AHM[649.3660]VTLDYTVQVPGTGR     AHM[655.3735]VTLDYTVQVPGTGR   \n",
       "199            KEQESEVDM[649.3660]K            KEQESEVDM[655.3735]K   \n",
       "200       RGVM[649.3660]LAVDAVIAELK       RGVM[655.3735]LAVDAVIAELK   \n",
       "201           M[649.3660]QLLEIITTDK           M[655.3735]QLLEIITTDK   \n",
       "202              DHFEEAM[649.3660]R              DHFEEAM[655.3735]R   \n",
       "\n",
       "     1 Log2 Ratio HL  10 Log2 Ratio HL  11 Log2 Ratio HL  12 Log2 Ratio HL  \\\n",
       "0                NaN               NaN          3.269016               NaN   \n",
       "1                NaN          2.783695          3.114945          2.697822   \n",
       "2                NaN               NaN               NaN               NaN   \n",
       "3           2.394458          2.380664          2.682897               NaN   \n",
       "4                NaN               NaN          2.580647               NaN   \n",
       "..               ...               ...               ...               ...   \n",
       "198        -1.822067         -1.687501         -1.706146         -1.647231   \n",
       "199        -1.759694               NaN         -1.821773         -1.960428   \n",
       "200        -1.812349         -1.980371         -2.307386         -2.492858   \n",
       "201        -2.286954         -2.700322         -1.941018         -1.860306   \n",
       "202              NaN         -2.891986         -2.129614               NaN   \n",
       "\n",
       "     2 Log2 Ratio HL  3 Log2 Ratio HL  ...  8 Log2 Ratio HL  9 Log2 Ratio HL  \\\n",
       "0                NaN              NaN  ...              NaN              NaN   \n",
       "1                NaN              NaN  ...              NaN              NaN   \n",
       "2                NaN         2.383482  ...              NaN              NaN   \n",
       "3           2.435014         2.412394  ...         2.839492         2.607501   \n",
       "4           2.446614         2.450718  ...              NaN              NaN   \n",
       "..               ...              ...  ...              ...              ...   \n",
       "198        -1.793966        -1.407199  ...        -1.613004        -1.760150   \n",
       "199        -1.781592        -1.988603  ...        -1.891596        -1.881011   \n",
       "200        -1.405837        -2.094407  ...        -1.829820        -1.935236   \n",
       "201        -2.069488              NaN  ...        -2.326194        -2.205174   \n",
       "202              NaN              NaN  ...        -2.351995              NaN   \n",
       "\n",
       "                   Protein  Protein ID   Entry Name    Gene  \\\n",
       "0     sp|Q8C196|CPSM_MOUSE      Q8C196   CPSM_MOUSE    Cps1   \n",
       "1    sp|Q07417|ACADS_MOUSE      Q07417  ACADS_MOUSE   Acads   \n",
       "2     sp|Q91YI0|ARLY_MOUSE      Q91YI0   ARLY_MOUSE     Asl   \n",
       "3     sp|P50247|SAHH_MOUSE      P50247   SAHH_MOUSE    Ahcy   \n",
       "4    sp|P33267|CP2F2_MOUSE      P33267  CP2F2_MOUSE  Cyp2f2   \n",
       "..                     ...         ...          ...     ...   \n",
       "198   sp|Q9QXF8|GNMT_MOUSE      Q9QXF8   GNMT_MOUSE    Gnmt   \n",
       "199  sp|Q8K3J1|NDUS8_MOUSE      Q8K3J1  NDUS8_MOUSE  Ndufs8   \n",
       "200   sp|P63038|CH60_MOUSE      P63038   CH60_MOUSE   Hspd1   \n",
       "201   sp|Q8BMS1|ECHA_MOUSE      Q8BMS1   ECHA_MOUSE   Hadha   \n",
       "202   sp|Q01853|TERA_MOUSE      Q01853   TERA_MOUSE     Vcp   \n",
       "\n",
       "                                   Protein Description       p-value  \\\n",
       "0    Carbamoyl-phosphate synthase [ammonia], mitoch...  4.058191e-03   \n",
       "1    Short-chain specific acyl-CoA dehydrogenase, m...  9.647931e-06   \n",
       "2                              Argininosuccinate lyase  4.283587e-02   \n",
       "3                               Adenosylhomocysteinase  8.561078e-10   \n",
       "4                                  Cytochrome P450 2F2  2.032232e-04   \n",
       "..                                                 ...           ...   \n",
       "198                        Glycine N-methyltransferase  1.880219e-08   \n",
       "199  NADH dehydrogenase [ubiquinone] iron-sulfur pr...  7.034064e-14   \n",
       "200           60 kDa heat shock protein, mitochondrial  7.289026e-11   \n",
       "201  Trifunctional enzyme subunit alpha, mitochondrial  5.582467e-09   \n",
       "202          Transitional endoplasmic reticulum ATPase  6.484059e-04   \n",
       "\n",
       "       neglogp Log2HL avg  \n",
       "0     2.391668   3.289988  \n",
       "1     5.015566   2.789195  \n",
       "2     1.368192   2.555706  \n",
       "3     9.067472   2.538998  \n",
       "4     3.692027   2.389057  \n",
       "..         ...        ...  \n",
       "198   7.725792  -1.857135  \n",
       "199  13.152794  -1.871741  \n",
       "200  10.137330  -1.950460  \n",
       "201   8.253174  -2.270072  \n",
       "202   3.188153  -2.507726  \n",
       "\n",
       "[203 rows x 24 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Canonicalize data\n",
    "#peptides[\"Assigned Modifications\"] = peptides[\"Assigned Modifications\"].str.replace(\"(\", \"[\", regex=False)\n",
    "#peptides[\"Assigned Modifications\"] = peptides[\"Assigned Modifications\"].str.replace(\")\", \"]\", regex=False)\n",
    "\n",
    "peptides.drop(columns=[\"Unnamed: 24\", \"Protein ID.1\"], axis=1, inplace=True)\n",
    "peptides"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide Sequence</th>\n",
       "      <th>Modified Peptide</th>\n",
       "      <th>Light Modified Peptide</th>\n",
       "      <th>Heavy Modified Peptide</th>\n",
       "      <th>1 Log2 Ratio HL</th>\n",
       "      <th>10 Log2 Ratio HL</th>\n",
       "      <th>11 Log2 Ratio HL</th>\n",
       "      <th>12 Log2 Ratio HL</th>\n",
       "      <th>2 Log2 Ratio HL</th>\n",
       "      <th>3 Log2 Ratio HL</th>\n",
       "      <th>...</th>\n",
       "      <th>9 Log2 Ratio HL</th>\n",
       "      <th>Protein</th>\n",
       "      <th>Protein ID</th>\n",
       "      <th>Entry Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Protein Description</th>\n",
       "      <th>p-value</th>\n",
       "      <th>neglogp</th>\n",
       "      <th>Log2HL avg</th>\n",
       "      <th>label</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVM[649.3660]IR</td>\n",
       "      <td>AADTIGYPVM[655.3735]IR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.269016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sp|Q8C196|CPSM_MOUSE</td>\n",
       "      <td>Q8C196</td>\n",
       "      <td>CPSM_MOUSE</td>\n",
       "      <td>Cps1</td>\n",
       "      <td>Carbamoyl-phosphate synthase [ammonia], mitoch...</td>\n",
       "      <td>4.058191e-03</td>\n",
       "      <td>2.391668</td>\n",
       "      <td>3.289988</td>\n",
       "      <td>green</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAM[649.3660]QTLDMGR</td>\n",
       "      <td>IAM[655.3735]QTLDMGR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.783695</td>\n",
       "      <td>3.114945</td>\n",
       "      <td>2.697822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sp|Q07417|ACADS_MOUSE</td>\n",
       "      <td>Q07417</td>\n",
       "      <td>ACADS_MOUSE</td>\n",
       "      <td>Acads</td>\n",
       "      <td>Short-chain specific acyl-CoA dehydrogenase, m...</td>\n",
       "      <td>9.647931e-06</td>\n",
       "      <td>5.015566</td>\n",
       "      <td>2.789195</td>\n",
       "      <td>green</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIM[649.3660]EK</td>\n",
       "      <td>FVGAVDPIM[655.3735]EK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.383482</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sp|Q91YI0|ARLY_MOUSE</td>\n",
       "      <td>Q91YI0</td>\n",
       "      <td>ARLY_MOUSE</td>\n",
       "      <td>Asl</td>\n",
       "      <td>Argininosuccinate lyase</td>\n",
       "      <td>4.283587e-02</td>\n",
       "      <td>1.368192</td>\n",
       "      <td>2.555706</td>\n",
       "      <td>green</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[649.3660]PINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[655.3735]PINGPFKPDHYRY</td>\n",
       "      <td>2.394458</td>\n",
       "      <td>2.380664</td>\n",
       "      <td>2.682897</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.435014</td>\n",
       "      <td>2.412394</td>\n",
       "      <td>...</td>\n",
       "      <td>2.607501</td>\n",
       "      <td>sp|P50247|SAHH_MOUSE</td>\n",
       "      <td>P50247</td>\n",
       "      <td>SAHH_MOUSE</td>\n",
       "      <td>Ahcy</td>\n",
       "      <td>Adenosylhomocysteinase</td>\n",
       "      <td>8.561078e-10</td>\n",
       "      <td>9.067472</td>\n",
       "      <td>2.538998</td>\n",
       "      <td>green</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPM[649.3660]NLPHR</td>\n",
       "      <td>FADVIPM[655.3735]NLPHR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.580647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.446614</td>\n",
       "      <td>2.450718</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sp|P33267|CP2F2_MOUSE</td>\n",
       "      <td>P33267</td>\n",
       "      <td>CP2F2_MOUSE</td>\n",
       "      <td>Cyp2f2</td>\n",
       "      <td>Cytochrome P450 2F2</td>\n",
       "      <td>2.032232e-04</td>\n",
       "      <td>3.692027</td>\n",
       "      <td>2.389057</td>\n",
       "      <td>green</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[649.3660]VTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[655.3735]VTLDYTVQVPGTGR</td>\n",
       "      <td>-1.822067</td>\n",
       "      <td>-1.687501</td>\n",
       "      <td>-1.706146</td>\n",
       "      <td>-1.647231</td>\n",
       "      <td>-1.793966</td>\n",
       "      <td>-1.407199</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.760150</td>\n",
       "      <td>sp|Q9QXF8|GNMT_MOUSE</td>\n",
       "      <td>Q9QXF8</td>\n",
       "      <td>GNMT_MOUSE</td>\n",
       "      <td>Gnmt</td>\n",
       "      <td>Glycine N-methyltransferase</td>\n",
       "      <td>1.880219e-08</td>\n",
       "      <td>7.725792</td>\n",
       "      <td>-1.857135</td>\n",
       "      <td>yellow</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDM[649.3660]K</td>\n",
       "      <td>KEQESEVDM[655.3735]K</td>\n",
       "      <td>-1.759694</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.821773</td>\n",
       "      <td>-1.960428</td>\n",
       "      <td>-1.781592</td>\n",
       "      <td>-1.988603</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.881011</td>\n",
       "      <td>sp|Q8K3J1|NDUS8_MOUSE</td>\n",
       "      <td>Q8K3J1</td>\n",
       "      <td>NDUS8_MOUSE</td>\n",
       "      <td>Ndufs8</td>\n",
       "      <td>NADH dehydrogenase [ubiquinone] iron-sulfur pr...</td>\n",
       "      <td>7.034064e-14</td>\n",
       "      <td>13.152794</td>\n",
       "      <td>-1.871741</td>\n",
       "      <td>yellow</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVM[649.3660]LAVDAVIAELK</td>\n",
       "      <td>RGVM[655.3735]LAVDAVIAELK</td>\n",
       "      <td>-1.812349</td>\n",
       "      <td>-1.980371</td>\n",
       "      <td>-2.307386</td>\n",
       "      <td>-2.492858</td>\n",
       "      <td>-1.405837</td>\n",
       "      <td>-2.094407</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.935236</td>\n",
       "      <td>sp|P63038|CH60_MOUSE</td>\n",
       "      <td>P63038</td>\n",
       "      <td>CH60_MOUSE</td>\n",
       "      <td>Hspd1</td>\n",
       "      <td>60 kDa heat shock protein, mitochondrial</td>\n",
       "      <td>7.289026e-11</td>\n",
       "      <td>10.137330</td>\n",
       "      <td>-1.950460</td>\n",
       "      <td>yellow</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>M[649.3660]QLLEIITTDK</td>\n",
       "      <td>M[655.3735]QLLEIITTDK</td>\n",
       "      <td>-2.286954</td>\n",
       "      <td>-2.700322</td>\n",
       "      <td>-1.941018</td>\n",
       "      <td>-1.860306</td>\n",
       "      <td>-2.069488</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.205174</td>\n",
       "      <td>sp|Q8BMS1|ECHA_MOUSE</td>\n",
       "      <td>Q8BMS1</td>\n",
       "      <td>ECHA_MOUSE</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondrial</td>\n",
       "      <td>5.582467e-09</td>\n",
       "      <td>8.253174</td>\n",
       "      <td>-2.270072</td>\n",
       "      <td>yellow</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAM[649.3660]R</td>\n",
       "      <td>DHFEEAM[655.3735]R</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.891986</td>\n",
       "      <td>-2.129614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>sp|Q01853|TERA_MOUSE</td>\n",
       "      <td>Q01853</td>\n",
       "      <td>TERA_MOUSE</td>\n",
       "      <td>Vcp</td>\n",
       "      <td>Transitional endoplasmic reticulum ATPase</td>\n",
       "      <td>6.484059e-04</td>\n",
       "      <td>3.188153</td>\n",
       "      <td>-2.507726</td>\n",
       "      <td>yellow</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>203 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Peptide Sequence      Modified Peptide  \\\n",
       "0            AADTIGYPVMIR          AADTIGYPVMIR   \n",
       "1              IAMQTLDMGR            IAMQTLDMGR   \n",
       "2             FVGAVDPIMEK           FVGAVDPIMEK   \n",
       "3    QAQYLGMPINGPFKPDHYRY  QAQYLGMPINGPFKPDHYRY   \n",
       "4            FADVIPMNLPHR          FADVIPMNLPHR   \n",
       "..                    ...                   ...   \n",
       "198     AHMVTLDYTVQVPGTGR     AHMVTLDYTVQVPGTGR   \n",
       "199            KEQESEVDMK            KEQESEVDMK   \n",
       "200       RGVMLAVDAVIAELK       RGVMLAVDAVIAELK   \n",
       "201           MQLLEIITTDK           MQLLEIITTDK   \n",
       "202              DHFEEAMR              DHFEEAMR   \n",
       "\n",
       "             Light Modified Peptide          Heavy Modified Peptide  \\\n",
       "0            AADTIGYPVM[649.3660]IR          AADTIGYPVM[655.3735]IR   \n",
       "1              IAM[649.3660]QTLDMGR            IAM[655.3735]QTLDMGR   \n",
       "2             FVGAVDPIM[649.3660]EK           FVGAVDPIM[655.3735]EK   \n",
       "3    QAQYLGM[649.3660]PINGPFKPDHYRY  QAQYLGM[655.3735]PINGPFKPDHYRY   \n",
       "4            FADVIPM[649.3660]NLPHR          FADVIPM[655.3735]NLPHR   \n",
       "..                              ...                             ...   \n",
       "198     AHM[649.3660]VTLDYTVQVPGTGR     AHM[655.3735]VTLDYTVQVPGTGR   \n",
       "199            KEQESEVDM[649.3660]K            KEQESEVDM[655.3735]K   \n",
       "200       RGVM[649.3660]LAVDAVIAELK       RGVM[655.3735]LAVDAVIAELK   \n",
       "201           M[649.3660]QLLEIITTDK           M[655.3735]QLLEIITTDK   \n",
       "202              DHFEEAM[649.3660]R              DHFEEAM[655.3735]R   \n",
       "\n",
       "     1 Log2 Ratio HL  10 Log2 Ratio HL  11 Log2 Ratio HL  12 Log2 Ratio HL  \\\n",
       "0                NaN               NaN          3.269016               NaN   \n",
       "1                NaN          2.783695          3.114945          2.697822   \n",
       "2                NaN               NaN               NaN               NaN   \n",
       "3           2.394458          2.380664          2.682897               NaN   \n",
       "4                NaN               NaN          2.580647               NaN   \n",
       "..               ...               ...               ...               ...   \n",
       "198        -1.822067         -1.687501         -1.706146         -1.647231   \n",
       "199        -1.759694               NaN         -1.821773         -1.960428   \n",
       "200        -1.812349         -1.980371         -2.307386         -2.492858   \n",
       "201        -2.286954         -2.700322         -1.941018         -1.860306   \n",
       "202              NaN         -2.891986         -2.129614               NaN   \n",
       "\n",
       "     2 Log2 Ratio HL  3 Log2 Ratio HL  ...  9 Log2 Ratio HL  \\\n",
       "0                NaN              NaN  ...              NaN   \n",
       "1                NaN              NaN  ...              NaN   \n",
       "2                NaN         2.383482  ...              NaN   \n",
       "3           2.435014         2.412394  ...         2.607501   \n",
       "4           2.446614         2.450718  ...              NaN   \n",
       "..               ...              ...  ...              ...   \n",
       "198        -1.793966        -1.407199  ...        -1.760150   \n",
       "199        -1.781592        -1.988603  ...        -1.881011   \n",
       "200        -1.405837        -2.094407  ...        -1.935236   \n",
       "201        -2.069488              NaN  ...        -2.205174   \n",
       "202              NaN              NaN  ...              NaN   \n",
       "\n",
       "                   Protein  Protein ID   Entry Name    Gene  \\\n",
       "0     sp|Q8C196|CPSM_MOUSE      Q8C196   CPSM_MOUSE    Cps1   \n",
       "1    sp|Q07417|ACADS_MOUSE      Q07417  ACADS_MOUSE   Acads   \n",
       "2     sp|Q91YI0|ARLY_MOUSE      Q91YI0   ARLY_MOUSE     Asl   \n",
       "3     sp|P50247|SAHH_MOUSE      P50247   SAHH_MOUSE    Ahcy   \n",
       "4    sp|P33267|CP2F2_MOUSE      P33267  CP2F2_MOUSE  Cyp2f2   \n",
       "..                     ...         ...          ...     ...   \n",
       "198   sp|Q9QXF8|GNMT_MOUSE      Q9QXF8   GNMT_MOUSE    Gnmt   \n",
       "199  sp|Q8K3J1|NDUS8_MOUSE      Q8K3J1  NDUS8_MOUSE  Ndufs8   \n",
       "200   sp|P63038|CH60_MOUSE      P63038   CH60_MOUSE   Hspd1   \n",
       "201   sp|Q8BMS1|ECHA_MOUSE      Q8BMS1   ECHA_MOUSE   Hadha   \n",
       "202   sp|Q01853|TERA_MOUSE      Q01853   TERA_MOUSE     Vcp   \n",
       "\n",
       "                                   Protein Description       p-value  \\\n",
       "0    Carbamoyl-phosphate synthase [ammonia], mitoch...  4.058191e-03   \n",
       "1    Short-chain specific acyl-CoA dehydrogenase, m...  9.647931e-06   \n",
       "2                              Argininosuccinate lyase  4.283587e-02   \n",
       "3                               Adenosylhomocysteinase  8.561078e-10   \n",
       "4                                  Cytochrome P450 2F2  2.032232e-04   \n",
       "..                                                 ...           ...   \n",
       "198                        Glycine N-methyltransferase  1.880219e-08   \n",
       "199  NADH dehydrogenase [ubiquinone] iron-sulfur pr...  7.034064e-14   \n",
       "200           60 kDa heat shock protein, mitochondrial  7.289026e-11   \n",
       "201  Trifunctional enzyme subunit alpha, mitochondrial  5.582467e-09   \n",
       "202          Transitional endoplasmic reticulum ATPase  6.484059e-04   \n",
       "\n",
       "       neglogp Log2HL avg   label  \n",
       "0     2.391668   3.289988   green  \n",
       "1     5.015566   2.789195   green  \n",
       "2     1.368192   2.555706   green  \n",
       "3     9.067472   2.538998   green  \n",
       "4     3.692027   2.389057   green  \n",
       "..         ...        ...     ...  \n",
       "198   7.725792  -1.857135  yellow  \n",
       "199  13.152794  -1.871741  yellow  \n",
       "200  10.137330  -1.950460  yellow  \n",
       "201   8.253174  -2.270072  yellow  \n",
       "202   3.188153  -2.507726  yellow  \n",
       "\n",
       "[203 rows x 25 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "label_col_data = [\"green\"] * 39 + [\"white\"] * 143 + [\"yellow\"] * 21\n",
    "label_col = pd.Series(label_col_data)\n",
    "peptides[\"label\"] = label_col\n",
    "peptides"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# helper function to get full amino acid sequence for a protein\n",
    "def get_full_protein_seq(cID):\n",
    "    baseUrl=\"http://www.uniprot.org/uniprot/\"\n",
    "    currentUrl=baseUrl+cID+\".fasta\"\n",
    "    response = r.post(currentUrl)\n",
    "    cData=''.join(response.text)\n",
    "    \n",
    "    Seq=StringIO(cData)\n",
    "    pSeq=list(SeqIO.parse(Seq,\"fasta\"))\n",
    "\n",
    "    return str(pSeq[0].seq)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# test - get a single amino acid sequence\n",
    "#first_protein_ID = peptides[\"Protein ID\"].iloc[0]\n",
    "#test_sequence = get_full_protein_seq(first_protein_ID)\n",
    "#print(test_sequence[575:587])\n",
    "#print(peptides[\"Peptide Sequence\"].iloc[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# get whole amino acid sequences for methionine peptides\n",
    "# SLOW - ONLY DO THIS ONCE - CONVERT TO CSV FILE, THEN RE-LOAD FROM THAT FILE\n",
    "\n",
    "#tqdm.pandas()\n",
    "#peptides_completed_sequence = peptides.copy()\n",
    "#peptides_completed_sequence[\"Complete Sequence\"] = peptides_completed_sequence[\"Protein ID\"].progress_apply(get_full_protein_seq)\n",
    "#peptides_completed_sequence\n",
    "\n",
    "# NOTE: WE CAN SPEED THIS UP BY ONLY GETTING UNIQUE PROTEINS (lots of repeats in the dataset)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#peptides_completed_sequence.to_csv(os.path.join(datasets_path, \"RvsS_peptides_completed_sequence.csv\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide Sequence</th>\n",
       "      <th>Modified Peptide</th>\n",
       "      <th>Light Modified Peptide</th>\n",
       "      <th>Heavy Modified Peptide</th>\n",
       "      <th>1 Log2 Ratio HL</th>\n",
       "      <th>10 Log2 Ratio HL</th>\n",
       "      <th>11 Log2 Ratio HL</th>\n",
       "      <th>12 Log2 Ratio HL</th>\n",
       "      <th>2 Log2 Ratio HL</th>\n",
       "      <th>3 Log2 Ratio HL</th>\n",
       "      <th>...</th>\n",
       "      <th>Protein</th>\n",
       "      <th>Protein ID</th>\n",
       "      <th>Entry Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Protein Description</th>\n",
       "      <th>p-value</th>\n",
       "      <th>neglogp</th>\n",
       "      <th>Log2HL avg</th>\n",
       "      <th>label</th>\n",
       "      <th>Complete Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVM[649.3660]IR</td>\n",
       "      <td>AADTIGYPVM[655.3735]IR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.269016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|Q8C196|CPSM_MOUSE</td>\n",
       "      <td>Q8C196</td>\n",
       "      <td>CPSM_MOUSE</td>\n",
       "      <td>Cps1</td>\n",
       "      <td>Carbamoyl-phosphate synthase [ammonia], mitoch...</td>\n",
       "      <td>4.058191e-03</td>\n",
       "      <td>2.391668</td>\n",
       "      <td>3.289988</td>\n",
       "      <td>green</td>\n",
       "      <td>MTRILTACKVVKTLKSGFGFANVTTKRQWDFSRPGIRLLSVKAKTA...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAM[649.3660]QTLDMGR</td>\n",
       "      <td>IAM[655.3735]QTLDMGR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.783695</td>\n",
       "      <td>3.114945</td>\n",
       "      <td>2.697822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|Q07417|ACADS_MOUSE</td>\n",
       "      <td>Q07417</td>\n",
       "      <td>ACADS_MOUSE</td>\n",
       "      <td>Acads</td>\n",
       "      <td>Short-chain specific acyl-CoA dehydrogenase, m...</td>\n",
       "      <td>9.647931e-06</td>\n",
       "      <td>5.015566</td>\n",
       "      <td>2.789195</td>\n",
       "      <td>green</td>\n",
       "      <td>MAAALLARARGPLRRALGVRDWRRLHTVYQSVELPETHQMLRQTCR...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIM[649.3660]EK</td>\n",
       "      <td>FVGAVDPIM[655.3735]EK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.383482</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|Q91YI0|ARLY_MOUSE</td>\n",
       "      <td>Q91YI0</td>\n",
       "      <td>ARLY_MOUSE</td>\n",
       "      <td>Asl</td>\n",
       "      <td>Argininosuccinate lyase</td>\n",
       "      <td>4.283587e-02</td>\n",
       "      <td>1.368192</td>\n",
       "      <td>2.555706</td>\n",
       "      <td>green</td>\n",
       "      <td>MASESGKLWGGRFVGAVDPIMEKFNSSISYDRHLWNVDVQGSKAYS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[649.3660]PINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[655.3735]PINGPFKPDHYRY</td>\n",
       "      <td>2.394458</td>\n",
       "      <td>2.380664</td>\n",
       "      <td>2.682897</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.435014</td>\n",
       "      <td>2.412394</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|P50247|SAHH_MOUSE</td>\n",
       "      <td>P50247</td>\n",
       "      <td>SAHH_MOUSE</td>\n",
       "      <td>Ahcy</td>\n",
       "      <td>Adenosylhomocysteinase</td>\n",
       "      <td>8.561078e-10</td>\n",
       "      <td>9.067472</td>\n",
       "      <td>2.538998</td>\n",
       "      <td>green</td>\n",
       "      <td>MSDKLPYKVADIGLAAWGRKALDIAENEMPGLMRMREMYSASKPLK...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPM[649.3660]NLPHR</td>\n",
       "      <td>FADVIPM[655.3735]NLPHR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.580647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.446614</td>\n",
       "      <td>2.450718</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|P33267|CP2F2_MOUSE</td>\n",
       "      <td>P33267</td>\n",
       "      <td>CP2F2_MOUSE</td>\n",
       "      <td>Cyp2f2</td>\n",
       "      <td>Cytochrome P450 2F2</td>\n",
       "      <td>2.032232e-04</td>\n",
       "      <td>3.692027</td>\n",
       "      <td>2.389057</td>\n",
       "      <td>green</td>\n",
       "      <td>MDGVSTAILLLLLAVISLSLTFSSRGKGQLPPGPKPLPILGNLLQL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[649.3660]VTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[655.3735]VTLDYTVQVPGTGR</td>\n",
       "      <td>-1.822067</td>\n",
       "      <td>-1.687501</td>\n",
       "      <td>-1.706146</td>\n",
       "      <td>-1.647231</td>\n",
       "      <td>-1.793966</td>\n",
       "      <td>-1.407199</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|Q9QXF8|GNMT_MOUSE</td>\n",
       "      <td>Q9QXF8</td>\n",
       "      <td>GNMT_MOUSE</td>\n",
       "      <td>Gnmt</td>\n",
       "      <td>Glycine N-methyltransferase</td>\n",
       "      <td>1.880219e-08</td>\n",
       "      <td>7.725792</td>\n",
       "      <td>-1.857135</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MVDSVYRTRSLGVAAEGLPDQYADGEAARVWQLYIGDTRSRTAEYK...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDM[649.3660]K</td>\n",
       "      <td>KEQESEVDM[655.3735]K</td>\n",
       "      <td>-1.759694</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.821773</td>\n",
       "      <td>-1.960428</td>\n",
       "      <td>-1.781592</td>\n",
       "      <td>-1.988603</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|Q8K3J1|NDUS8_MOUSE</td>\n",
       "      <td>Q8K3J1</td>\n",
       "      <td>NDUS8_MOUSE</td>\n",
       "      <td>Ndufs8</td>\n",
       "      <td>NADH dehydrogenase [ubiquinone] iron-sulfur pr...</td>\n",
       "      <td>7.034064e-14</td>\n",
       "      <td>13.152794</td>\n",
       "      <td>-1.871741</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MYRLSSSMLPRALAQAMRTGHLNGQSLHSSAVAATYKYVNKKEQES...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVM[649.3660]LAVDAVIAELK</td>\n",
       "      <td>RGVM[655.3735]LAVDAVIAELK</td>\n",
       "      <td>-1.812349</td>\n",
       "      <td>-1.980371</td>\n",
       "      <td>-2.307386</td>\n",
       "      <td>-2.492858</td>\n",
       "      <td>-1.405837</td>\n",
       "      <td>-2.094407</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|P63038|CH60_MOUSE</td>\n",
       "      <td>P63038</td>\n",
       "      <td>CH60_MOUSE</td>\n",
       "      <td>Hspd1</td>\n",
       "      <td>60 kDa heat shock protein, mitochondrial</td>\n",
       "      <td>7.289026e-11</td>\n",
       "      <td>10.137330</td>\n",
       "      <td>-1.950460</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MLRLPTVLRQMRPVSRALAPHLTRAYAKDVKFGADARALMLQGVDL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>M[649.3660]QLLEIITTDK</td>\n",
       "      <td>M[655.3735]QLLEIITTDK</td>\n",
       "      <td>-2.286954</td>\n",
       "      <td>-2.700322</td>\n",
       "      <td>-1.941018</td>\n",
       "      <td>-1.860306</td>\n",
       "      <td>-2.069488</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|Q8BMS1|ECHA_MOUSE</td>\n",
       "      <td>Q8BMS1</td>\n",
       "      <td>ECHA_MOUSE</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondrial</td>\n",
       "      <td>5.582467e-09</td>\n",
       "      <td>8.253174</td>\n",
       "      <td>-2.270072</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MVASRAIGSLSRFSAFRILRSRGCICRSFTTSSALLTRTHINYGVK...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAM[649.3660]R</td>\n",
       "      <td>DHFEEAM[655.3735]R</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.891986</td>\n",
       "      <td>-2.129614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>sp|Q01853|TERA_MOUSE</td>\n",
       "      <td>Q01853</td>\n",
       "      <td>TERA_MOUSE</td>\n",
       "      <td>Vcp</td>\n",
       "      <td>Transitional endoplasmic reticulum ATPase</td>\n",
       "      <td>6.484059e-04</td>\n",
       "      <td>3.188153</td>\n",
       "      <td>-2.507726</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MASGADSKGDDLSTAILKQKNRPNRLIVDEAINEDNSVVSLSQPKM...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>203 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Peptide Sequence      Modified Peptide  \\\n",
       "0            AADTIGYPVMIR          AADTIGYPVMIR   \n",
       "1              IAMQTLDMGR            IAMQTLDMGR   \n",
       "2             FVGAVDPIMEK           FVGAVDPIMEK   \n",
       "3    QAQYLGMPINGPFKPDHYRY  QAQYLGMPINGPFKPDHYRY   \n",
       "4            FADVIPMNLPHR          FADVIPMNLPHR   \n",
       "..                    ...                   ...   \n",
       "198     AHMVTLDYTVQVPGTGR     AHMVTLDYTVQVPGTGR   \n",
       "199            KEQESEVDMK            KEQESEVDMK   \n",
       "200       RGVMLAVDAVIAELK       RGVMLAVDAVIAELK   \n",
       "201           MQLLEIITTDK           MQLLEIITTDK   \n",
       "202              DHFEEAMR              DHFEEAMR   \n",
       "\n",
       "             Light Modified Peptide          Heavy Modified Peptide  \\\n",
       "0            AADTIGYPVM[649.3660]IR          AADTIGYPVM[655.3735]IR   \n",
       "1              IAM[649.3660]QTLDMGR            IAM[655.3735]QTLDMGR   \n",
       "2             FVGAVDPIM[649.3660]EK           FVGAVDPIM[655.3735]EK   \n",
       "3    QAQYLGM[649.3660]PINGPFKPDHYRY  QAQYLGM[655.3735]PINGPFKPDHYRY   \n",
       "4            FADVIPM[649.3660]NLPHR          FADVIPM[655.3735]NLPHR   \n",
       "..                              ...                             ...   \n",
       "198     AHM[649.3660]VTLDYTVQVPGTGR     AHM[655.3735]VTLDYTVQVPGTGR   \n",
       "199            KEQESEVDM[649.3660]K            KEQESEVDM[655.3735]K   \n",
       "200       RGVM[649.3660]LAVDAVIAELK       RGVM[655.3735]LAVDAVIAELK   \n",
       "201           M[649.3660]QLLEIITTDK           M[655.3735]QLLEIITTDK   \n",
       "202              DHFEEAM[649.3660]R              DHFEEAM[655.3735]R   \n",
       "\n",
       "     1 Log2 Ratio HL  10 Log2 Ratio HL  11 Log2 Ratio HL  12 Log2 Ratio HL  \\\n",
       "0                NaN               NaN          3.269016               NaN   \n",
       "1                NaN          2.783695          3.114945          2.697822   \n",
       "2                NaN               NaN               NaN               NaN   \n",
       "3           2.394458          2.380664          2.682897               NaN   \n",
       "4                NaN               NaN          2.580647               NaN   \n",
       "..               ...               ...               ...               ...   \n",
       "198        -1.822067         -1.687501         -1.706146         -1.647231   \n",
       "199        -1.759694               NaN         -1.821773         -1.960428   \n",
       "200        -1.812349         -1.980371         -2.307386         -2.492858   \n",
       "201        -2.286954         -2.700322         -1.941018         -1.860306   \n",
       "202              NaN         -2.891986         -2.129614               NaN   \n",
       "\n",
       "     2 Log2 Ratio HL  3 Log2 Ratio HL  ...                Protein  Protein ID  \\\n",
       "0                NaN              NaN  ...   sp|Q8C196|CPSM_MOUSE      Q8C196   \n",
       "1                NaN              NaN  ...  sp|Q07417|ACADS_MOUSE      Q07417   \n",
       "2                NaN         2.383482  ...   sp|Q91YI0|ARLY_MOUSE      Q91YI0   \n",
       "3           2.435014         2.412394  ...   sp|P50247|SAHH_MOUSE      P50247   \n",
       "4           2.446614         2.450718  ...  sp|P33267|CP2F2_MOUSE      P33267   \n",
       "..               ...              ...  ...                    ...         ...   \n",
       "198        -1.793966        -1.407199  ...   sp|Q9QXF8|GNMT_MOUSE      Q9QXF8   \n",
       "199        -1.781592        -1.988603  ...  sp|Q8K3J1|NDUS8_MOUSE      Q8K3J1   \n",
       "200        -1.405837        -2.094407  ...   sp|P63038|CH60_MOUSE      P63038   \n",
       "201        -2.069488              NaN  ...   sp|Q8BMS1|ECHA_MOUSE      Q8BMS1   \n",
       "202              NaN              NaN  ...   sp|Q01853|TERA_MOUSE      Q01853   \n",
       "\n",
       "      Entry Name    Gene                                Protein Description  \\\n",
       "0     CPSM_MOUSE    Cps1  Carbamoyl-phosphate synthase [ammonia], mitoch...   \n",
       "1    ACADS_MOUSE   Acads  Short-chain specific acyl-CoA dehydrogenase, m...   \n",
       "2     ARLY_MOUSE     Asl                            Argininosuccinate lyase   \n",
       "3     SAHH_MOUSE    Ahcy                             Adenosylhomocysteinase   \n",
       "4    CP2F2_MOUSE  Cyp2f2                                Cytochrome P450 2F2   \n",
       "..           ...     ...                                                ...   \n",
       "198   GNMT_MOUSE    Gnmt                        Glycine N-methyltransferase   \n",
       "199  NDUS8_MOUSE  Ndufs8  NADH dehydrogenase [ubiquinone] iron-sulfur pr...   \n",
       "200   CH60_MOUSE   Hspd1           60 kDa heat shock protein, mitochondrial   \n",
       "201   ECHA_MOUSE   Hadha  Trifunctional enzyme subunit alpha, mitochondrial   \n",
       "202   TERA_MOUSE     Vcp          Transitional endoplasmic reticulum ATPase   \n",
       "\n",
       "          p-value    neglogp Log2HL avg   label  \\\n",
       "0    4.058191e-03   2.391668   3.289988   green   \n",
       "1    9.647931e-06   5.015566   2.789195   green   \n",
       "2    4.283587e-02   1.368192   2.555706   green   \n",
       "3    8.561078e-10   9.067472   2.538998   green   \n",
       "4    2.032232e-04   3.692027   2.389057   green   \n",
       "..            ...        ...        ...     ...   \n",
       "198  1.880219e-08   7.725792  -1.857135  yellow   \n",
       "199  7.034064e-14  13.152794  -1.871741  yellow   \n",
       "200  7.289026e-11  10.137330  -1.950460  yellow   \n",
       "201  5.582467e-09   8.253174  -2.270072  yellow   \n",
       "202  6.484059e-04   3.188153  -2.507726  yellow   \n",
       "\n",
       "                                     Complete Sequence  \n",
       "0    MTRILTACKVVKTLKSGFGFANVTTKRQWDFSRPGIRLLSVKAKTA...  \n",
       "1    MAAALLARARGPLRRALGVRDWRRLHTVYQSVELPETHQMLRQTCR...  \n",
       "2    MASESGKLWGGRFVGAVDPIMEKFNSSISYDRHLWNVDVQGSKAYS...  \n",
       "3    MSDKLPYKVADIGLAAWGRKALDIAENEMPGLMRMREMYSASKPLK...  \n",
       "4    MDGVSTAILLLLLAVISLSLTFSSRGKGQLPPGPKPLPILGNLLQL...  \n",
       "..                                                 ...  \n",
       "198  MVDSVYRTRSLGVAAEGLPDQYADGEAARVWQLYIGDTRSRTAEYK...  \n",
       "199  MYRLSSSMLPRALAQAMRTGHLNGQSLHSSAVAATYKYVNKKEQES...  \n",
       "200  MLRLPTVLRQMRPVSRALAPHLTRAYAKDVKFGADARALMLQGVDL...  \n",
       "201  MVASRAIGSLSRFSAFRILRSRGCICRSFTTSSALLTRTHINYGVK...  \n",
       "202  MASGADSKGDDLSTAILKQKNRPNRLIVDEAINEDNSVVSLSQPKM...  \n",
       "\n",
       "[203 rows x 26 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "path = os.path.join(datasets_path, \"RvsS_peptides_completed_sequence.csv\")\n",
    "peptides_completed_sequence = pd.read_csv(path)\n",
    "peptides_completed_sequence.set_index(\"Unnamed: 0\", inplace=True)\n",
    "peptides_completed_sequence.index.name = None\n",
    "peptides_completed_sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "M\\[649\\.3660\\]|M\\[655\\.3735\\]\n"
     ]
    }
   ],
   "source": [
    "# create regex pattern to identify desired modifications\n",
    "def create_modifications_pattern(modifications):\n",
    "\n",
    "    split_mod = modifications[0].split(\".\")\n",
    "    whole = split_mod[0]\n",
    "    mantissa = split_mod[1]\n",
    "    pattern = r\"M\\[{}\\.{}\\]\".format(whole, mantissa)\n",
    "\n",
    "    for i in range(1, len(modifications)):\n",
    "        split_mod = modifications[i].split(\".\")\n",
    "        whole = split_mod[0]\n",
    "        mantissa = split_mod[1]\n",
    "        pattern += r\"|M\\[{}\\.{}\\]\".format(whole, mantissa)\n",
    "    \n",
    "    return pattern\n",
    "\n",
    "modifications_pattern = create_modifications_pattern(modifications)\n",
    "print(modifications_pattern)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide Sequence</th>\n",
       "      <th>Modified Peptide</th>\n",
       "      <th>Light Modified Peptide</th>\n",
       "      <th>Heavy Modified Peptide</th>\n",
       "      <th>1 Log2 Ratio HL</th>\n",
       "      <th>10 Log2 Ratio HL</th>\n",
       "      <th>11 Log2 Ratio HL</th>\n",
       "      <th>12 Log2 Ratio HL</th>\n",
       "      <th>2 Log2 Ratio HL</th>\n",
       "      <th>3 Log2 Ratio HL</th>\n",
       "      <th>...</th>\n",
       "      <th>Protein ID</th>\n",
       "      <th>Entry Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Protein Description</th>\n",
       "      <th>p-value</th>\n",
       "      <th>neglogp</th>\n",
       "      <th>Log2HL avg</th>\n",
       "      <th>label</th>\n",
       "      <th>Complete Sequence</th>\n",
       "      <th>Sequence Location</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVM[649.3660]IR</td>\n",
       "      <td>AADTIGYPVM[655.3735]IR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.269016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Q8C196</td>\n",
       "      <td>CPSM_MOUSE</td>\n",
       "      <td>Cps1</td>\n",
       "      <td>Carbamoyl-phosphate synthase [ammonia], mitoch...</td>\n",
       "      <td>4.058191e-03</td>\n",
       "      <td>2.391668</td>\n",
       "      <td>3.289988</td>\n",
       "      <td>green</td>\n",
       "      <td>MTRILTACKVVKTLKSGFGFANVTTKRQWDFSRPGIRLLSVKAKTA...</td>\n",
       "      <td>575</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAM[649.3660]QTLDMGR</td>\n",
       "      <td>IAM[655.3735]QTLDMGR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.783695</td>\n",
       "      <td>3.114945</td>\n",
       "      <td>2.697822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Q07417</td>\n",
       "      <td>ACADS_MOUSE</td>\n",
       "      <td>Acads</td>\n",
       "      <td>Short-chain specific acyl-CoA dehydrogenase, m...</td>\n",
       "      <td>9.647931e-06</td>\n",
       "      <td>5.015566</td>\n",
       "      <td>2.789195</td>\n",
       "      <td>green</td>\n",
       "      <td>MAAALLARARGPLRRALGVRDWRRLHTVYQSVELPETHQMLRQTCR...</td>\n",
       "      <td>262</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIM[649.3660]EK</td>\n",
       "      <td>FVGAVDPIM[655.3735]EK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.383482</td>\n",
       "      <td>...</td>\n",
       "      <td>Q91YI0</td>\n",
       "      <td>ARLY_MOUSE</td>\n",
       "      <td>Asl</td>\n",
       "      <td>Argininosuccinate lyase</td>\n",
       "      <td>4.283587e-02</td>\n",
       "      <td>1.368192</td>\n",
       "      <td>2.555706</td>\n",
       "      <td>green</td>\n",
       "      <td>MASESGKLWGGRFVGAVDPIMEKFNSSISYDRHLWNVDVQGSKAYS...</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[649.3660]PINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[655.3735]PINGPFKPDHYRY</td>\n",
       "      <td>2.394458</td>\n",
       "      <td>2.380664</td>\n",
       "      <td>2.682897</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.435014</td>\n",
       "      <td>2.412394</td>\n",
       "      <td>...</td>\n",
       "      <td>P50247</td>\n",
       "      <td>SAHH_MOUSE</td>\n",
       "      <td>Ahcy</td>\n",
       "      <td>Adenosylhomocysteinase</td>\n",
       "      <td>8.561078e-10</td>\n",
       "      <td>9.067472</td>\n",
       "      <td>2.538998</td>\n",
       "      <td>green</td>\n",
       "      <td>MSDKLPYKVADIGLAAWGRKALDIAENEMPGLMRMREMYSASKPLK...</td>\n",
       "      <td>412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPM[649.3660]NLPHR</td>\n",
       "      <td>FADVIPM[655.3735]NLPHR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.580647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.446614</td>\n",
       "      <td>2.450718</td>\n",
       "      <td>...</td>\n",
       "      <td>P33267</td>\n",
       "      <td>CP2F2_MOUSE</td>\n",
       "      <td>Cyp2f2</td>\n",
       "      <td>Cytochrome P450 2F2</td>\n",
       "      <td>2.032232e-04</td>\n",
       "      <td>3.692027</td>\n",
       "      <td>2.389057</td>\n",
       "      <td>green</td>\n",
       "      <td>MDGVSTAILLLLLAVISLSLTFSSRGKGQLPPGPKPLPILGNLLQL...</td>\n",
       "      <td>358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[649.3660]VTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[655.3735]VTLDYTVQVPGTGR</td>\n",
       "      <td>-1.822067</td>\n",
       "      <td>-1.687501</td>\n",
       "      <td>-1.706146</td>\n",
       "      <td>-1.647231</td>\n",
       "      <td>-1.793966</td>\n",
       "      <td>-1.407199</td>\n",
       "      <td>...</td>\n",
       "      <td>Q9QXF8</td>\n",
       "      <td>GNMT_MOUSE</td>\n",
       "      <td>Gnmt</td>\n",
       "      <td>Glycine N-methyltransferase</td>\n",
       "      <td>1.880219e-08</td>\n",
       "      <td>7.725792</td>\n",
       "      <td>-1.857135</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MVDSVYRTRSLGVAAEGLPDQYADGEAARVWQLYIGDTRSRTAEYK...</td>\n",
       "      <td>213</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDM[649.3660]K</td>\n",
       "      <td>KEQESEVDM[655.3735]K</td>\n",
       "      <td>-1.759694</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.821773</td>\n",
       "      <td>-1.960428</td>\n",
       "      <td>-1.781592</td>\n",
       "      <td>-1.988603</td>\n",
       "      <td>...</td>\n",
       "      <td>Q8K3J1</td>\n",
       "      <td>NDUS8_MOUSE</td>\n",
       "      <td>Ndufs8</td>\n",
       "      <td>NADH dehydrogenase [ubiquinone] iron-sulfur pr...</td>\n",
       "      <td>7.034064e-14</td>\n",
       "      <td>13.152794</td>\n",
       "      <td>-1.871741</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MYRLSSSMLPRALAQAMRTGHLNGQSLHSSAVAATYKYVNKKEQES...</td>\n",
       "      <td>41</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVM[649.3660]LAVDAVIAELK</td>\n",
       "      <td>RGVM[655.3735]LAVDAVIAELK</td>\n",
       "      <td>-1.812349</td>\n",
       "      <td>-1.980371</td>\n",
       "      <td>-2.307386</td>\n",
       "      <td>-2.492858</td>\n",
       "      <td>-1.405837</td>\n",
       "      <td>-2.094407</td>\n",
       "      <td>...</td>\n",
       "      <td>P63038</td>\n",
       "      <td>CH60_MOUSE</td>\n",
       "      <td>Hspd1</td>\n",
       "      <td>60 kDa heat shock protein, mitochondrial</td>\n",
       "      <td>7.289026e-11</td>\n",
       "      <td>10.137330</td>\n",
       "      <td>-1.950460</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MLRLPTVLRQMRPVSRALAPHLTRAYAKDVKFGADARALMLQGVDL...</td>\n",
       "      <td>141</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>M[649.3660]QLLEIITTDK</td>\n",
       "      <td>M[655.3735]QLLEIITTDK</td>\n",
       "      <td>-2.286954</td>\n",
       "      <td>-2.700322</td>\n",
       "      <td>-1.941018</td>\n",
       "      <td>-1.860306</td>\n",
       "      <td>-2.069488</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Q8BMS1</td>\n",
       "      <td>ECHA_MOUSE</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondrial</td>\n",
       "      <td>5.582467e-09</td>\n",
       "      <td>8.253174</td>\n",
       "      <td>-2.270072</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MVASRAIGSLSRFSAFRILRSRGCICRSFTTSSALLTRTHINYGVK...</td>\n",
       "      <td>505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAM[649.3660]R</td>\n",
       "      <td>DHFEEAM[655.3735]R</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.891986</td>\n",
       "      <td>-2.129614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Q01853</td>\n",
       "      <td>TERA_MOUSE</td>\n",
       "      <td>Vcp</td>\n",
       "      <td>Transitional endoplasmic reticulum ATPase</td>\n",
       "      <td>6.484059e-04</td>\n",
       "      <td>3.188153</td>\n",
       "      <td>-2.507726</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MASGADSKGDDLSTAILKQKNRPNRLIVDEAINEDNSVVSLSQPKM...</td>\n",
       "      <td>733</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>203 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Peptide Sequence      Modified Peptide  \\\n",
       "0            AADTIGYPVMIR          AADTIGYPVMIR   \n",
       "1              IAMQTLDMGR            IAMQTLDMGR   \n",
       "2             FVGAVDPIMEK           FVGAVDPIMEK   \n",
       "3    QAQYLGMPINGPFKPDHYRY  QAQYLGMPINGPFKPDHYRY   \n",
       "4            FADVIPMNLPHR          FADVIPMNLPHR   \n",
       "..                    ...                   ...   \n",
       "198     AHMVTLDYTVQVPGTGR     AHMVTLDYTVQVPGTGR   \n",
       "199            KEQESEVDMK            KEQESEVDMK   \n",
       "200       RGVMLAVDAVIAELK       RGVMLAVDAVIAELK   \n",
       "201           MQLLEIITTDK           MQLLEIITTDK   \n",
       "202              DHFEEAMR              DHFEEAMR   \n",
       "\n",
       "             Light Modified Peptide          Heavy Modified Peptide  \\\n",
       "0            AADTIGYPVM[649.3660]IR          AADTIGYPVM[655.3735]IR   \n",
       "1              IAM[649.3660]QTLDMGR            IAM[655.3735]QTLDMGR   \n",
       "2             FVGAVDPIM[649.3660]EK           FVGAVDPIM[655.3735]EK   \n",
       "3    QAQYLGM[649.3660]PINGPFKPDHYRY  QAQYLGM[655.3735]PINGPFKPDHYRY   \n",
       "4            FADVIPM[649.3660]NLPHR          FADVIPM[655.3735]NLPHR   \n",
       "..                              ...                             ...   \n",
       "198     AHM[649.3660]VTLDYTVQVPGTGR     AHM[655.3735]VTLDYTVQVPGTGR   \n",
       "199            KEQESEVDM[649.3660]K            KEQESEVDM[655.3735]K   \n",
       "200       RGVM[649.3660]LAVDAVIAELK       RGVM[655.3735]LAVDAVIAELK   \n",
       "201           M[649.3660]QLLEIITTDK           M[655.3735]QLLEIITTDK   \n",
       "202              DHFEEAM[649.3660]R              DHFEEAM[655.3735]R   \n",
       "\n",
       "     1 Log2 Ratio HL  10 Log2 Ratio HL  11 Log2 Ratio HL  12 Log2 Ratio HL  \\\n",
       "0                NaN               NaN          3.269016               NaN   \n",
       "1                NaN          2.783695          3.114945          2.697822   \n",
       "2                NaN               NaN               NaN               NaN   \n",
       "3           2.394458          2.380664          2.682897               NaN   \n",
       "4                NaN               NaN          2.580647               NaN   \n",
       "..               ...               ...               ...               ...   \n",
       "198        -1.822067         -1.687501         -1.706146         -1.647231   \n",
       "199        -1.759694               NaN         -1.821773         -1.960428   \n",
       "200        -1.812349         -1.980371         -2.307386         -2.492858   \n",
       "201        -2.286954         -2.700322         -1.941018         -1.860306   \n",
       "202              NaN         -2.891986         -2.129614               NaN   \n",
       "\n",
       "     2 Log2 Ratio HL  3 Log2 Ratio HL  ...  Protein ID   Entry Name    Gene  \\\n",
       "0                NaN              NaN  ...      Q8C196   CPSM_MOUSE    Cps1   \n",
       "1                NaN              NaN  ...      Q07417  ACADS_MOUSE   Acads   \n",
       "2                NaN         2.383482  ...      Q91YI0   ARLY_MOUSE     Asl   \n",
       "3           2.435014         2.412394  ...      P50247   SAHH_MOUSE    Ahcy   \n",
       "4           2.446614         2.450718  ...      P33267  CP2F2_MOUSE  Cyp2f2   \n",
       "..               ...              ...  ...         ...          ...     ...   \n",
       "198        -1.793966        -1.407199  ...      Q9QXF8   GNMT_MOUSE    Gnmt   \n",
       "199        -1.781592        -1.988603  ...      Q8K3J1  NDUS8_MOUSE  Ndufs8   \n",
       "200        -1.405837        -2.094407  ...      P63038   CH60_MOUSE   Hspd1   \n",
       "201        -2.069488              NaN  ...      Q8BMS1   ECHA_MOUSE   Hadha   \n",
       "202              NaN              NaN  ...      Q01853   TERA_MOUSE     Vcp   \n",
       "\n",
       "                                   Protein Description       p-value  \\\n",
       "0    Carbamoyl-phosphate synthase [ammonia], mitoch...  4.058191e-03   \n",
       "1    Short-chain specific acyl-CoA dehydrogenase, m...  9.647931e-06   \n",
       "2                              Argininosuccinate lyase  4.283587e-02   \n",
       "3                               Adenosylhomocysteinase  8.561078e-10   \n",
       "4                                  Cytochrome P450 2F2  2.032232e-04   \n",
       "..                                                 ...           ...   \n",
       "198                        Glycine N-methyltransferase  1.880219e-08   \n",
       "199  NADH dehydrogenase [ubiquinone] iron-sulfur pr...  7.034064e-14   \n",
       "200           60 kDa heat shock protein, mitochondrial  7.289026e-11   \n",
       "201  Trifunctional enzyme subunit alpha, mitochondrial  5.582467e-09   \n",
       "202          Transitional endoplasmic reticulum ATPase  6.484059e-04   \n",
       "\n",
       "       neglogp Log2HL avg   label  \\\n",
       "0     2.391668   3.289988   green   \n",
       "1     5.015566   2.789195   green   \n",
       "2     1.368192   2.555706   green   \n",
       "3     9.067472   2.538998   green   \n",
       "4     3.692027   2.389057   green   \n",
       "..         ...        ...     ...   \n",
       "198   7.725792  -1.857135  yellow   \n",
       "199  13.152794  -1.871741  yellow   \n",
       "200  10.137330  -1.950460  yellow   \n",
       "201   8.253174  -2.270072  yellow   \n",
       "202   3.188153  -2.507726  yellow   \n",
       "\n",
       "                                     Complete Sequence Sequence Location  \n",
       "0    MTRILTACKVVKTLKSGFGFANVTTKRQWDFSRPGIRLLSVKAKTA...               575  \n",
       "1    MAAALLARARGPLRRALGVRDWRRLHTVYQSVELPETHQMLRQTCR...               262  \n",
       "2    MASESGKLWGGRFVGAVDPIMEKFNSSISYDRHLWNVDVQGSKAYS...                12  \n",
       "3    MSDKLPYKVADIGLAAWGRKALDIAENEMPGLMRMREMYSASKPLK...               412  \n",
       "4    MDGVSTAILLLLLAVISLSLTFSSRGKGQLPPGPKPLPILGNLLQL...               358  \n",
       "..                                                 ...               ...  \n",
       "198  MVDSVYRTRSLGVAAEGLPDQYADGEAARVWQLYIGDTRSRTAEYK...               213  \n",
       "199  MYRLSSSMLPRALAQAMRTGHLNGQSLHSSAVAATYKYVNKKEQES...                41  \n",
       "200  MLRLPTVLRQMRPVSRALAPHLTRAYAKDVKFGADARALMLQGVDL...               141  \n",
       "201  MVASRAIGSLSRFSAFRILRSRGCICRSFTTSSALLTRTHINYGVK...               505  \n",
       "202  MASGADSKGDDLSTAILKQKNRPNRLIVDEAINEDNSVVSLSQPKM...               733  \n",
       "\n",
       "[203 rows x 27 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peptides_completed_sequence[\"Sequence Location\"] = pd.Series([a.find(b) for a, b in zip(peptides_completed_sequence[\"Complete Sequence\"], peptides_completed_sequence[\"Peptide Sequence\"])])\n",
    "peptides_completed_sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide Sequence</th>\n",
       "      <th>Modified Peptide</th>\n",
       "      <th>Light Modified Peptide</th>\n",
       "      <th>Heavy Modified Peptide</th>\n",
       "      <th>1 Log2 Ratio HL</th>\n",
       "      <th>10 Log2 Ratio HL</th>\n",
       "      <th>11 Log2 Ratio HL</th>\n",
       "      <th>12 Log2 Ratio HL</th>\n",
       "      <th>2 Log2 Ratio HL</th>\n",
       "      <th>3 Log2 Ratio HL</th>\n",
       "      <th>...</th>\n",
       "      <th>Entry Name</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Protein Description</th>\n",
       "      <th>p-value</th>\n",
       "      <th>neglogp</th>\n",
       "      <th>Log2HL avg</th>\n",
       "      <th>label</th>\n",
       "      <th>Complete Sequence</th>\n",
       "      <th>Sequence Location</th>\n",
       "      <th>Sequence Length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVM[649.3660]IR</td>\n",
       "      <td>AADTIGYPVM[655.3735]IR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.269016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>CPSM_MOUSE</td>\n",
       "      <td>Cps1</td>\n",
       "      <td>Carbamoyl-phosphate synthase [ammonia], mitoch...</td>\n",
       "      <td>4.058191e-03</td>\n",
       "      <td>2.391668</td>\n",
       "      <td>3.289988</td>\n",
       "      <td>green</td>\n",
       "      <td>MTRILTACKVVKTLKSGFGFANVTTKRQWDFSRPGIRLLSVKAKTA...</td>\n",
       "      <td>575</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAM[649.3660]QTLDMGR</td>\n",
       "      <td>IAM[655.3735]QTLDMGR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.783695</td>\n",
       "      <td>3.114945</td>\n",
       "      <td>2.697822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ACADS_MOUSE</td>\n",
       "      <td>Acads</td>\n",
       "      <td>Short-chain specific acyl-CoA dehydrogenase, m...</td>\n",
       "      <td>9.647931e-06</td>\n",
       "      <td>5.015566</td>\n",
       "      <td>2.789195</td>\n",
       "      <td>green</td>\n",
       "      <td>MAAALLARARGPLRRALGVRDWRRLHTVYQSVELPETHQMLRQTCR...</td>\n",
       "      <td>262</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIM[649.3660]EK</td>\n",
       "      <td>FVGAVDPIM[655.3735]EK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.383482</td>\n",
       "      <td>...</td>\n",
       "      <td>ARLY_MOUSE</td>\n",
       "      <td>Asl</td>\n",
       "      <td>Argininosuccinate lyase</td>\n",
       "      <td>4.283587e-02</td>\n",
       "      <td>1.368192</td>\n",
       "      <td>2.555706</td>\n",
       "      <td>green</td>\n",
       "      <td>MASESGKLWGGRFVGAVDPIMEKFNSSISYDRHLWNVDVQGSKAYS...</td>\n",
       "      <td>12</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[649.3660]PINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[655.3735]PINGPFKPDHYRY</td>\n",
       "      <td>2.394458</td>\n",
       "      <td>2.380664</td>\n",
       "      <td>2.682897</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.435014</td>\n",
       "      <td>2.412394</td>\n",
       "      <td>...</td>\n",
       "      <td>SAHH_MOUSE</td>\n",
       "      <td>Ahcy</td>\n",
       "      <td>Adenosylhomocysteinase</td>\n",
       "      <td>8.561078e-10</td>\n",
       "      <td>9.067472</td>\n",
       "      <td>2.538998</td>\n",
       "      <td>green</td>\n",
       "      <td>MSDKLPYKVADIGLAAWGRKALDIAENEMPGLMRMREMYSASKPLK...</td>\n",
       "      <td>412</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPM[649.3660]NLPHR</td>\n",
       "      <td>FADVIPM[655.3735]NLPHR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.580647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.446614</td>\n",
       "      <td>2.450718</td>\n",
       "      <td>...</td>\n",
       "      <td>CP2F2_MOUSE</td>\n",
       "      <td>Cyp2f2</td>\n",
       "      <td>Cytochrome P450 2F2</td>\n",
       "      <td>2.032232e-04</td>\n",
       "      <td>3.692027</td>\n",
       "      <td>2.389057</td>\n",
       "      <td>green</td>\n",
       "      <td>MDGVSTAILLLLLAVISLSLTFSSRGKGQLPPGPKPLPILGNLLQL...</td>\n",
       "      <td>358</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[649.3660]VTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[655.3735]VTLDYTVQVPGTGR</td>\n",
       "      <td>-1.822067</td>\n",
       "      <td>-1.687501</td>\n",
       "      <td>-1.706146</td>\n",
       "      <td>-1.647231</td>\n",
       "      <td>-1.793966</td>\n",
       "      <td>-1.407199</td>\n",
       "      <td>...</td>\n",
       "      <td>GNMT_MOUSE</td>\n",
       "      <td>Gnmt</td>\n",
       "      <td>Glycine N-methyltransferase</td>\n",
       "      <td>1.880219e-08</td>\n",
       "      <td>7.725792</td>\n",
       "      <td>-1.857135</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MVDSVYRTRSLGVAAEGLPDQYADGEAARVWQLYIGDTRSRTAEYK...</td>\n",
       "      <td>213</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDM[649.3660]K</td>\n",
       "      <td>KEQESEVDM[655.3735]K</td>\n",
       "      <td>-1.759694</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.821773</td>\n",
       "      <td>-1.960428</td>\n",
       "      <td>-1.781592</td>\n",
       "      <td>-1.988603</td>\n",
       "      <td>...</td>\n",
       "      <td>NDUS8_MOUSE</td>\n",
       "      <td>Ndufs8</td>\n",
       "      <td>NADH dehydrogenase [ubiquinone] iron-sulfur pr...</td>\n",
       "      <td>7.034064e-14</td>\n",
       "      <td>13.152794</td>\n",
       "      <td>-1.871741</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MYRLSSSMLPRALAQAMRTGHLNGQSLHSSAVAATYKYVNKKEQES...</td>\n",
       "      <td>41</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVM[649.3660]LAVDAVIAELK</td>\n",
       "      <td>RGVM[655.3735]LAVDAVIAELK</td>\n",
       "      <td>-1.812349</td>\n",
       "      <td>-1.980371</td>\n",
       "      <td>-2.307386</td>\n",
       "      <td>-2.492858</td>\n",
       "      <td>-1.405837</td>\n",
       "      <td>-2.094407</td>\n",
       "      <td>...</td>\n",
       "      <td>CH60_MOUSE</td>\n",
       "      <td>Hspd1</td>\n",
       "      <td>60 kDa heat shock protein, mitochondrial</td>\n",
       "      <td>7.289026e-11</td>\n",
       "      <td>10.137330</td>\n",
       "      <td>-1.950460</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MLRLPTVLRQMRPVSRALAPHLTRAYAKDVKFGADARALMLQGVDL...</td>\n",
       "      <td>141</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>M[649.3660]QLLEIITTDK</td>\n",
       "      <td>M[655.3735]QLLEIITTDK</td>\n",
       "      <td>-2.286954</td>\n",
       "      <td>-2.700322</td>\n",
       "      <td>-1.941018</td>\n",
       "      <td>-1.860306</td>\n",
       "      <td>-2.069488</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>ECHA_MOUSE</td>\n",
       "      <td>Hadha</td>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondrial</td>\n",
       "      <td>5.582467e-09</td>\n",
       "      <td>8.253174</td>\n",
       "      <td>-2.270072</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MVASRAIGSLSRFSAFRILRSRGCICRSFTTSSALLTRTHINYGVK...</td>\n",
       "      <td>505</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAM[649.3660]R</td>\n",
       "      <td>DHFEEAM[655.3735]R</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.891986</td>\n",
       "      <td>-2.129614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>TERA_MOUSE</td>\n",
       "      <td>Vcp</td>\n",
       "      <td>Transitional endoplasmic reticulum ATPase</td>\n",
       "      <td>6.484059e-04</td>\n",
       "      <td>3.188153</td>\n",
       "      <td>-2.507726</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MASGADSKGDDLSTAILKQKNRPNRLIVDEAINEDNSVVSLSQPKM...</td>\n",
       "      <td>733</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>203 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Peptide Sequence      Modified Peptide  \\\n",
       "0            AADTIGYPVMIR          AADTIGYPVMIR   \n",
       "1              IAMQTLDMGR            IAMQTLDMGR   \n",
       "2             FVGAVDPIMEK           FVGAVDPIMEK   \n",
       "3    QAQYLGMPINGPFKPDHYRY  QAQYLGMPINGPFKPDHYRY   \n",
       "4            FADVIPMNLPHR          FADVIPMNLPHR   \n",
       "..                    ...                   ...   \n",
       "198     AHMVTLDYTVQVPGTGR     AHMVTLDYTVQVPGTGR   \n",
       "199            KEQESEVDMK            KEQESEVDMK   \n",
       "200       RGVMLAVDAVIAELK       RGVMLAVDAVIAELK   \n",
       "201           MQLLEIITTDK           MQLLEIITTDK   \n",
       "202              DHFEEAMR              DHFEEAMR   \n",
       "\n",
       "             Light Modified Peptide          Heavy Modified Peptide  \\\n",
       "0            AADTIGYPVM[649.3660]IR          AADTIGYPVM[655.3735]IR   \n",
       "1              IAM[649.3660]QTLDMGR            IAM[655.3735]QTLDMGR   \n",
       "2             FVGAVDPIM[649.3660]EK           FVGAVDPIM[655.3735]EK   \n",
       "3    QAQYLGM[649.3660]PINGPFKPDHYRY  QAQYLGM[655.3735]PINGPFKPDHYRY   \n",
       "4            FADVIPM[649.3660]NLPHR          FADVIPM[655.3735]NLPHR   \n",
       "..                              ...                             ...   \n",
       "198     AHM[649.3660]VTLDYTVQVPGTGR     AHM[655.3735]VTLDYTVQVPGTGR   \n",
       "199            KEQESEVDM[649.3660]K            KEQESEVDM[655.3735]K   \n",
       "200       RGVM[649.3660]LAVDAVIAELK       RGVM[655.3735]LAVDAVIAELK   \n",
       "201           M[649.3660]QLLEIITTDK           M[655.3735]QLLEIITTDK   \n",
       "202              DHFEEAM[649.3660]R              DHFEEAM[655.3735]R   \n",
       "\n",
       "     1 Log2 Ratio HL  10 Log2 Ratio HL  11 Log2 Ratio HL  12 Log2 Ratio HL  \\\n",
       "0                NaN               NaN          3.269016               NaN   \n",
       "1                NaN          2.783695          3.114945          2.697822   \n",
       "2                NaN               NaN               NaN               NaN   \n",
       "3           2.394458          2.380664          2.682897               NaN   \n",
       "4                NaN               NaN          2.580647               NaN   \n",
       "..               ...               ...               ...               ...   \n",
       "198        -1.822067         -1.687501         -1.706146         -1.647231   \n",
       "199        -1.759694               NaN         -1.821773         -1.960428   \n",
       "200        -1.812349         -1.980371         -2.307386         -2.492858   \n",
       "201        -2.286954         -2.700322         -1.941018         -1.860306   \n",
       "202              NaN         -2.891986         -2.129614               NaN   \n",
       "\n",
       "     2 Log2 Ratio HL  3 Log2 Ratio HL  ...   Entry Name    Gene  \\\n",
       "0                NaN              NaN  ...   CPSM_MOUSE    Cps1   \n",
       "1                NaN              NaN  ...  ACADS_MOUSE   Acads   \n",
       "2                NaN         2.383482  ...   ARLY_MOUSE     Asl   \n",
       "3           2.435014         2.412394  ...   SAHH_MOUSE    Ahcy   \n",
       "4           2.446614         2.450718  ...  CP2F2_MOUSE  Cyp2f2   \n",
       "..               ...              ...  ...          ...     ...   \n",
       "198        -1.793966        -1.407199  ...   GNMT_MOUSE    Gnmt   \n",
       "199        -1.781592        -1.988603  ...  NDUS8_MOUSE  Ndufs8   \n",
       "200        -1.405837        -2.094407  ...   CH60_MOUSE   Hspd1   \n",
       "201        -2.069488              NaN  ...   ECHA_MOUSE   Hadha   \n",
       "202              NaN              NaN  ...   TERA_MOUSE     Vcp   \n",
       "\n",
       "                                   Protein Description       p-value  \\\n",
       "0    Carbamoyl-phosphate synthase [ammonia], mitoch...  4.058191e-03   \n",
       "1    Short-chain specific acyl-CoA dehydrogenase, m...  9.647931e-06   \n",
       "2                              Argininosuccinate lyase  4.283587e-02   \n",
       "3                               Adenosylhomocysteinase  8.561078e-10   \n",
       "4                                  Cytochrome P450 2F2  2.032232e-04   \n",
       "..                                                 ...           ...   \n",
       "198                        Glycine N-methyltransferase  1.880219e-08   \n",
       "199  NADH dehydrogenase [ubiquinone] iron-sulfur pr...  7.034064e-14   \n",
       "200           60 kDa heat shock protein, mitochondrial  7.289026e-11   \n",
       "201  Trifunctional enzyme subunit alpha, mitochondrial  5.582467e-09   \n",
       "202          Transitional endoplasmic reticulum ATPase  6.484059e-04   \n",
       "\n",
       "       neglogp  Log2HL avg   label  \\\n",
       "0     2.391668    3.289988   green   \n",
       "1     5.015566    2.789195   green   \n",
       "2     1.368192    2.555706   green   \n",
       "3     9.067472    2.538998   green   \n",
       "4     3.692027    2.389057   green   \n",
       "..         ...         ...     ...   \n",
       "198   7.725792   -1.857135  yellow   \n",
       "199  13.152794   -1.871741  yellow   \n",
       "200  10.137330   -1.950460  yellow   \n",
       "201   8.253174   -2.270072  yellow   \n",
       "202   3.188153   -2.507726  yellow   \n",
       "\n",
       "                                     Complete Sequence Sequence Location  \\\n",
       "0    MTRILTACKVVKTLKSGFGFANVTTKRQWDFSRPGIRLLSVKAKTA...               575   \n",
       "1    MAAALLARARGPLRRALGVRDWRRLHTVYQSVELPETHQMLRQTCR...               262   \n",
       "2    MASESGKLWGGRFVGAVDPIMEKFNSSISYDRHLWNVDVQGSKAYS...                12   \n",
       "3    MSDKLPYKVADIGLAAWGRKALDIAENEMPGLMRMREMYSASKPLK...               412   \n",
       "4    MDGVSTAILLLLLAVISLSLTFSSRGKGQLPPGPKPLPILGNLLQL...               358   \n",
       "..                                                 ...               ...   \n",
       "198  MVDSVYRTRSLGVAAEGLPDQYADGEAARVWQLYIGDTRSRTAEYK...               213   \n",
       "199  MYRLSSSMLPRALAQAMRTGHLNGQSLHSSAVAATYKYVNKKEQES...                41   \n",
       "200  MLRLPTVLRQMRPVSRALAPHLTRAYAKDVKFGADARALMLQGVDL...               141   \n",
       "201  MVASRAIGSLSRFSAFRILRSRGCICRSFTTSSALLTRTHINYGVK...               505   \n",
       "202  MASGADSKGDDLSTAILKQKNRPNRLIVDEAINEDNSVVSLSQPKM...               733   \n",
       "\n",
       "    Sequence Length  \n",
       "0                12  \n",
       "1                10  \n",
       "2                11  \n",
       "3                20  \n",
       "4                12  \n",
       "..              ...  \n",
       "198              17  \n",
       "199              10  \n",
       "200              15  \n",
       "201              11  \n",
       "202               8  \n",
       "\n",
       "[203 rows x 28 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peptides_completed_sequence[\"Sequence Length\"] = peptides_completed_sequence[\"Peptide Sequence\"].str.len()\n",
    "peptides_completed_sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Peptide Sequence\n",
       "True    203\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "temp = [A[B:B+C] for A, B, C in zip(peptides_completed_sequence[\"Complete Sequence\"], peptides_completed_sequence[\"Sequence Location\"], peptides_completed_sequence[\"Sequence Length\"])]\n",
    "(temp == peptides_completed_sequence[\"Peptide Sequence\"]).value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(.*)(M\\[649\\.3660\\]|M\\[655\\.3735\\])\n"
     ]
    }
   ],
   "source": [
    "# create regex pattern to identify desired modifications\n",
    "left_prefix_pattern = \"(.*)(\" + modifications_pattern + \")\"\n",
    "print(left_prefix_pattern)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide Sequence</th>\n",
       "      <th>Modified Peptide</th>\n",
       "      <th>Light Modified Peptide</th>\n",
       "      <th>Heavy Modified Peptide</th>\n",
       "      <th>1 Log2 Ratio HL</th>\n",
       "      <th>10 Log2 Ratio HL</th>\n",
       "      <th>11 Log2 Ratio HL</th>\n",
       "      <th>12 Log2 Ratio HL</th>\n",
       "      <th>2 Log2 Ratio HL</th>\n",
       "      <th>3 Log2 Ratio HL</th>\n",
       "      <th>...</th>\n",
       "      <th>Protein Description</th>\n",
       "      <th>p-value</th>\n",
       "      <th>neglogp</th>\n",
       "      <th>Log2HL avg</th>\n",
       "      <th>label</th>\n",
       "      <th>Complete Sequence</th>\n",
       "      <th>Sequence Location</th>\n",
       "      <th>Sequence Length</th>\n",
       "      <th>Left Prefix</th>\n",
       "      <th>Left Prefix Length</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVM[649.3660]IR</td>\n",
       "      <td>AADTIGYPVM[655.3735]IR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.269016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Carbamoyl-phosphate synthase [ammonia], mitoch...</td>\n",
       "      <td>4.058191e-03</td>\n",
       "      <td>2.391668</td>\n",
       "      <td>3.289988</td>\n",
       "      <td>green</td>\n",
       "      <td>MTRILTACKVVKTLKSGFGFANVTTKRQWDFSRPGIRLLSVKAKTA...</td>\n",
       "      <td>575</td>\n",
       "      <td>12</td>\n",
       "      <td>AADTIGYPV</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAM[649.3660]QTLDMGR</td>\n",
       "      <td>IAM[655.3735]QTLDMGR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.783695</td>\n",
       "      <td>3.114945</td>\n",
       "      <td>2.697822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Short-chain specific acyl-CoA dehydrogenase, m...</td>\n",
       "      <td>9.647931e-06</td>\n",
       "      <td>5.015566</td>\n",
       "      <td>2.789195</td>\n",
       "      <td>green</td>\n",
       "      <td>MAAALLARARGPLRRALGVRDWRRLHTVYQSVELPETHQMLRQTCR...</td>\n",
       "      <td>262</td>\n",
       "      <td>10</td>\n",
       "      <td>IA</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIM[649.3660]EK</td>\n",
       "      <td>FVGAVDPIM[655.3735]EK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.383482</td>\n",
       "      <td>...</td>\n",
       "      <td>Argininosuccinate lyase</td>\n",
       "      <td>4.283587e-02</td>\n",
       "      <td>1.368192</td>\n",
       "      <td>2.555706</td>\n",
       "      <td>green</td>\n",
       "      <td>MASESGKLWGGRFVGAVDPIMEKFNSSISYDRHLWNVDVQGSKAYS...</td>\n",
       "      <td>12</td>\n",
       "      <td>11</td>\n",
       "      <td>FVGAVDPI</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[649.3660]PINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[655.3735]PINGPFKPDHYRY</td>\n",
       "      <td>2.394458</td>\n",
       "      <td>2.380664</td>\n",
       "      <td>2.682897</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.435014</td>\n",
       "      <td>2.412394</td>\n",
       "      <td>...</td>\n",
       "      <td>Adenosylhomocysteinase</td>\n",
       "      <td>8.561078e-10</td>\n",
       "      <td>9.067472</td>\n",
       "      <td>2.538998</td>\n",
       "      <td>green</td>\n",
       "      <td>MSDKLPYKVADIGLAAWGRKALDIAENEMPGLMRMREMYSASKPLK...</td>\n",
       "      <td>412</td>\n",
       "      <td>20</td>\n",
       "      <td>QAQYLG</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPM[649.3660]NLPHR</td>\n",
       "      <td>FADVIPM[655.3735]NLPHR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.580647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.446614</td>\n",
       "      <td>2.450718</td>\n",
       "      <td>...</td>\n",
       "      <td>Cytochrome P450 2F2</td>\n",
       "      <td>2.032232e-04</td>\n",
       "      <td>3.692027</td>\n",
       "      <td>2.389057</td>\n",
       "      <td>green</td>\n",
       "      <td>MDGVSTAILLLLLAVISLSLTFSSRGKGQLPPGPKPLPILGNLLQL...</td>\n",
       "      <td>358</td>\n",
       "      <td>12</td>\n",
       "      <td>FADVIP</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[649.3660]VTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[655.3735]VTLDYTVQVPGTGR</td>\n",
       "      <td>-1.822067</td>\n",
       "      <td>-1.687501</td>\n",
       "      <td>-1.706146</td>\n",
       "      <td>-1.647231</td>\n",
       "      <td>-1.793966</td>\n",
       "      <td>-1.407199</td>\n",
       "      <td>...</td>\n",
       "      <td>Glycine N-methyltransferase</td>\n",
       "      <td>1.880219e-08</td>\n",
       "      <td>7.725792</td>\n",
       "      <td>-1.857135</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MVDSVYRTRSLGVAAEGLPDQYADGEAARVWQLYIGDTRSRTAEYK...</td>\n",
       "      <td>213</td>\n",
       "      <td>17</td>\n",
       "      <td>AH</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDM[649.3660]K</td>\n",
       "      <td>KEQESEVDM[655.3735]K</td>\n",
       "      <td>-1.759694</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.821773</td>\n",
       "      <td>-1.960428</td>\n",
       "      <td>-1.781592</td>\n",
       "      <td>-1.988603</td>\n",
       "      <td>...</td>\n",
       "      <td>NADH dehydrogenase [ubiquinone] iron-sulfur pr...</td>\n",
       "      <td>7.034064e-14</td>\n",
       "      <td>13.152794</td>\n",
       "      <td>-1.871741</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MYRLSSSMLPRALAQAMRTGHLNGQSLHSSAVAATYKYVNKKEQES...</td>\n",
       "      <td>41</td>\n",
       "      <td>10</td>\n",
       "      <td>KEQESEVD</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVM[649.3660]LAVDAVIAELK</td>\n",
       "      <td>RGVM[655.3735]LAVDAVIAELK</td>\n",
       "      <td>-1.812349</td>\n",
       "      <td>-1.980371</td>\n",
       "      <td>-2.307386</td>\n",
       "      <td>-2.492858</td>\n",
       "      <td>-1.405837</td>\n",
       "      <td>-2.094407</td>\n",
       "      <td>...</td>\n",
       "      <td>60 kDa heat shock protein, mitochondrial</td>\n",
       "      <td>7.289026e-11</td>\n",
       "      <td>10.137330</td>\n",
       "      <td>-1.950460</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MLRLPTVLRQMRPVSRALAPHLTRAYAKDVKFGADARALMLQGVDL...</td>\n",
       "      <td>141</td>\n",
       "      <td>15</td>\n",
       "      <td>RGV</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>M[649.3660]QLLEIITTDK</td>\n",
       "      <td>M[655.3735]QLLEIITTDK</td>\n",
       "      <td>-2.286954</td>\n",
       "      <td>-2.700322</td>\n",
       "      <td>-1.941018</td>\n",
       "      <td>-1.860306</td>\n",
       "      <td>-2.069488</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Trifunctional enzyme subunit alpha, mitochondrial</td>\n",
       "      <td>5.582467e-09</td>\n",
       "      <td>8.253174</td>\n",
       "      <td>-2.270072</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MVASRAIGSLSRFSAFRILRSRGCICRSFTTSSALLTRTHINYGVK...</td>\n",
       "      <td>505</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAM[649.3660]R</td>\n",
       "      <td>DHFEEAM[655.3735]R</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.891986</td>\n",
       "      <td>-2.129614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Transitional endoplasmic reticulum ATPase</td>\n",
       "      <td>6.484059e-04</td>\n",
       "      <td>3.188153</td>\n",
       "      <td>-2.507726</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MASGADSKGDDLSTAILKQKNRPNRLIVDEAINEDNSVVSLSQPKM...</td>\n",
       "      <td>733</td>\n",
       "      <td>8</td>\n",
       "      <td>DHFEEA</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>203 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Peptide Sequence      Modified Peptide  \\\n",
       "0            AADTIGYPVMIR          AADTIGYPVMIR   \n",
       "1              IAMQTLDMGR            IAMQTLDMGR   \n",
       "2             FVGAVDPIMEK           FVGAVDPIMEK   \n",
       "3    QAQYLGMPINGPFKPDHYRY  QAQYLGMPINGPFKPDHYRY   \n",
       "4            FADVIPMNLPHR          FADVIPMNLPHR   \n",
       "..                    ...                   ...   \n",
       "198     AHMVTLDYTVQVPGTGR     AHMVTLDYTVQVPGTGR   \n",
       "199            KEQESEVDMK            KEQESEVDMK   \n",
       "200       RGVMLAVDAVIAELK       RGVMLAVDAVIAELK   \n",
       "201           MQLLEIITTDK           MQLLEIITTDK   \n",
       "202              DHFEEAMR              DHFEEAMR   \n",
       "\n",
       "             Light Modified Peptide          Heavy Modified Peptide  \\\n",
       "0            AADTIGYPVM[649.3660]IR          AADTIGYPVM[655.3735]IR   \n",
       "1              IAM[649.3660]QTLDMGR            IAM[655.3735]QTLDMGR   \n",
       "2             FVGAVDPIM[649.3660]EK           FVGAVDPIM[655.3735]EK   \n",
       "3    QAQYLGM[649.3660]PINGPFKPDHYRY  QAQYLGM[655.3735]PINGPFKPDHYRY   \n",
       "4            FADVIPM[649.3660]NLPHR          FADVIPM[655.3735]NLPHR   \n",
       "..                              ...                             ...   \n",
       "198     AHM[649.3660]VTLDYTVQVPGTGR     AHM[655.3735]VTLDYTVQVPGTGR   \n",
       "199            KEQESEVDM[649.3660]K            KEQESEVDM[655.3735]K   \n",
       "200       RGVM[649.3660]LAVDAVIAELK       RGVM[655.3735]LAVDAVIAELK   \n",
       "201           M[649.3660]QLLEIITTDK           M[655.3735]QLLEIITTDK   \n",
       "202              DHFEEAM[649.3660]R              DHFEEAM[655.3735]R   \n",
       "\n",
       "     1 Log2 Ratio HL  10 Log2 Ratio HL  11 Log2 Ratio HL  12 Log2 Ratio HL  \\\n",
       "0                NaN               NaN          3.269016               NaN   \n",
       "1                NaN          2.783695          3.114945          2.697822   \n",
       "2                NaN               NaN               NaN               NaN   \n",
       "3           2.394458          2.380664          2.682897               NaN   \n",
       "4                NaN               NaN          2.580647               NaN   \n",
       "..               ...               ...               ...               ...   \n",
       "198        -1.822067         -1.687501         -1.706146         -1.647231   \n",
       "199        -1.759694               NaN         -1.821773         -1.960428   \n",
       "200        -1.812349         -1.980371         -2.307386         -2.492858   \n",
       "201        -2.286954         -2.700322         -1.941018         -1.860306   \n",
       "202              NaN         -2.891986         -2.129614               NaN   \n",
       "\n",
       "     2 Log2 Ratio HL  3 Log2 Ratio HL  ...  \\\n",
       "0                NaN              NaN  ...   \n",
       "1                NaN              NaN  ...   \n",
       "2                NaN         2.383482  ...   \n",
       "3           2.435014         2.412394  ...   \n",
       "4           2.446614         2.450718  ...   \n",
       "..               ...              ...  ...   \n",
       "198        -1.793966        -1.407199  ...   \n",
       "199        -1.781592        -1.988603  ...   \n",
       "200        -1.405837        -2.094407  ...   \n",
       "201        -2.069488              NaN  ...   \n",
       "202              NaN              NaN  ...   \n",
       "\n",
       "                                   Protein Description       p-value  \\\n",
       "0    Carbamoyl-phosphate synthase [ammonia], mitoch...  4.058191e-03   \n",
       "1    Short-chain specific acyl-CoA dehydrogenase, m...  9.647931e-06   \n",
       "2                              Argininosuccinate lyase  4.283587e-02   \n",
       "3                               Adenosylhomocysteinase  8.561078e-10   \n",
       "4                                  Cytochrome P450 2F2  2.032232e-04   \n",
       "..                                                 ...           ...   \n",
       "198                        Glycine N-methyltransferase  1.880219e-08   \n",
       "199  NADH dehydrogenase [ubiquinone] iron-sulfur pr...  7.034064e-14   \n",
       "200           60 kDa heat shock protein, mitochondrial  7.289026e-11   \n",
       "201  Trifunctional enzyme subunit alpha, mitochondrial  5.582467e-09   \n",
       "202          Transitional endoplasmic reticulum ATPase  6.484059e-04   \n",
       "\n",
       "       neglogp  Log2HL avg   label  \\\n",
       "0     2.391668    3.289988   green   \n",
       "1     5.015566    2.789195   green   \n",
       "2     1.368192    2.555706   green   \n",
       "3     9.067472    2.538998   green   \n",
       "4     3.692027    2.389057   green   \n",
       "..         ...         ...     ...   \n",
       "198   7.725792   -1.857135  yellow   \n",
       "199  13.152794   -1.871741  yellow   \n",
       "200  10.137330   -1.950460  yellow   \n",
       "201   8.253174   -2.270072  yellow   \n",
       "202   3.188153   -2.507726  yellow   \n",
       "\n",
       "                                     Complete Sequence Sequence Location  \\\n",
       "0    MTRILTACKVVKTLKSGFGFANVTTKRQWDFSRPGIRLLSVKAKTA...               575   \n",
       "1    MAAALLARARGPLRRALGVRDWRRLHTVYQSVELPETHQMLRQTCR...               262   \n",
       "2    MASESGKLWGGRFVGAVDPIMEKFNSSISYDRHLWNVDVQGSKAYS...                12   \n",
       "3    MSDKLPYKVADIGLAAWGRKALDIAENEMPGLMRMREMYSASKPLK...               412   \n",
       "4    MDGVSTAILLLLLAVISLSLTFSSRGKGQLPPGPKPLPILGNLLQL...               358   \n",
       "..                                                 ...               ...   \n",
       "198  MVDSVYRTRSLGVAAEGLPDQYADGEAARVWQLYIGDTRSRTAEYK...               213   \n",
       "199  MYRLSSSMLPRALAQAMRTGHLNGQSLHSSAVAATYKYVNKKEQES...                41   \n",
       "200  MLRLPTVLRQMRPVSRALAPHLTRAYAKDVKFGADARALMLQGVDL...               141   \n",
       "201  MVASRAIGSLSRFSAFRILRSRGCICRSFTTSSALLTRTHINYGVK...               505   \n",
       "202  MASGADSKGDDLSTAILKQKNRPNRLIVDEAINEDNSVVSLSQPKM...               733   \n",
       "\n",
       "    Sequence Length Left Prefix Left Prefix Length  \n",
       "0                12   AADTIGYPV                  9  \n",
       "1                10          IA                  2  \n",
       "2                11    FVGAVDPI                  8  \n",
       "3                20      QAQYLG                  6  \n",
       "4                12      FADVIP                  6  \n",
       "..              ...         ...                ...  \n",
       "198              17          AH                  2  \n",
       "199              10    KEQESEVD                  8  \n",
       "200              15         RGV                  3  \n",
       "201              11                              0  \n",
       "202               8      DHFEEA                  6  \n",
       "\n",
       "[203 rows x 30 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# extract left prefix of modified methionine (for indexing purposes)\n",
    "IUPACCodes = \"ACDEFGHIKLMNPQRSTVWY\"\n",
    "filtering = lambda string: ''.join([char for char in string if char in IUPACCodes])\n",
    "\n",
    "peptides_completed_sequence[\"Left Prefix\"] = peptides_completed_sequence[\"Light Modified Peptide\"].str.extract(left_prefix_pattern)[0]\n",
    "peptides_completed_sequence[\"Left Prefix\"] = peptides_completed_sequence[\"Left Prefix\"].map(filtering)\n",
    "peptides_completed_sequence[\"Left Prefix Length\"] = peptides_completed_sequence[\"Left Prefix\"].str.len()\n",
    "\n",
    "peptides_completed_sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide Sequence</th>\n",
       "      <th>Modified Peptide</th>\n",
       "      <th>Light Modified Peptide</th>\n",
       "      <th>Heavy Modified Peptide</th>\n",
       "      <th>1 Log2 Ratio HL</th>\n",
       "      <th>10 Log2 Ratio HL</th>\n",
       "      <th>11 Log2 Ratio HL</th>\n",
       "      <th>12 Log2 Ratio HL</th>\n",
       "      <th>2 Log2 Ratio HL</th>\n",
       "      <th>3 Log2 Ratio HL</th>\n",
       "      <th>...</th>\n",
       "      <th>p-value</th>\n",
       "      <th>neglogp</th>\n",
       "      <th>Log2HL avg</th>\n",
       "      <th>label</th>\n",
       "      <th>Complete Sequence</th>\n",
       "      <th>Sequence Location</th>\n",
       "      <th>Sequence Length</th>\n",
       "      <th>Left Prefix</th>\n",
       "      <th>Left Prefix Length</th>\n",
       "      <th>Methionine Location</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVM[649.3660]IR</td>\n",
       "      <td>AADTIGYPVM[655.3735]IR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.269016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>4.058191e-03</td>\n",
       "      <td>2.391668</td>\n",
       "      <td>3.289988</td>\n",
       "      <td>green</td>\n",
       "      <td>MTRILTACKVVKTLKSGFGFANVTTKRQWDFSRPGIRLLSVKAKTA...</td>\n",
       "      <td>575</td>\n",
       "      <td>12</td>\n",
       "      <td>AADTIGYPV</td>\n",
       "      <td>9</td>\n",
       "      <td>584</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAM[649.3660]QTLDMGR</td>\n",
       "      <td>IAM[655.3735]QTLDMGR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.783695</td>\n",
       "      <td>3.114945</td>\n",
       "      <td>2.697822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>9.647931e-06</td>\n",
       "      <td>5.015566</td>\n",
       "      <td>2.789195</td>\n",
       "      <td>green</td>\n",
       "      <td>MAAALLARARGPLRRALGVRDWRRLHTVYQSVELPETHQMLRQTCR...</td>\n",
       "      <td>262</td>\n",
       "      <td>10</td>\n",
       "      <td>IA</td>\n",
       "      <td>2</td>\n",
       "      <td>264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIM[649.3660]EK</td>\n",
       "      <td>FVGAVDPIM[655.3735]EK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.383482</td>\n",
       "      <td>...</td>\n",
       "      <td>4.283587e-02</td>\n",
       "      <td>1.368192</td>\n",
       "      <td>2.555706</td>\n",
       "      <td>green</td>\n",
       "      <td>MASESGKLWGGRFVGAVDPIMEKFNSSISYDRHLWNVDVQGSKAYS...</td>\n",
       "      <td>12</td>\n",
       "      <td>11</td>\n",
       "      <td>FVGAVDPI</td>\n",
       "      <td>8</td>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[649.3660]PINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[655.3735]PINGPFKPDHYRY</td>\n",
       "      <td>2.394458</td>\n",
       "      <td>2.380664</td>\n",
       "      <td>2.682897</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.435014</td>\n",
       "      <td>2.412394</td>\n",
       "      <td>...</td>\n",
       "      <td>8.561078e-10</td>\n",
       "      <td>9.067472</td>\n",
       "      <td>2.538998</td>\n",
       "      <td>green</td>\n",
       "      <td>MSDKLPYKVADIGLAAWGRKALDIAENEMPGLMRMREMYSASKPLK...</td>\n",
       "      <td>412</td>\n",
       "      <td>20</td>\n",
       "      <td>QAQYLG</td>\n",
       "      <td>6</td>\n",
       "      <td>418</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPM[649.3660]NLPHR</td>\n",
       "      <td>FADVIPM[655.3735]NLPHR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.580647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.446614</td>\n",
       "      <td>2.450718</td>\n",
       "      <td>...</td>\n",
       "      <td>2.032232e-04</td>\n",
       "      <td>3.692027</td>\n",
       "      <td>2.389057</td>\n",
       "      <td>green</td>\n",
       "      <td>MDGVSTAILLLLLAVISLSLTFSSRGKGQLPPGPKPLPILGNLLQL...</td>\n",
       "      <td>358</td>\n",
       "      <td>12</td>\n",
       "      <td>FADVIP</td>\n",
       "      <td>6</td>\n",
       "      <td>364</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[649.3660]VTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[655.3735]VTLDYTVQVPGTGR</td>\n",
       "      <td>-1.822067</td>\n",
       "      <td>-1.687501</td>\n",
       "      <td>-1.706146</td>\n",
       "      <td>-1.647231</td>\n",
       "      <td>-1.793966</td>\n",
       "      <td>-1.407199</td>\n",
       "      <td>...</td>\n",
       "      <td>1.880219e-08</td>\n",
       "      <td>7.725792</td>\n",
       "      <td>-1.857135</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MVDSVYRTRSLGVAAEGLPDQYADGEAARVWQLYIGDTRSRTAEYK...</td>\n",
       "      <td>213</td>\n",
       "      <td>17</td>\n",
       "      <td>AH</td>\n",
       "      <td>2</td>\n",
       "      <td>215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDM[649.3660]K</td>\n",
       "      <td>KEQESEVDM[655.3735]K</td>\n",
       "      <td>-1.759694</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.821773</td>\n",
       "      <td>-1.960428</td>\n",
       "      <td>-1.781592</td>\n",
       "      <td>-1.988603</td>\n",
       "      <td>...</td>\n",
       "      <td>7.034064e-14</td>\n",
       "      <td>13.152794</td>\n",
       "      <td>-1.871741</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MYRLSSSMLPRALAQAMRTGHLNGQSLHSSAVAATYKYVNKKEQES...</td>\n",
       "      <td>41</td>\n",
       "      <td>10</td>\n",
       "      <td>KEQESEVD</td>\n",
       "      <td>8</td>\n",
       "      <td>49</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVM[649.3660]LAVDAVIAELK</td>\n",
       "      <td>RGVM[655.3735]LAVDAVIAELK</td>\n",
       "      <td>-1.812349</td>\n",
       "      <td>-1.980371</td>\n",
       "      <td>-2.307386</td>\n",
       "      <td>-2.492858</td>\n",
       "      <td>-1.405837</td>\n",
       "      <td>-2.094407</td>\n",
       "      <td>...</td>\n",
       "      <td>7.289026e-11</td>\n",
       "      <td>10.137330</td>\n",
       "      <td>-1.950460</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MLRLPTVLRQMRPVSRALAPHLTRAYAKDVKFGADARALMLQGVDL...</td>\n",
       "      <td>141</td>\n",
       "      <td>15</td>\n",
       "      <td>RGV</td>\n",
       "      <td>3</td>\n",
       "      <td>144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>M[649.3660]QLLEIITTDK</td>\n",
       "      <td>M[655.3735]QLLEIITTDK</td>\n",
       "      <td>-2.286954</td>\n",
       "      <td>-2.700322</td>\n",
       "      <td>-1.941018</td>\n",
       "      <td>-1.860306</td>\n",
       "      <td>-2.069488</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>5.582467e-09</td>\n",
       "      <td>8.253174</td>\n",
       "      <td>-2.270072</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MVASRAIGSLSRFSAFRILRSRGCICRSFTTSSALLTRTHINYGVK...</td>\n",
       "      <td>505</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>505</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAM[649.3660]R</td>\n",
       "      <td>DHFEEAM[655.3735]R</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.891986</td>\n",
       "      <td>-2.129614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>6.484059e-04</td>\n",
       "      <td>3.188153</td>\n",
       "      <td>-2.507726</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MASGADSKGDDLSTAILKQKNRPNRLIVDEAINEDNSVVSLSQPKM...</td>\n",
       "      <td>733</td>\n",
       "      <td>8</td>\n",
       "      <td>DHFEEA</td>\n",
       "      <td>6</td>\n",
       "      <td>739</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>203 rows × 31 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Peptide Sequence      Modified Peptide  \\\n",
       "0            AADTIGYPVMIR          AADTIGYPVMIR   \n",
       "1              IAMQTLDMGR            IAMQTLDMGR   \n",
       "2             FVGAVDPIMEK           FVGAVDPIMEK   \n",
       "3    QAQYLGMPINGPFKPDHYRY  QAQYLGMPINGPFKPDHYRY   \n",
       "4            FADVIPMNLPHR          FADVIPMNLPHR   \n",
       "..                    ...                   ...   \n",
       "198     AHMVTLDYTVQVPGTGR     AHMVTLDYTVQVPGTGR   \n",
       "199            KEQESEVDMK            KEQESEVDMK   \n",
       "200       RGVMLAVDAVIAELK       RGVMLAVDAVIAELK   \n",
       "201           MQLLEIITTDK           MQLLEIITTDK   \n",
       "202              DHFEEAMR              DHFEEAMR   \n",
       "\n",
       "             Light Modified Peptide          Heavy Modified Peptide  \\\n",
       "0            AADTIGYPVM[649.3660]IR          AADTIGYPVM[655.3735]IR   \n",
       "1              IAM[649.3660]QTLDMGR            IAM[655.3735]QTLDMGR   \n",
       "2             FVGAVDPIM[649.3660]EK           FVGAVDPIM[655.3735]EK   \n",
       "3    QAQYLGM[649.3660]PINGPFKPDHYRY  QAQYLGM[655.3735]PINGPFKPDHYRY   \n",
       "4            FADVIPM[649.3660]NLPHR          FADVIPM[655.3735]NLPHR   \n",
       "..                              ...                             ...   \n",
       "198     AHM[649.3660]VTLDYTVQVPGTGR     AHM[655.3735]VTLDYTVQVPGTGR   \n",
       "199            KEQESEVDM[649.3660]K            KEQESEVDM[655.3735]K   \n",
       "200       RGVM[649.3660]LAVDAVIAELK       RGVM[655.3735]LAVDAVIAELK   \n",
       "201           M[649.3660]QLLEIITTDK           M[655.3735]QLLEIITTDK   \n",
       "202              DHFEEAM[649.3660]R              DHFEEAM[655.3735]R   \n",
       "\n",
       "     1 Log2 Ratio HL  10 Log2 Ratio HL  11 Log2 Ratio HL  12 Log2 Ratio HL  \\\n",
       "0                NaN               NaN          3.269016               NaN   \n",
       "1                NaN          2.783695          3.114945          2.697822   \n",
       "2                NaN               NaN               NaN               NaN   \n",
       "3           2.394458          2.380664          2.682897               NaN   \n",
       "4                NaN               NaN          2.580647               NaN   \n",
       "..               ...               ...               ...               ...   \n",
       "198        -1.822067         -1.687501         -1.706146         -1.647231   \n",
       "199        -1.759694               NaN         -1.821773         -1.960428   \n",
       "200        -1.812349         -1.980371         -2.307386         -2.492858   \n",
       "201        -2.286954         -2.700322         -1.941018         -1.860306   \n",
       "202              NaN         -2.891986         -2.129614               NaN   \n",
       "\n",
       "     2 Log2 Ratio HL  3 Log2 Ratio HL  ...       p-value    neglogp  \\\n",
       "0                NaN              NaN  ...  4.058191e-03   2.391668   \n",
       "1                NaN              NaN  ...  9.647931e-06   5.015566   \n",
       "2                NaN         2.383482  ...  4.283587e-02   1.368192   \n",
       "3           2.435014         2.412394  ...  8.561078e-10   9.067472   \n",
       "4           2.446614         2.450718  ...  2.032232e-04   3.692027   \n",
       "..               ...              ...  ...           ...        ...   \n",
       "198        -1.793966        -1.407199  ...  1.880219e-08   7.725792   \n",
       "199        -1.781592        -1.988603  ...  7.034064e-14  13.152794   \n",
       "200        -1.405837        -2.094407  ...  7.289026e-11  10.137330   \n",
       "201        -2.069488              NaN  ...  5.582467e-09   8.253174   \n",
       "202              NaN              NaN  ...  6.484059e-04   3.188153   \n",
       "\n",
       "     Log2HL avg   label                                  Complete Sequence  \\\n",
       "0      3.289988   green  MTRILTACKVVKTLKSGFGFANVTTKRQWDFSRPGIRLLSVKAKTA...   \n",
       "1      2.789195   green  MAAALLARARGPLRRALGVRDWRRLHTVYQSVELPETHQMLRQTCR...   \n",
       "2      2.555706   green  MASESGKLWGGRFVGAVDPIMEKFNSSISYDRHLWNVDVQGSKAYS...   \n",
       "3      2.538998   green  MSDKLPYKVADIGLAAWGRKALDIAENEMPGLMRMREMYSASKPLK...   \n",
       "4      2.389057   green  MDGVSTAILLLLLAVISLSLTFSSRGKGQLPPGPKPLPILGNLLQL...   \n",
       "..          ...     ...                                                ...   \n",
       "198   -1.857135  yellow  MVDSVYRTRSLGVAAEGLPDQYADGEAARVWQLYIGDTRSRTAEYK...   \n",
       "199   -1.871741  yellow  MYRLSSSMLPRALAQAMRTGHLNGQSLHSSAVAATYKYVNKKEQES...   \n",
       "200   -1.950460  yellow  MLRLPTVLRQMRPVSRALAPHLTRAYAKDVKFGADARALMLQGVDL...   \n",
       "201   -2.270072  yellow  MVASRAIGSLSRFSAFRILRSRGCICRSFTTSSALLTRTHINYGVK...   \n",
       "202   -2.507726  yellow  MASGADSKGDDLSTAILKQKNRPNRLIVDEAINEDNSVVSLSQPKM...   \n",
       "\n",
       "     Sequence Location Sequence Length Left Prefix Left Prefix Length  \\\n",
       "0                  575              12   AADTIGYPV                  9   \n",
       "1                  262              10          IA                  2   \n",
       "2                   12              11    FVGAVDPI                  8   \n",
       "3                  412              20      QAQYLG                  6   \n",
       "4                  358              12      FADVIP                  6   \n",
       "..                 ...             ...         ...                ...   \n",
       "198                213              17          AH                  2   \n",
       "199                 41              10    KEQESEVD                  8   \n",
       "200                141              15         RGV                  3   \n",
       "201                505              11                              0   \n",
       "202                733               8      DHFEEA                  6   \n",
       "\n",
       "    Methionine Location  \n",
       "0                   584  \n",
       "1                   264  \n",
       "2                    20  \n",
       "3                   418  \n",
       "4                   364  \n",
       "..                  ...  \n",
       "198                 215  \n",
       "199                  49  \n",
       "200                 144  \n",
       "201                 505  \n",
       "202                 739  \n",
       "\n",
       "[203 rows x 31 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "peptides_completed_sequence[\"Methionine Location\"] = peptides_completed_sequence[\"Sequence Location\"] + peptides_completed_sequence[\"Left Prefix Length\"]\n",
    "peptides_completed_sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide Sequence</th>\n",
       "      <th>Modified Peptide</th>\n",
       "      <th>Light Modified Peptide</th>\n",
       "      <th>Heavy Modified Peptide</th>\n",
       "      <th>1 Log2 Ratio HL</th>\n",
       "      <th>10 Log2 Ratio HL</th>\n",
       "      <th>11 Log2 Ratio HL</th>\n",
       "      <th>12 Log2 Ratio HL</th>\n",
       "      <th>2 Log2 Ratio HL</th>\n",
       "      <th>3 Log2 Ratio HL</th>\n",
       "      <th>...</th>\n",
       "      <th>Log2HL avg</th>\n",
       "      <th>label</th>\n",
       "      <th>Complete Sequence</th>\n",
       "      <th>Sequence Location</th>\n",
       "      <th>Sequence Length</th>\n",
       "      <th>Left Prefix</th>\n",
       "      <th>Left Prefix Length</th>\n",
       "      <th>Methionine Location</th>\n",
       "      <th>Left 20</th>\n",
       "      <th>Right 20</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVM[649.3660]IR</td>\n",
       "      <td>AADTIGYPVM[655.3735]IR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.269016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>3.289988</td>\n",
       "      <td>green</td>\n",
       "      <td>MTRILTACKVVKTLKSGFGFANVTTKRQWDFSRPGIRLLSVKAKTA...</td>\n",
       "      <td>575</td>\n",
       "      <td>12</td>\n",
       "      <td>AADTIGYPV</td>\n",
       "      <td>9</td>\n",
       "      <td>584</td>\n",
       "      <td>SFAVESMEDALKAADTIGYP</td>\n",
       "      <td>IRSAYALGGLGSGICPNKET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAM[649.3660]QTLDMGR</td>\n",
       "      <td>IAM[655.3735]QTLDMGR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.783695</td>\n",
       "      <td>3.114945</td>\n",
       "      <td>2.697822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>2.789195</td>\n",
       "      <td>green</td>\n",
       "      <td>MAAALLARARGPLRRALGVRDWRRLHTVYQSVELPETHQMLRQTCR...</td>\n",
       "      <td>262</td>\n",
       "      <td>10</td>\n",
       "      <td>IA</td>\n",
       "      <td>2</td>\n",
       "      <td>264</td>\n",
       "      <td>EDCRIPKENLLGEPGMGFKI</td>\n",
       "      <td>QTLDMGRIGIASQALGIAQA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIM[649.3660]EK</td>\n",
       "      <td>FVGAVDPIM[655.3735]EK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.383482</td>\n",
       "      <td>...</td>\n",
       "      <td>2.555706</td>\n",
       "      <td>green</td>\n",
       "      <td>MASESGKLWGGRFVGAVDPIMEKFNSSISYDRHLWNVDVQGSKAYS...</td>\n",
       "      <td>12</td>\n",
       "      <td>11</td>\n",
       "      <td>FVGAVDPI</td>\n",
       "      <td>8</td>\n",
       "      <td>20</td>\n",
       "      <td>MASESGKLWGGRFVGAVDP</td>\n",
       "      <td>EKFNSSISYDRHLWNVDVQG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[649.3660]PINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[655.3735]PINGPFKPDHYRY</td>\n",
       "      <td>2.394458</td>\n",
       "      <td>2.380664</td>\n",
       "      <td>2.682897</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.435014</td>\n",
       "      <td>2.412394</td>\n",
       "      <td>...</td>\n",
       "      <td>2.538998</td>\n",
       "      <td>green</td>\n",
       "      <td>MSDKLPYKVADIGLAAWGRKALDIAENEMPGLMRMREMYSASKPLK...</td>\n",
       "      <td>412</td>\n",
       "      <td>20</td>\n",
       "      <td>QAQYLG</td>\n",
       "      <td>6</td>\n",
       "      <td>418</td>\n",
       "      <td>HLGKLNVKLTKLTEKQAQYL</td>\n",
       "      <td>PINGPFKPDHYRY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPM[649.3660]NLPHR</td>\n",
       "      <td>FADVIPM[655.3735]NLPHR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.580647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.446614</td>\n",
       "      <td>2.450718</td>\n",
       "      <td>...</td>\n",
       "      <td>2.389057</td>\n",
       "      <td>green</td>\n",
       "      <td>MDGVSTAILLLLLAVISLSLTFSSRGKGQLPPGPKPLPILGNLLQL...</td>\n",
       "      <td>358</td>\n",
       "      <td>12</td>\n",
       "      <td>FADVIP</td>\n",
       "      <td>6</td>\n",
       "      <td>364</td>\n",
       "      <td>TSMPYTDAVIHEVQRFADVI</td>\n",
       "      <td>NLPHRVTRDTPFRGFLIPKG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[649.3660]VTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[655.3735]VTLDYTVQVPGTGR</td>\n",
       "      <td>-1.822067</td>\n",
       "      <td>-1.687501</td>\n",
       "      <td>-1.706146</td>\n",
       "      <td>-1.647231</td>\n",
       "      <td>-1.793966</td>\n",
       "      <td>-1.407199</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.857135</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MVDSVYRTRSLGVAAEGLPDQYADGEAARVWQLYIGDTRSRTAEYK...</td>\n",
       "      <td>213</td>\n",
       "      <td>17</td>\n",
       "      <td>AH</td>\n",
       "      <td>2</td>\n",
       "      <td>215</td>\n",
       "      <td>YKSDLTKDITTSVLTVNNKA</td>\n",
       "      <td>VTLDYTVQVPGTGRDGSPGF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDM[649.3660]K</td>\n",
       "      <td>KEQESEVDM[655.3735]K</td>\n",
       "      <td>-1.759694</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.821773</td>\n",
       "      <td>-1.960428</td>\n",
       "      <td>-1.781592</td>\n",
       "      <td>-1.988603</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.871741</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MYRLSSSMLPRALAQAMRTGHLNGQSLHSSAVAATYKYVNKKEQES...</td>\n",
       "      <td>41</td>\n",
       "      <td>10</td>\n",
       "      <td>KEQESEVD</td>\n",
       "      <td>8</td>\n",
       "      <td>49</td>\n",
       "      <td>SSAVAATYKYVNKKEQESEV</td>\n",
       "      <td>KSATDNAARILMWTELIRGL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVM[649.3660]LAVDAVIAELK</td>\n",
       "      <td>RGVM[655.3735]LAVDAVIAELK</td>\n",
       "      <td>-1.812349</td>\n",
       "      <td>-1.980371</td>\n",
       "      <td>-2.307386</td>\n",
       "      <td>-2.492858</td>\n",
       "      <td>-1.405837</td>\n",
       "      <td>-2.094407</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.950460</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MLRLPTVLRQMRPVSRALAPHLTRAYAKDVKFGADARALMLQGVDL...</td>\n",
       "      <td>141</td>\n",
       "      <td>15</td>\n",
       "      <td>RGV</td>\n",
       "      <td>3</td>\n",
       "      <td>144</td>\n",
       "      <td>AKEGFEKISKGANPVEIRRG</td>\n",
       "      <td>LAVDAVIAELKKQSKPVTTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>M[649.3660]QLLEIITTDK</td>\n",
       "      <td>M[655.3735]QLLEIITTDK</td>\n",
       "      <td>-2.286954</td>\n",
       "      <td>-2.700322</td>\n",
       "      <td>-1.941018</td>\n",
       "      <td>-1.860306</td>\n",
       "      <td>-2.069488</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.270072</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MVASRAIGSLSRFSAFRILRSRGCICRSFTTSSALLTRTHINYGVK...</td>\n",
       "      <td>505</td>\n",
       "      <td>11</td>\n",
       "      <td></td>\n",
       "      <td>0</td>\n",
       "      <td>505</td>\n",
       "      <td>AAVSKRPEKVIGMHYFSPVD</td>\n",
       "      <td>QLLEIITTDKTSKDTTASAV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAM[649.3660]R</td>\n",
       "      <td>DHFEEAM[655.3735]R</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.891986</td>\n",
       "      <td>-2.129614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.507726</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MASGADSKGDDLSTAILKQKNRPNRLIVDEAINEDNSVVSLSQPKM...</td>\n",
       "      <td>733</td>\n",
       "      <td>8</td>\n",
       "      <td>DHFEEA</td>\n",
       "      <td>6</td>\n",
       "      <td>739</td>\n",
       "      <td>AMEVEEDDPVPEIRRDHFEE</td>\n",
       "      <td>RFARRSVSDNDIRKYEMFAQ</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>203 rows × 33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Peptide Sequence      Modified Peptide  \\\n",
       "0            AADTIGYPVMIR          AADTIGYPVMIR   \n",
       "1              IAMQTLDMGR            IAMQTLDMGR   \n",
       "2             FVGAVDPIMEK           FVGAVDPIMEK   \n",
       "3    QAQYLGMPINGPFKPDHYRY  QAQYLGMPINGPFKPDHYRY   \n",
       "4            FADVIPMNLPHR          FADVIPMNLPHR   \n",
       "..                    ...                   ...   \n",
       "198     AHMVTLDYTVQVPGTGR     AHMVTLDYTVQVPGTGR   \n",
       "199            KEQESEVDMK            KEQESEVDMK   \n",
       "200       RGVMLAVDAVIAELK       RGVMLAVDAVIAELK   \n",
       "201           MQLLEIITTDK           MQLLEIITTDK   \n",
       "202              DHFEEAMR              DHFEEAMR   \n",
       "\n",
       "             Light Modified Peptide          Heavy Modified Peptide  \\\n",
       "0            AADTIGYPVM[649.3660]IR          AADTIGYPVM[655.3735]IR   \n",
       "1              IAM[649.3660]QTLDMGR            IAM[655.3735]QTLDMGR   \n",
       "2             FVGAVDPIM[649.3660]EK           FVGAVDPIM[655.3735]EK   \n",
       "3    QAQYLGM[649.3660]PINGPFKPDHYRY  QAQYLGM[655.3735]PINGPFKPDHYRY   \n",
       "4            FADVIPM[649.3660]NLPHR          FADVIPM[655.3735]NLPHR   \n",
       "..                              ...                             ...   \n",
       "198     AHM[649.3660]VTLDYTVQVPGTGR     AHM[655.3735]VTLDYTVQVPGTGR   \n",
       "199            KEQESEVDM[649.3660]K            KEQESEVDM[655.3735]K   \n",
       "200       RGVM[649.3660]LAVDAVIAELK       RGVM[655.3735]LAVDAVIAELK   \n",
       "201           M[649.3660]QLLEIITTDK           M[655.3735]QLLEIITTDK   \n",
       "202              DHFEEAM[649.3660]R              DHFEEAM[655.3735]R   \n",
       "\n",
       "     1 Log2 Ratio HL  10 Log2 Ratio HL  11 Log2 Ratio HL  12 Log2 Ratio HL  \\\n",
       "0                NaN               NaN          3.269016               NaN   \n",
       "1                NaN          2.783695          3.114945          2.697822   \n",
       "2                NaN               NaN               NaN               NaN   \n",
       "3           2.394458          2.380664          2.682897               NaN   \n",
       "4                NaN               NaN          2.580647               NaN   \n",
       "..               ...               ...               ...               ...   \n",
       "198        -1.822067         -1.687501         -1.706146         -1.647231   \n",
       "199        -1.759694               NaN         -1.821773         -1.960428   \n",
       "200        -1.812349         -1.980371         -2.307386         -2.492858   \n",
       "201        -2.286954         -2.700322         -1.941018         -1.860306   \n",
       "202              NaN         -2.891986         -2.129614               NaN   \n",
       "\n",
       "     2 Log2 Ratio HL  3 Log2 Ratio HL  ...  Log2HL avg   label  \\\n",
       "0                NaN              NaN  ...    3.289988   green   \n",
       "1                NaN              NaN  ...    2.789195   green   \n",
       "2                NaN         2.383482  ...    2.555706   green   \n",
       "3           2.435014         2.412394  ...    2.538998   green   \n",
       "4           2.446614         2.450718  ...    2.389057   green   \n",
       "..               ...              ...  ...         ...     ...   \n",
       "198        -1.793966        -1.407199  ...   -1.857135  yellow   \n",
       "199        -1.781592        -1.988603  ...   -1.871741  yellow   \n",
       "200        -1.405837        -2.094407  ...   -1.950460  yellow   \n",
       "201        -2.069488              NaN  ...   -2.270072  yellow   \n",
       "202              NaN              NaN  ...   -2.507726  yellow   \n",
       "\n",
       "                                     Complete Sequence  Sequence Location  \\\n",
       "0    MTRILTACKVVKTLKSGFGFANVTTKRQWDFSRPGIRLLSVKAKTA...                575   \n",
       "1    MAAALLARARGPLRRALGVRDWRRLHTVYQSVELPETHQMLRQTCR...                262   \n",
       "2    MASESGKLWGGRFVGAVDPIMEKFNSSISYDRHLWNVDVQGSKAYS...                 12   \n",
       "3    MSDKLPYKVADIGLAAWGRKALDIAENEMPGLMRMREMYSASKPLK...                412   \n",
       "4    MDGVSTAILLLLLAVISLSLTFSSRGKGQLPPGPKPLPILGNLLQL...                358   \n",
       "..                                                 ...                ...   \n",
       "198  MVDSVYRTRSLGVAAEGLPDQYADGEAARVWQLYIGDTRSRTAEYK...                213   \n",
       "199  MYRLSSSMLPRALAQAMRTGHLNGQSLHSSAVAATYKYVNKKEQES...                 41   \n",
       "200  MLRLPTVLRQMRPVSRALAPHLTRAYAKDVKFGADARALMLQGVDL...                141   \n",
       "201  MVASRAIGSLSRFSAFRILRSRGCICRSFTTSSALLTRTHINYGVK...                505   \n",
       "202  MASGADSKGDDLSTAILKQKNRPNRLIVDEAINEDNSVVSLSQPKM...                733   \n",
       "\n",
       "     Sequence Length  Left Prefix Left Prefix Length Methionine Location  \\\n",
       "0                 12    AADTIGYPV                  9                 584   \n",
       "1                 10           IA                  2                 264   \n",
       "2                 11     FVGAVDPI                  8                  20   \n",
       "3                 20       QAQYLG                  6                 418   \n",
       "4                 12       FADVIP                  6                 364   \n",
       "..               ...          ...                ...                 ...   \n",
       "198               17           AH                  2                 215   \n",
       "199               10     KEQESEVD                  8                  49   \n",
       "200               15          RGV                  3                 144   \n",
       "201               11                               0                 505   \n",
       "202                8       DHFEEA                  6                 739   \n",
       "\n",
       "                  Left 20              Right 20  \n",
       "0    SFAVESMEDALKAADTIGYP  IRSAYALGGLGSGICPNKET  \n",
       "1    EDCRIPKENLLGEPGMGFKI  QTLDMGRIGIASQALGIAQA  \n",
       "2     MASESGKLWGGRFVGAVDP  EKFNSSISYDRHLWNVDVQG  \n",
       "3    HLGKLNVKLTKLTEKQAQYL         PINGPFKPDHYRY  \n",
       "4    TSMPYTDAVIHEVQRFADVI  NLPHRVTRDTPFRGFLIPKG  \n",
       "..                    ...                   ...  \n",
       "198  YKSDLTKDITTSVLTVNNKA  VTLDYTVQVPGTGRDGSPGF  \n",
       "199  SSAVAATYKYVNKKEQESEV  KSATDNAARILMWTELIRGL  \n",
       "200  AKEGFEKISKGANPVEIRRG  LAVDAVIAELKKQSKPVTTP  \n",
       "201  AAVSKRPEKVIGMHYFSPVD  QLLEIITTDKTSKDTTASAV  \n",
       "202  AMEVEEDDPVPEIRRDHFEE  RFARRSVSDNDIRKYEMFAQ  \n",
       "\n",
       "[203 rows x 33 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Compute left/right analysis sequences based on threshold\n",
    "peptides_completed_sequence[f\"Left {analysis_threshold}\"] = [A[B-1-analysis_threshold:B-1]  if (B - 1 - analysis_threshold >= 0) else A[0:B-1] for A, B in zip(peptides_completed_sequence[\"Complete Sequence\"], peptides_completed_sequence[\"Methionine Location\"])]\n",
    "peptides_completed_sequence[f\"Right {analysis_threshold}\"] = [A[B+1:B+1+analysis_threshold] for A, B in zip(peptides_completed_sequence[\"Complete Sequence\"], peptides_completed_sequence[\"Methionine Location\"])]\n",
    "peptides_completed_sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "#peptides_completed_sequence.to_csv(os.path.join(datasets_path, \"RvsS_peptides_completed_sequence_with_thresholds.csv\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide Sequence</th>\n",
       "      <th>Modified Peptide</th>\n",
       "      <th>Light Modified Peptide</th>\n",
       "      <th>Heavy Modified Peptide</th>\n",
       "      <th>1 Log2 Ratio HL</th>\n",
       "      <th>10 Log2 Ratio HL</th>\n",
       "      <th>11 Log2 Ratio HL</th>\n",
       "      <th>12 Log2 Ratio HL</th>\n",
       "      <th>2 Log2 Ratio HL</th>\n",
       "      <th>3 Log2 Ratio HL</th>\n",
       "      <th>...</th>\n",
       "      <th>Log2HL avg</th>\n",
       "      <th>label</th>\n",
       "      <th>Complete Sequence</th>\n",
       "      <th>Sequence Location</th>\n",
       "      <th>Sequence Length</th>\n",
       "      <th>Left Prefix</th>\n",
       "      <th>Left Prefix Length</th>\n",
       "      <th>Methionine Location</th>\n",
       "      <th>Left 20</th>\n",
       "      <th>Right 20</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVM[649.3660]IR</td>\n",
       "      <td>AADTIGYPVM[655.3735]IR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.269016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>3.289988</td>\n",
       "      <td>green</td>\n",
       "      <td>MTRILTACKVVKTLKSGFGFANVTTKRQWDFSRPGIRLLSVKAKTA...</td>\n",
       "      <td>575</td>\n",
       "      <td>12</td>\n",
       "      <td>AADTIGYPV</td>\n",
       "      <td>9</td>\n",
       "      <td>584</td>\n",
       "      <td>SFAVESMEDALKAADTIGYP</td>\n",
       "      <td>IRSAYALGGLGSGICPNKET</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAM[649.3660]QTLDMGR</td>\n",
       "      <td>IAM[655.3735]QTLDMGR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.783695</td>\n",
       "      <td>3.114945</td>\n",
       "      <td>2.697822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>2.789195</td>\n",
       "      <td>green</td>\n",
       "      <td>MAAALLARARGPLRRALGVRDWRRLHTVYQSVELPETHQMLRQTCR...</td>\n",
       "      <td>262</td>\n",
       "      <td>10</td>\n",
       "      <td>IA</td>\n",
       "      <td>2</td>\n",
       "      <td>264</td>\n",
       "      <td>EDCRIPKENLLGEPGMGFKI</td>\n",
       "      <td>QTLDMGRIGIASQALGIAQA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIM[649.3660]EK</td>\n",
       "      <td>FVGAVDPIM[655.3735]EK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.383482</td>\n",
       "      <td>...</td>\n",
       "      <td>2.555706</td>\n",
       "      <td>green</td>\n",
       "      <td>MASESGKLWGGRFVGAVDPIMEKFNSSISYDRHLWNVDVQGSKAYS...</td>\n",
       "      <td>12</td>\n",
       "      <td>11</td>\n",
       "      <td>FVGAVDPI</td>\n",
       "      <td>8</td>\n",
       "      <td>20</td>\n",
       "      <td>MASESGKLWGGRFVGAVDP</td>\n",
       "      <td>EKFNSSISYDRHLWNVDVQG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[649.3660]PINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[655.3735]PINGPFKPDHYRY</td>\n",
       "      <td>2.394458</td>\n",
       "      <td>2.380664</td>\n",
       "      <td>2.682897</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.435014</td>\n",
       "      <td>2.412394</td>\n",
       "      <td>...</td>\n",
       "      <td>2.538998</td>\n",
       "      <td>green</td>\n",
       "      <td>MSDKLPYKVADIGLAAWGRKALDIAENEMPGLMRMREMYSASKPLK...</td>\n",
       "      <td>412</td>\n",
       "      <td>20</td>\n",
       "      <td>QAQYLG</td>\n",
       "      <td>6</td>\n",
       "      <td>418</td>\n",
       "      <td>HLGKLNVKLTKLTEKQAQYL</td>\n",
       "      <td>PINGPFKPDHYRY</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPM[649.3660]NLPHR</td>\n",
       "      <td>FADVIPM[655.3735]NLPHR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.580647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.446614</td>\n",
       "      <td>2.450718</td>\n",
       "      <td>...</td>\n",
       "      <td>2.389057</td>\n",
       "      <td>green</td>\n",
       "      <td>MDGVSTAILLLLLAVISLSLTFSSRGKGQLPPGPKPLPILGNLLQL...</td>\n",
       "      <td>358</td>\n",
       "      <td>12</td>\n",
       "      <td>FADVIP</td>\n",
       "      <td>6</td>\n",
       "      <td>364</td>\n",
       "      <td>TSMPYTDAVIHEVQRFADVI</td>\n",
       "      <td>NLPHRVTRDTPFRGFLIPKG</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[649.3660]VTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[655.3735]VTLDYTVQVPGTGR</td>\n",
       "      <td>-1.822067</td>\n",
       "      <td>-1.687501</td>\n",
       "      <td>-1.706146</td>\n",
       "      <td>-1.647231</td>\n",
       "      <td>-1.793966</td>\n",
       "      <td>-1.407199</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.857135</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MVDSVYRTRSLGVAAEGLPDQYADGEAARVWQLYIGDTRSRTAEYK...</td>\n",
       "      <td>213</td>\n",
       "      <td>17</td>\n",
       "      <td>AH</td>\n",
       "      <td>2</td>\n",
       "      <td>215</td>\n",
       "      <td>YKSDLTKDITTSVLTVNNKA</td>\n",
       "      <td>VTLDYTVQVPGTGRDGSPGF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDM[649.3660]K</td>\n",
       "      <td>KEQESEVDM[655.3735]K</td>\n",
       "      <td>-1.759694</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.821773</td>\n",
       "      <td>-1.960428</td>\n",
       "      <td>-1.781592</td>\n",
       "      <td>-1.988603</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.871741</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MYRLSSSMLPRALAQAMRTGHLNGQSLHSSAVAATYKYVNKKEQES...</td>\n",
       "      <td>41</td>\n",
       "      <td>10</td>\n",
       "      <td>KEQESEVD</td>\n",
       "      <td>8</td>\n",
       "      <td>49</td>\n",
       "      <td>SSAVAATYKYVNKKEQESEV</td>\n",
       "      <td>KSATDNAARILMWTELIRGL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVM[649.3660]LAVDAVIAELK</td>\n",
       "      <td>RGVM[655.3735]LAVDAVIAELK</td>\n",
       "      <td>-1.812349</td>\n",
       "      <td>-1.980371</td>\n",
       "      <td>-2.307386</td>\n",
       "      <td>-2.492858</td>\n",
       "      <td>-1.405837</td>\n",
       "      <td>-2.094407</td>\n",
       "      <td>...</td>\n",
       "      <td>-1.950460</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MLRLPTVLRQMRPVSRALAPHLTRAYAKDVKFGADARALMLQGVDL...</td>\n",
       "      <td>141</td>\n",
       "      <td>15</td>\n",
       "      <td>RGV</td>\n",
       "      <td>3</td>\n",
       "      <td>144</td>\n",
       "      <td>AKEGFEKISKGANPVEIRRG</td>\n",
       "      <td>LAVDAVIAELKKQSKPVTTP</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>M[649.3660]QLLEIITTDK</td>\n",
       "      <td>M[655.3735]QLLEIITTDK</td>\n",
       "      <td>-2.286954</td>\n",
       "      <td>-2.700322</td>\n",
       "      <td>-1.941018</td>\n",
       "      <td>-1.860306</td>\n",
       "      <td>-2.069488</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.270072</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MVASRAIGSLSRFSAFRILRSRGCICRSFTTSSALLTRTHINYGVK...</td>\n",
       "      <td>505</td>\n",
       "      <td>11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0</td>\n",
       "      <td>505</td>\n",
       "      <td>AAVSKRPEKVIGMHYFSPVD</td>\n",
       "      <td>QLLEIITTDKTSKDTTASAV</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAM[649.3660]R</td>\n",
       "      <td>DHFEEAM[655.3735]R</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.891986</td>\n",
       "      <td>-2.129614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>-2.507726</td>\n",
       "      <td>yellow</td>\n",
       "      <td>MASGADSKGDDLSTAILKQKNRPNRLIVDEAINEDNSVVSLSQPKM...</td>\n",
       "      <td>733</td>\n",
       "      <td>8</td>\n",
       "      <td>DHFEEA</td>\n",
       "      <td>6</td>\n",
       "      <td>739</td>\n",
       "      <td>AMEVEEDDPVPEIRRDHFEE</td>\n",
       "      <td>RFARRSVSDNDIRKYEMFAQ</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>203 rows × 33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Peptide Sequence      Modified Peptide  \\\n",
       "0            AADTIGYPVMIR          AADTIGYPVMIR   \n",
       "1              IAMQTLDMGR            IAMQTLDMGR   \n",
       "2             FVGAVDPIMEK           FVGAVDPIMEK   \n",
       "3    QAQYLGMPINGPFKPDHYRY  QAQYLGMPINGPFKPDHYRY   \n",
       "4            FADVIPMNLPHR          FADVIPMNLPHR   \n",
       "..                    ...                   ...   \n",
       "198     AHMVTLDYTVQVPGTGR     AHMVTLDYTVQVPGTGR   \n",
       "199            KEQESEVDMK            KEQESEVDMK   \n",
       "200       RGVMLAVDAVIAELK       RGVMLAVDAVIAELK   \n",
       "201           MQLLEIITTDK           MQLLEIITTDK   \n",
       "202              DHFEEAMR              DHFEEAMR   \n",
       "\n",
       "             Light Modified Peptide          Heavy Modified Peptide  \\\n",
       "0            AADTIGYPVM[649.3660]IR          AADTIGYPVM[655.3735]IR   \n",
       "1              IAM[649.3660]QTLDMGR            IAM[655.3735]QTLDMGR   \n",
       "2             FVGAVDPIM[649.3660]EK           FVGAVDPIM[655.3735]EK   \n",
       "3    QAQYLGM[649.3660]PINGPFKPDHYRY  QAQYLGM[655.3735]PINGPFKPDHYRY   \n",
       "4            FADVIPM[649.3660]NLPHR          FADVIPM[655.3735]NLPHR   \n",
       "..                              ...                             ...   \n",
       "198     AHM[649.3660]VTLDYTVQVPGTGR     AHM[655.3735]VTLDYTVQVPGTGR   \n",
       "199            KEQESEVDM[649.3660]K            KEQESEVDM[655.3735]K   \n",
       "200       RGVM[649.3660]LAVDAVIAELK       RGVM[655.3735]LAVDAVIAELK   \n",
       "201           M[649.3660]QLLEIITTDK           M[655.3735]QLLEIITTDK   \n",
       "202              DHFEEAM[649.3660]R              DHFEEAM[655.3735]R   \n",
       "\n",
       "     1 Log2 Ratio HL  10 Log2 Ratio HL  11 Log2 Ratio HL  12 Log2 Ratio HL  \\\n",
       "0                NaN               NaN          3.269016               NaN   \n",
       "1                NaN          2.783695          3.114945          2.697822   \n",
       "2                NaN               NaN               NaN               NaN   \n",
       "3           2.394458          2.380664          2.682897               NaN   \n",
       "4                NaN               NaN          2.580647               NaN   \n",
       "..               ...               ...               ...               ...   \n",
       "198        -1.822067         -1.687501         -1.706146         -1.647231   \n",
       "199        -1.759694               NaN         -1.821773         -1.960428   \n",
       "200        -1.812349         -1.980371         -2.307386         -2.492858   \n",
       "201        -2.286954         -2.700322         -1.941018         -1.860306   \n",
       "202              NaN         -2.891986         -2.129614               NaN   \n",
       "\n",
       "     2 Log2 Ratio HL  3 Log2 Ratio HL  ...  Log2HL avg   label  \\\n",
       "0                NaN              NaN  ...    3.289988   green   \n",
       "1                NaN              NaN  ...    2.789195   green   \n",
       "2                NaN         2.383482  ...    2.555706   green   \n",
       "3           2.435014         2.412394  ...    2.538998   green   \n",
       "4           2.446614         2.450718  ...    2.389057   green   \n",
       "..               ...              ...  ...         ...     ...   \n",
       "198        -1.793966        -1.407199  ...   -1.857135  yellow   \n",
       "199        -1.781592        -1.988603  ...   -1.871741  yellow   \n",
       "200        -1.405837        -2.094407  ...   -1.950460  yellow   \n",
       "201        -2.069488              NaN  ...   -2.270072  yellow   \n",
       "202              NaN              NaN  ...   -2.507726  yellow   \n",
       "\n",
       "                                     Complete Sequence  Sequence Location  \\\n",
       "0    MTRILTACKVVKTLKSGFGFANVTTKRQWDFSRPGIRLLSVKAKTA...                575   \n",
       "1    MAAALLARARGPLRRALGVRDWRRLHTVYQSVELPETHQMLRQTCR...                262   \n",
       "2    MASESGKLWGGRFVGAVDPIMEKFNSSISYDRHLWNVDVQGSKAYS...                 12   \n",
       "3    MSDKLPYKVADIGLAAWGRKALDIAENEMPGLMRMREMYSASKPLK...                412   \n",
       "4    MDGVSTAILLLLLAVISLSLTFSSRGKGQLPPGPKPLPILGNLLQL...                358   \n",
       "..                                                 ...                ...   \n",
       "198  MVDSVYRTRSLGVAAEGLPDQYADGEAARVWQLYIGDTRSRTAEYK...                213   \n",
       "199  MYRLSSSMLPRALAQAMRTGHLNGQSLHSSAVAATYKYVNKKEQES...                 41   \n",
       "200  MLRLPTVLRQMRPVSRALAPHLTRAYAKDVKFGADARALMLQGVDL...                141   \n",
       "201  MVASRAIGSLSRFSAFRILRSRGCICRSFTTSSALLTRTHINYGVK...                505   \n",
       "202  MASGADSKGDDLSTAILKQKNRPNRLIVDEAINEDNSVVSLSQPKM...                733   \n",
       "\n",
       "     Sequence Length  Left Prefix Left Prefix Length Methionine Location  \\\n",
       "0                 12    AADTIGYPV                  9                 584   \n",
       "1                 10           IA                  2                 264   \n",
       "2                 11     FVGAVDPI                  8                  20   \n",
       "3                 20       QAQYLG                  6                 418   \n",
       "4                 12       FADVIP                  6                 364   \n",
       "..               ...          ...                ...                 ...   \n",
       "198               17           AH                  2                 215   \n",
       "199               10     KEQESEVD                  8                  49   \n",
       "200               15          RGV                  3                 144   \n",
       "201               11          NaN                  0                 505   \n",
       "202                8       DHFEEA                  6                 739   \n",
       "\n",
       "                  Left 20              Right 20  \n",
       "0    SFAVESMEDALKAADTIGYP  IRSAYALGGLGSGICPNKET  \n",
       "1    EDCRIPKENLLGEPGMGFKI  QTLDMGRIGIASQALGIAQA  \n",
       "2     MASESGKLWGGRFVGAVDP  EKFNSSISYDRHLWNVDVQG  \n",
       "3    HLGKLNVKLTKLTEKQAQYL         PINGPFKPDHYRY  \n",
       "4    TSMPYTDAVIHEVQRFADVI  NLPHRVTRDTPFRGFLIPKG  \n",
       "..                    ...                   ...  \n",
       "198  YKSDLTKDITTSVLTVNNKA  VTLDYTVQVPGTGRDGSPGF  \n",
       "199  SSAVAATYKYVNKKEQESEV  KSATDNAARILMWTELIRGL  \n",
       "200  AKEGFEKISKGANPVEIRRG  LAVDAVIAELKKQSKPVTTP  \n",
       "201  AAVSKRPEKVIGMHYFSPVD  QLLEIITTDKTSKDTTASAV  \n",
       "202  AMEVEEDDPVPEIRRDHFEE  RFARRSVSDNDIRKYEMFAQ  \n",
       "\n",
       "[203 rows x 33 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "path = os.path.join(datasets_path, \"RvsS_peptides_completed_sequence_with_thresholds.csv\")\n",
    "peptides_completed_sequence = pd.read_csv(path)\n",
    "peptides_completed_sequence.set_index(\"Unnamed: 0\", inplace=True)\n",
    "peptides_completed_sequence.index.name = None\n",
    "peptides_completed_sequence"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Download Alphafold Data - Labeled Methionines"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/ritwiksrinivas/Desktop/Projects/methionine-analysis/alphafold_data\n",
      "/Users/ritwiksrinivas/Desktop/Projects/methionine-analysis/alphafold_data/cif\n",
      "/Users/ritwiksrinivas/Desktop/Projects/methionine-analysis/alphafold_data/pae\n"
     ]
    }
   ],
   "source": [
    "# path for alphafold protein data\n",
    "alphafold_path_str = \"../alphafold_data/\"\n",
    "alphafold_path = os.path.abspath(alphafold_path_str)\n",
    "\n",
    "cif_dir = os.path.join(alphafold_path, \"cif\")\n",
    "pae_dir = os.path.join(alphafold_path, \"pae\")\n",
    "\n",
    "print(alphafold_path)\n",
    "print(cif_dir)\n",
    "print(pae_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Q8C196', 'Q07417', 'Q91YI0', 'P50247', 'P33267', 'P97872',\n",
       "       'Q9DBT9', 'P32020', 'P32020', 'P50136', 'P26443', 'Q91VS7',\n",
       "       'Q920A5', 'O55125', 'Q8C196', 'Q05920', 'Q91VR2', 'Q91VR2',\n",
       "       'P51881', 'P56480', 'P97872', 'Q61425', 'P47962', 'P53395',\n",
       "       'Q921G7', 'Q9CQR4', 'Q921G7', 'P63017', 'Q62425', 'P20029',\n",
       "       'O35423', 'Q61102', 'Q8C196', 'Q9WUR2', 'Q9CQQ7', 'P55096',\n",
       "       'P01942', 'O35490', 'P38647', 'Q60991', 'P56480', 'Q9Z2I8',\n",
       "       'Q9EQ20', 'Q9Z1J3', 'Q5U458', 'O35129', 'Q05920', 'Q99JY0',\n",
       "       'P55096', 'Q91VS7', 'Q99JY0', 'Q9WTP6', 'Q99LC5', 'P16460',\n",
       "       'Q9CZW5', 'P08226', 'P51881', 'P08226', 'Q03265', 'P55096',\n",
       "       'P16460', 'Q99LC5', 'P63038', 'Q8CAQ8', 'O35386', 'Q9Z2I0',\n",
       "       'P54869', 'P56480', 'Q9Z2I8', 'Q8C196', 'P97742', 'P55096',\n",
       "       'P63038', 'Q9DCN1', 'P56480', 'Q05920', 'Q9WUM5', 'P63038',\n",
       "       'Q05920', 'Q91VS7', 'P56480', 'P60710', 'Q9Z2I8', 'Q8C196',\n",
       "       'P68033', 'Q9CQ69', 'Q9DBJ1', 'Q9CR62', 'P19783', 'P63038',\n",
       "       'Q8QZY2', 'P08226', 'O70579', 'Q9CQN1', 'Q8BH95', 'Q03265',\n",
       "       'P32020', 'P51660', 'Q8C196', 'Q8BGY7', 'P55096', 'P56480',\n",
       "       'Q9EQ20', 'P55096', 'Q91VS7', 'Q8CC88', 'P55096', 'Q9CRB9',\n",
       "       'Q9Z0X1', 'Q9QXD1', 'Q2TPA8', 'P24270', 'O35490', 'Q9CPQ8',\n",
       "       'Q80XN0', 'Q925I1', 'Q03265', 'Q8CIM7', 'P24270', 'Q9Z1P6',\n",
       "       'Q8VC30', 'Q9R0H0', 'Q8C196', 'P51658', 'P32020', 'P63038',\n",
       "       'Q9CQQ7', 'P16332', 'Q80XN0', 'Q80XN0', 'Q8VDN2', 'Q9DBJ1',\n",
       "       'Q9CQQ7', 'Q91VR2', 'P33267', 'P38647', 'Q8BJ64', 'P42125',\n",
       "       'Q99MR8', 'P54869', 'Q80XN0', 'P38647', 'P29758', 'Q80XN0',\n",
       "       'P32020', 'Q9R0H0', 'Q5FW57', 'P32020', 'P08226', 'Q99JY0',\n",
       "       'P55096', 'P63038', 'P38647', 'Q07417', 'P62821', 'Q9CQQ7',\n",
       "       'P26443', 'P52825', 'P50136', 'Q8BGY7', 'Q9JKR6', 'Q99KR3',\n",
       "       'P63038', 'P50136', 'Q80XN0', 'P26443', 'Q61733', 'Q2TPA8',\n",
       "       'P26443', 'Q61335', 'Q7TNG8', 'Q9DCM2', 'Q99JY0', 'Q8CFA2',\n",
       "       'Q8C196', 'Q64433', 'P45952', 'Q9JKR6', 'Q9DCM2', 'P08249',\n",
       "       'P50136', 'O35459', 'P97450', 'P55096', 'Q9CQQ7', 'Q99MR8',\n",
       "       'Q9CYR0', 'Q64310', 'Q8K1Z0', 'Q9CPU0', 'Q2TPA8', 'Q8CAQ8',\n",
       "       'Q80XL6', 'P97807', 'P11725', 'Q91VR2', 'Q8R164', 'Q8VC30',\n",
       "       'Q9QXF8', 'Q8K3J1', 'P63038', 'Q8BMS1', 'Q01853'], dtype=object)"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# set uniprot IDs to use\n",
    "uniprotIDs = peptides_completed_sequence[\"Protein ID\"].values\n",
    "uniprotIDs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 203/203 [00:00<00:00, 77102.57it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-04-23 18:27:50> Valid proteins: 0\n",
      "2024-04-23 18:27:50> Invalid proteins: 0\n",
      "2024-04-23 18:27:50> Existing proteins: 203\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "# download cif data for proteins\n",
    "# SLOW THE FIRST TIME\n",
    "valid_proteins_cif, invalid_proteins_cif, existing_proteins_cif = download_alphafold_cif(\n",
    "    proteins=uniprotIDs,\n",
    "    out_folder=cif_dir\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 203/203 [00:00<00:00, 38310.18it/s]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-04-23 18:27:50> Valid proteins: 0\n",
      "2024-04-23 18:27:50> Invalid proteins: 0\n",
      "2024-04-23 18:27:50> Existing proteins: 203\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "# download pae data for proteins\n",
    "# SLOW THE FIRST TIME\n",
    "valid_proteins_pae, invalid_proteins_pae, existing_proteins_pae = download_alphafold_pae(\n",
    "    proteins=uniprotIDs,\n",
    "    out_folder=pae_dir, \n",
    ")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Construct Alphafold Dataframe (Calculate Accessibilities) - Labeled Methionines"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  0%|          | 0/210 [00:00<?, ?it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "  3%|▎         | 6/210 [00:00<00:03, 56.69it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "  6%|▌         | 12/210 [00:00<00:06, 28.96it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 10%|█         | 22/210 [00:00<00:04, 46.87it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 14%|█▍        | 30/210 [00:00<00:03, 47.17it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 17%|█▋        | 36/210 [00:00<00:05, 34.37it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 20%|██        | 42/210 [00:01<00:04, 35.67it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 22%|██▏       | 47/210 [00:01<00:05, 28.72it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 24%|██▍       | 51/210 [00:01<00:05, 27.78it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 27%|██▋       | 56/210 [00:01<00:05, 29.86it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 29%|██▊       | 60/210 [00:01<00:05, 28.09it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 30%|███       | 63/210 [00:02<00:06, 23.07it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 31%|███▏      | 66/210 [00:02<00:06, 21.43it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 34%|███▍      | 71/210 [00:02<00:05, 25.00it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 35%|███▌      | 74/210 [00:02<00:05, 24.49it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 37%|███▋      | 77/210 [00:02<00:05, 23.50it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 38%|███▊      | 80/210 [00:02<00:05, 22.99it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 40%|███▉      | 83/210 [00:03<00:07, 17.47it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 40%|████      | 85/210 [00:03<00:08, 13.90it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 42%|████▏     | 88/210 [00:03<00:07, 15.67it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 43%|████▎     | 91/210 [00:03<00:06, 17.38it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 45%|████▌     | 95/210 [00:03<00:06, 18.65it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 47%|████▋     | 99/210 [00:03<00:05, 21.08it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 50%|█████     | 106/210 [00:04<00:03, 28.32it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 52%|█████▏    | 109/210 [00:04<00:04, 24.91it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 56%|█████▌    | 117/210 [00:04<00:02, 31.39it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 58%|█████▊    | 121/210 [00:04<00:02, 32.50it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 60%|█████▉    | 125/210 [00:05<00:05, 14.75it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 62%|██████▏   | 130/210 [00:05<00:04, 18.43it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 64%|██████▍   | 135/210 [00:05<00:03, 22.68it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 66%|██████▌   | 139/210 [00:05<00:02, 24.37it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 68%|██████▊   | 143/210 [00:05<00:02, 25.01it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 71%|███████▏  | 150/210 [00:05<00:01, 31.45it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 73%|███████▎  | 154/210 [00:06<00:02, 22.88it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 76%|███████▌  | 159/210 [00:06<00:02, 23.43it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 79%|███████▊  | 165/210 [00:06<00:01, 26.52it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 81%|████████  | 170/210 [00:06<00:01, 30.24it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 83%|████████▎ | 174/210 [00:06<00:01, 29.54it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 89%|████████▊ | 186/210 [00:06<00:00, 42.12it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 91%|█████████ | 191/210 [00:07<00:00, 36.96it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 93%|█████████▎| 195/210 [00:07<00:00, 29.18it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 96%|█████████▌| 201/210 [00:07<00:00, 32.17it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 98%|█████████▊| 205/210 [00:07<00:00, 30.88it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "100%|█████████▉| 209/210 [00:07<00:00, 23.89it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "100%|██████████| 210/210 [00:08<00:00, 26.06it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>protein_number</th>\n",
       "      <th>AA</th>\n",
       "      <th>position</th>\n",
       "      <th>quality</th>\n",
       "      <th>x_coord_c</th>\n",
       "      <th>x_coord_ca</th>\n",
       "      <th>x_coord_cb</th>\n",
       "      <th>x_coord_n</th>\n",
       "      <th>y_coord_c</th>\n",
       "      <th>...</th>\n",
       "      <th>z_coord_ca</th>\n",
       "      <th>z_coord_cb</th>\n",
       "      <th>z_coord_n</th>\n",
       "      <th>secondary_structure</th>\n",
       "      <th>structure_group</th>\n",
       "      <th>BEND</th>\n",
       "      <th>HELX</th>\n",
       "      <th>STRN</th>\n",
       "      <th>TURN</th>\n",
       "      <th>unstructured</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>1</td>\n",
       "      <td>45.38</td>\n",
       "      <td>-35.172</td>\n",
       "      <td>-33.707</td>\n",
       "      <td>-32.901</td>\n",
       "      <td>-33.031</td>\n",
       "      <td>3.159</td>\n",
       "      <td>...</td>\n",
       "      <td>51.060</td>\n",
       "      <td>49.845</td>\n",
       "      <td>51.772</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>50.78</td>\n",
       "      <td>-38.162</td>\n",
       "      <td>-37.117</td>\n",
       "      <td>-37.283</td>\n",
       "      <td>-35.709</td>\n",
       "      <td>2.229</td>\n",
       "      <td>...</td>\n",
       "      <td>50.385</td>\n",
       "      <td>50.224</td>\n",
       "      <td>50.740</td>\n",
       "      <td>TURN_TY1_P</td>\n",
       "      <td>TURN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>Q</td>\n",
       "      <td>3</td>\n",
       "      <td>52.40</td>\n",
       "      <td>-38.996</td>\n",
       "      <td>-38.671</td>\n",
       "      <td>-38.089</td>\n",
       "      <td>-37.769</td>\n",
       "      <td>4.596</td>\n",
       "      <td>...</td>\n",
       "      <td>53.616</td>\n",
       "      <td>55.022</td>\n",
       "      <td>52.617</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>51.23</td>\n",
       "      <td>-39.258</td>\n",
       "      <td>-38.274</td>\n",
       "      <td>-36.904</td>\n",
       "      <td>-38.080</td>\n",
       "      <td>7.104</td>\n",
       "      <td>...</td>\n",
       "      <td>52.434</td>\n",
       "      <td>52.182</td>\n",
       "      <td>52.692</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>L</td>\n",
       "      <td>5</td>\n",
       "      <td>50.02</td>\n",
       "      <td>-42.031</td>\n",
       "      <td>-40.574</td>\n",
       "      <td>-40.297</td>\n",
       "      <td>-39.604</td>\n",
       "      <td>6.273</td>\n",
       "      <td>...</td>\n",
       "      <td>49.354</td>\n",
       "      <td>48.180</td>\n",
       "      <td>50.434</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52926</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>S</td>\n",
       "      <td>429</td>\n",
       "      <td>94.38</td>\n",
       "      <td>26.176</td>\n",
       "      <td>27.466</td>\n",
       "      <td>27.350</td>\n",
       "      <td>27.774</td>\n",
       "      <td>14.096</td>\n",
       "      <td>...</td>\n",
       "      <td>-18.565</td>\n",
       "      <td>-18.413</td>\n",
       "      <td>-17.274</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52927</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>V</td>\n",
       "      <td>430</td>\n",
       "      <td>91.54</td>\n",
       "      <td>24.398</td>\n",
       "      <td>24.087</td>\n",
       "      <td>23.021</td>\n",
       "      <td>25.313</td>\n",
       "      <td>16.707</td>\n",
       "      <td>...</td>\n",
       "      <td>-18.909</td>\n",
       "      <td>-17.814</td>\n",
       "      <td>-18.401</td>\n",
       "      <td>BEND</td>\n",
       "      <td>BEND</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52928</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>A</td>\n",
       "      <td>431</td>\n",
       "      <td>73.79</td>\n",
       "      <td>23.429</td>\n",
       "      <td>23.961</td>\n",
       "      <td>23.247</td>\n",
       "      <td>23.808</td>\n",
       "      <td>19.412</td>\n",
       "      <td>...</td>\n",
       "      <td>-21.302</td>\n",
       "      <td>-22.660</td>\n",
       "      <td>-20.673</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52929</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>K</td>\n",
       "      <td>432</td>\n",
       "      <td>62.38</td>\n",
       "      <td>22.744</td>\n",
       "      <td>23.748</td>\n",
       "      <td>24.902</td>\n",
       "      <td>24.238</td>\n",
       "      <td>22.331</td>\n",
       "      <td>...</td>\n",
       "      <td>-19.459</td>\n",
       "      <td>-19.112</td>\n",
       "      <td>-20.119</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52930</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>K</td>\n",
       "      <td>433</td>\n",
       "      <td>51.30</td>\n",
       "      <td>20.532</td>\n",
       "      <td>20.426</td>\n",
       "      <td>19.015</td>\n",
       "      <td>21.457</td>\n",
       "      <td>24.555</td>\n",
       "      <td>...</td>\n",
       "      <td>-20.722</td>\n",
       "      <td>-20.474</td>\n",
       "      <td>-20.058</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52931 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      protein_id  protein_number AA  position  quality  x_coord_c  x_coord_ca  \\\n",
       "0         O35129               1  M         1    45.38    -35.172     -33.707   \n",
       "1         O35129               1  A         2    50.78    -38.162     -37.117   \n",
       "2         O35129               1  Q         3    52.40    -38.996     -38.671   \n",
       "3         O35129               1  N         4    51.23    -39.258     -38.274   \n",
       "4         O35129               1  L         5    50.02    -42.031     -40.574   \n",
       "...          ...             ... ..       ...      ...        ...         ...   \n",
       "52926     Q9Z2I8             113  S       429    94.38     26.176      27.466   \n",
       "52927     Q9Z2I8             113  V       430    91.54     24.398      24.087   \n",
       "52928     Q9Z2I8             113  A       431    73.79     23.429      23.961   \n",
       "52929     Q9Z2I8             113  K       432    62.38     22.744      23.748   \n",
       "52930     Q9Z2I8             113  K       433    51.30     20.532      20.426   \n",
       "\n",
       "       x_coord_cb  x_coord_n  y_coord_c  ...  z_coord_ca  z_coord_cb  \\\n",
       "0         -32.901    -33.031      3.159  ...      51.060      49.845   \n",
       "1         -37.283    -35.709      2.229  ...      50.385      50.224   \n",
       "2         -38.089    -37.769      4.596  ...      53.616      55.022   \n",
       "3         -36.904    -38.080      7.104  ...      52.434      52.182   \n",
       "4         -40.297    -39.604      6.273  ...      49.354      48.180   \n",
       "...           ...        ...        ...  ...         ...         ...   \n",
       "52926      27.350     27.774     14.096  ...     -18.565     -18.413   \n",
       "52927      23.021     25.313     16.707  ...     -18.909     -17.814   \n",
       "52928      23.247     23.808     19.412  ...     -21.302     -22.660   \n",
       "52929      24.902     24.238     22.331  ...     -19.459     -19.112   \n",
       "52930      19.015     21.457     24.555  ...     -20.722     -20.474   \n",
       "\n",
       "       z_coord_n  secondary_structure  structure_group  BEND  HELX STRN TURN  \\\n",
       "0         51.772         unstructured     unstructured     0     0    0    0   \n",
       "1         50.740           TURN_TY1_P             TURN     0     0    0    1   \n",
       "2         52.617         HELX_RH_AL_P             HELX     0     1    0    0   \n",
       "3         52.692         HELX_RH_AL_P             HELX     0     1    0    0   \n",
       "4         50.434         HELX_RH_AL_P             HELX     0     1    0    0   \n",
       "...          ...                  ...              ...   ...   ...  ...  ...   \n",
       "52926    -17.274         HELX_RH_AL_P             HELX     0     1    0    0   \n",
       "52927    -18.401                 BEND             BEND     1     0    0    0   \n",
       "52928    -20.673         unstructured     unstructured     0     0    0    0   \n",
       "52929    -20.119         unstructured     unstructured     0     0    0    0   \n",
       "52930    -20.058         unstructured     unstructured     0     0    0    0   \n",
       "\n",
       "       unstructured  \n",
       "0                 1  \n",
       "1                 0  \n",
       "2                 0  \n",
       "3                 0  \n",
       "4                 0  \n",
       "...             ...  \n",
       "52926             0  \n",
       "52927             0  \n",
       "52928             1  \n",
       "52929             1  \n",
       "52930             1  \n",
       "\n",
       "[52931 rows x 24 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# format alphafold data into dataframe\n",
    "alphafold_annotation = format_alphafold_data(\n",
    "    directory=cif_dir, \n",
    "    protein_ids=uniprotIDs)\n",
    "alphafold_annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 113/113 [00:03<00:00, 37.05it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>AA</th>\n",
       "      <th>position</th>\n",
       "      <th>nAA_24_180_pae</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O35129</td>\n",
       "      <td>M</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O35129</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O35129</td>\n",
       "      <td>Q</td>\n",
       "      <td>3</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O35129</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O35129</td>\n",
       "      <td>L</td>\n",
       "      <td>5</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52926</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>S</td>\n",
       "      <td>429</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52927</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>V</td>\n",
       "      <td>430</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52928</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>A</td>\n",
       "      <td>431</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52929</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>K</td>\n",
       "      <td>432</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52930</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>K</td>\n",
       "      <td>433</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52931 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      protein_id AA  position  nAA_24_180_pae\n",
       "0         O35129  M         1               7\n",
       "1         O35129  A         2               9\n",
       "2         O35129  Q         3              10\n",
       "3         O35129  N         4              11\n",
       "4         O35129  L         5              11\n",
       "...          ... ..       ...             ...\n",
       "52926     Q9Z2I8  S       429              91\n",
       "52927     Q9Z2I8  V       430              78\n",
       "52928     Q9Z2I8  A       431              36\n",
       "52929     Q9Z2I8  K       432              19\n",
       "52930     Q9Z2I8  K       433              11\n",
       "\n",
       "[52931 rows x 4 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# calculate full sphere exposure -> radius = 24\n",
    "full_sphere_exposure_24 = annotate_accessibility(\n",
    "    df=alphafold_annotation, \n",
    "    max_dist=24, \n",
    "    max_angle=180, \n",
    "    error_dir=pae_dir)\n",
    "full_sphere_exposure_24"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>protein_number</th>\n",
       "      <th>AA</th>\n",
       "      <th>position</th>\n",
       "      <th>quality</th>\n",
       "      <th>x_coord_c</th>\n",
       "      <th>x_coord_ca</th>\n",
       "      <th>x_coord_cb</th>\n",
       "      <th>x_coord_n</th>\n",
       "      <th>y_coord_c</th>\n",
       "      <th>...</th>\n",
       "      <th>z_coord_cb</th>\n",
       "      <th>z_coord_n</th>\n",
       "      <th>secondary_structure</th>\n",
       "      <th>structure_group</th>\n",
       "      <th>BEND</th>\n",
       "      <th>HELX</th>\n",
       "      <th>STRN</th>\n",
       "      <th>TURN</th>\n",
       "      <th>unstructured</th>\n",
       "      <th>nAA_24_180_pae</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>1</td>\n",
       "      <td>45.38</td>\n",
       "      <td>-35.172</td>\n",
       "      <td>-33.707</td>\n",
       "      <td>-32.901</td>\n",
       "      <td>-33.031</td>\n",
       "      <td>3.159</td>\n",
       "      <td>...</td>\n",
       "      <td>49.845</td>\n",
       "      <td>51.772</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>50.78</td>\n",
       "      <td>-38.162</td>\n",
       "      <td>-37.117</td>\n",
       "      <td>-37.283</td>\n",
       "      <td>-35.709</td>\n",
       "      <td>2.229</td>\n",
       "      <td>...</td>\n",
       "      <td>50.224</td>\n",
       "      <td>50.740</td>\n",
       "      <td>TURN_TY1_P</td>\n",
       "      <td>TURN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>Q</td>\n",
       "      <td>3</td>\n",
       "      <td>52.40</td>\n",
       "      <td>-38.996</td>\n",
       "      <td>-38.671</td>\n",
       "      <td>-38.089</td>\n",
       "      <td>-37.769</td>\n",
       "      <td>4.596</td>\n",
       "      <td>...</td>\n",
       "      <td>55.022</td>\n",
       "      <td>52.617</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>51.23</td>\n",
       "      <td>-39.258</td>\n",
       "      <td>-38.274</td>\n",
       "      <td>-36.904</td>\n",
       "      <td>-38.080</td>\n",
       "      <td>7.104</td>\n",
       "      <td>...</td>\n",
       "      <td>52.182</td>\n",
       "      <td>52.692</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>L</td>\n",
       "      <td>5</td>\n",
       "      <td>50.02</td>\n",
       "      <td>-42.031</td>\n",
       "      <td>-40.574</td>\n",
       "      <td>-40.297</td>\n",
       "      <td>-39.604</td>\n",
       "      <td>6.273</td>\n",
       "      <td>...</td>\n",
       "      <td>48.180</td>\n",
       "      <td>50.434</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52926</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>S</td>\n",
       "      <td>429</td>\n",
       "      <td>94.38</td>\n",
       "      <td>26.176</td>\n",
       "      <td>27.466</td>\n",
       "      <td>27.350</td>\n",
       "      <td>27.774</td>\n",
       "      <td>14.096</td>\n",
       "      <td>...</td>\n",
       "      <td>-18.413</td>\n",
       "      <td>-17.274</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52927</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>V</td>\n",
       "      <td>430</td>\n",
       "      <td>91.54</td>\n",
       "      <td>24.398</td>\n",
       "      <td>24.087</td>\n",
       "      <td>23.021</td>\n",
       "      <td>25.313</td>\n",
       "      <td>16.707</td>\n",
       "      <td>...</td>\n",
       "      <td>-17.814</td>\n",
       "      <td>-18.401</td>\n",
       "      <td>BEND</td>\n",
       "      <td>BEND</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>78</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52928</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>A</td>\n",
       "      <td>431</td>\n",
       "      <td>73.79</td>\n",
       "      <td>23.429</td>\n",
       "      <td>23.961</td>\n",
       "      <td>23.247</td>\n",
       "      <td>23.808</td>\n",
       "      <td>19.412</td>\n",
       "      <td>...</td>\n",
       "      <td>-22.660</td>\n",
       "      <td>-20.673</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>36</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52929</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>K</td>\n",
       "      <td>432</td>\n",
       "      <td>62.38</td>\n",
       "      <td>22.744</td>\n",
       "      <td>23.748</td>\n",
       "      <td>24.902</td>\n",
       "      <td>24.238</td>\n",
       "      <td>22.331</td>\n",
       "      <td>...</td>\n",
       "      <td>-19.112</td>\n",
       "      <td>-20.119</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>19</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52930</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>K</td>\n",
       "      <td>433</td>\n",
       "      <td>51.30</td>\n",
       "      <td>20.532</td>\n",
       "      <td>20.426</td>\n",
       "      <td>19.015</td>\n",
       "      <td>21.457</td>\n",
       "      <td>24.555</td>\n",
       "      <td>...</td>\n",
       "      <td>-20.474</td>\n",
       "      <td>-20.058</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52931 rows × 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      protein_id  protein_number AA  position  quality  x_coord_c  x_coord_ca  \\\n",
       "0         O35129               1  M         1    45.38    -35.172     -33.707   \n",
       "1         O35129               1  A         2    50.78    -38.162     -37.117   \n",
       "2         O35129               1  Q         3    52.40    -38.996     -38.671   \n",
       "3         O35129               1  N         4    51.23    -39.258     -38.274   \n",
       "4         O35129               1  L         5    50.02    -42.031     -40.574   \n",
       "...          ...             ... ..       ...      ...        ...         ...   \n",
       "52926     Q9Z2I8             113  S       429    94.38     26.176      27.466   \n",
       "52927     Q9Z2I8             113  V       430    91.54     24.398      24.087   \n",
       "52928     Q9Z2I8             113  A       431    73.79     23.429      23.961   \n",
       "52929     Q9Z2I8             113  K       432    62.38     22.744      23.748   \n",
       "52930     Q9Z2I8             113  K       433    51.30     20.532      20.426   \n",
       "\n",
       "       x_coord_cb  x_coord_n  y_coord_c  ...  z_coord_cb  z_coord_n  \\\n",
       "0         -32.901    -33.031      3.159  ...      49.845     51.772   \n",
       "1         -37.283    -35.709      2.229  ...      50.224     50.740   \n",
       "2         -38.089    -37.769      4.596  ...      55.022     52.617   \n",
       "3         -36.904    -38.080      7.104  ...      52.182     52.692   \n",
       "4         -40.297    -39.604      6.273  ...      48.180     50.434   \n",
       "...           ...        ...        ...  ...         ...        ...   \n",
       "52926      27.350     27.774     14.096  ...     -18.413    -17.274   \n",
       "52927      23.021     25.313     16.707  ...     -17.814    -18.401   \n",
       "52928      23.247     23.808     19.412  ...     -22.660    -20.673   \n",
       "52929      24.902     24.238     22.331  ...     -19.112    -20.119   \n",
       "52930      19.015     21.457     24.555  ...     -20.474    -20.058   \n",
       "\n",
       "       secondary_structure  structure_group  BEND  HELX  STRN TURN  \\\n",
       "0             unstructured     unstructured     0     0     0    0   \n",
       "1               TURN_TY1_P             TURN     0     0     0    1   \n",
       "2             HELX_RH_AL_P             HELX     0     1     0    0   \n",
       "3             HELX_RH_AL_P             HELX     0     1     0    0   \n",
       "4             HELX_RH_AL_P             HELX     0     1     0    0   \n",
       "...                    ...              ...   ...   ...   ...  ...   \n",
       "52926         HELX_RH_AL_P             HELX     0     1     0    0   \n",
       "52927                 BEND             BEND     1     0     0    0   \n",
       "52928         unstructured     unstructured     0     0     0    0   \n",
       "52929         unstructured     unstructured     0     0     0    0   \n",
       "52930         unstructured     unstructured     0     0     0    0   \n",
       "\n",
       "      unstructured  nAA_24_180_pae  \n",
       "0                1               7  \n",
       "1                0               9  \n",
       "2                0              10  \n",
       "3                0              11  \n",
       "4                0              11  \n",
       "...            ...             ...  \n",
       "52926            0              91  \n",
       "52927            0              78  \n",
       "52928            1              36  \n",
       "52929            1              19  \n",
       "52930            1              11  \n",
       "\n",
       "[52931 rows x 25 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alphafold_accessibility = alphafold_annotation.merge(\n",
    "    full_sphere_exposure_24, how='left', on=['protein_id','AA','position'])\n",
    "alphafold_accessibility"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 113/113 [00:00<00:00, 130.54it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>AA</th>\n",
       "      <th>position</th>\n",
       "      <th>nAA_12_70_pae</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O35129</td>\n",
       "      <td>M</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O35129</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O35129</td>\n",
       "      <td>Q</td>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O35129</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O35129</td>\n",
       "      <td>L</td>\n",
       "      <td>5</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52926</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>S</td>\n",
       "      <td>429</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52927</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>V</td>\n",
       "      <td>430</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52928</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>A</td>\n",
       "      <td>431</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52929</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>K</td>\n",
       "      <td>432</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52930</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>K</td>\n",
       "      <td>433</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52931 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      protein_id AA  position  nAA_12_70_pae\n",
       "0         O35129  M         1              0\n",
       "1         O35129  A         2              0\n",
       "2         O35129  Q         3              0\n",
       "3         O35129  N         4              1\n",
       "4         O35129  L         5              1\n",
       "...          ... ..       ...            ...\n",
       "52926     Q9Z2I8  S       429             12\n",
       "52927     Q9Z2I8  V       430              4\n",
       "52928     Q9Z2I8  A       431              0\n",
       "52929     Q9Z2I8  K       432              0\n",
       "52930     Q9Z2I8  K       433              0\n",
       "\n",
       "[52931 rows x 4 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# calculate full sphere exposure -> angle = 70, radius = 12\n",
    "part_sphere_exposure = annotate_accessibility(\n",
    "    df=alphafold_annotation, \n",
    "    max_dist=12, \n",
    "    max_angle=70, \n",
    "    error_dir=pae_dir)\n",
    "part_sphere_exposure"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>protein_number</th>\n",
       "      <th>AA</th>\n",
       "      <th>position</th>\n",
       "      <th>quality</th>\n",
       "      <th>x_coord_c</th>\n",
       "      <th>x_coord_ca</th>\n",
       "      <th>x_coord_cb</th>\n",
       "      <th>x_coord_n</th>\n",
       "      <th>y_coord_c</th>\n",
       "      <th>...</th>\n",
       "      <th>z_coord_n</th>\n",
       "      <th>secondary_structure</th>\n",
       "      <th>structure_group</th>\n",
       "      <th>BEND</th>\n",
       "      <th>HELX</th>\n",
       "      <th>STRN</th>\n",
       "      <th>TURN</th>\n",
       "      <th>unstructured</th>\n",
       "      <th>nAA_24_180_pae</th>\n",
       "      <th>nAA_12_70_pae</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>1</td>\n",
       "      <td>45.38</td>\n",
       "      <td>-35.172</td>\n",
       "      <td>-33.707</td>\n",
       "      <td>-32.901</td>\n",
       "      <td>-33.031</td>\n",
       "      <td>3.159</td>\n",
       "      <td>...</td>\n",
       "      <td>51.772</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>50.78</td>\n",
       "      <td>-38.162</td>\n",
       "      <td>-37.117</td>\n",
       "      <td>-37.283</td>\n",
       "      <td>-35.709</td>\n",
       "      <td>2.229</td>\n",
       "      <td>...</td>\n",
       "      <td>50.740</td>\n",
       "      <td>TURN_TY1_P</td>\n",
       "      <td>TURN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>Q</td>\n",
       "      <td>3</td>\n",
       "      <td>52.40</td>\n",
       "      <td>-38.996</td>\n",
       "      <td>-38.671</td>\n",
       "      <td>-38.089</td>\n",
       "      <td>-37.769</td>\n",
       "      <td>4.596</td>\n",
       "      <td>...</td>\n",
       "      <td>52.617</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>51.23</td>\n",
       "      <td>-39.258</td>\n",
       "      <td>-38.274</td>\n",
       "      <td>-36.904</td>\n",
       "      <td>-38.080</td>\n",
       "      <td>7.104</td>\n",
       "      <td>...</td>\n",
       "      <td>52.692</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>L</td>\n",
       "      <td>5</td>\n",
       "      <td>50.02</td>\n",
       "      <td>-42.031</td>\n",
       "      <td>-40.574</td>\n",
       "      <td>-40.297</td>\n",
       "      <td>-39.604</td>\n",
       "      <td>6.273</td>\n",
       "      <td>...</td>\n",
       "      <td>50.434</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52926</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>S</td>\n",
       "      <td>429</td>\n",
       "      <td>94.38</td>\n",
       "      <td>26.176</td>\n",
       "      <td>27.466</td>\n",
       "      <td>27.350</td>\n",
       "      <td>27.774</td>\n",
       "      <td>14.096</td>\n",
       "      <td>...</td>\n",
       "      <td>-17.274</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>12</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52927</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>V</td>\n",
       "      <td>430</td>\n",
       "      <td>91.54</td>\n",
       "      <td>24.398</td>\n",
       "      <td>24.087</td>\n",
       "      <td>23.021</td>\n",
       "      <td>25.313</td>\n",
       "      <td>16.707</td>\n",
       "      <td>...</td>\n",
       "      <td>-18.401</td>\n",
       "      <td>BEND</td>\n",
       "      <td>BEND</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>78</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52928</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>A</td>\n",
       "      <td>431</td>\n",
       "      <td>73.79</td>\n",
       "      <td>23.429</td>\n",
       "      <td>23.961</td>\n",
       "      <td>23.247</td>\n",
       "      <td>23.808</td>\n",
       "      <td>19.412</td>\n",
       "      <td>...</td>\n",
       "      <td>-20.673</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52929</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>K</td>\n",
       "      <td>432</td>\n",
       "      <td>62.38</td>\n",
       "      <td>22.744</td>\n",
       "      <td>23.748</td>\n",
       "      <td>24.902</td>\n",
       "      <td>24.238</td>\n",
       "      <td>22.331</td>\n",
       "      <td>...</td>\n",
       "      <td>-20.119</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52930</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>K</td>\n",
       "      <td>433</td>\n",
       "      <td>51.30</td>\n",
       "      <td>20.532</td>\n",
       "      <td>20.426</td>\n",
       "      <td>19.015</td>\n",
       "      <td>21.457</td>\n",
       "      <td>24.555</td>\n",
       "      <td>...</td>\n",
       "      <td>-20.058</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52931 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      protein_id  protein_number AA  position  quality  x_coord_c  x_coord_ca  \\\n",
       "0         O35129               1  M         1    45.38    -35.172     -33.707   \n",
       "1         O35129               1  A         2    50.78    -38.162     -37.117   \n",
       "2         O35129               1  Q         3    52.40    -38.996     -38.671   \n",
       "3         O35129               1  N         4    51.23    -39.258     -38.274   \n",
       "4         O35129               1  L         5    50.02    -42.031     -40.574   \n",
       "...          ...             ... ..       ...      ...        ...         ...   \n",
       "52926     Q9Z2I8             113  S       429    94.38     26.176      27.466   \n",
       "52927     Q9Z2I8             113  V       430    91.54     24.398      24.087   \n",
       "52928     Q9Z2I8             113  A       431    73.79     23.429      23.961   \n",
       "52929     Q9Z2I8             113  K       432    62.38     22.744      23.748   \n",
       "52930     Q9Z2I8             113  K       433    51.30     20.532      20.426   \n",
       "\n",
       "       x_coord_cb  x_coord_n  y_coord_c  ...  z_coord_n  secondary_structure  \\\n",
       "0         -32.901    -33.031      3.159  ...     51.772         unstructured   \n",
       "1         -37.283    -35.709      2.229  ...     50.740           TURN_TY1_P   \n",
       "2         -38.089    -37.769      4.596  ...     52.617         HELX_RH_AL_P   \n",
       "3         -36.904    -38.080      7.104  ...     52.692         HELX_RH_AL_P   \n",
       "4         -40.297    -39.604      6.273  ...     50.434         HELX_RH_AL_P   \n",
       "...           ...        ...        ...  ...        ...                  ...   \n",
       "52926      27.350     27.774     14.096  ...    -17.274         HELX_RH_AL_P   \n",
       "52927      23.021     25.313     16.707  ...    -18.401                 BEND   \n",
       "52928      23.247     23.808     19.412  ...    -20.673         unstructured   \n",
       "52929      24.902     24.238     22.331  ...    -20.119         unstructured   \n",
       "52930      19.015     21.457     24.555  ...    -20.058         unstructured   \n",
       "\n",
       "       structure_group  BEND  HELX  STRN  TURN unstructured nAA_24_180_pae  \\\n",
       "0         unstructured     0     0     0     0            1              7   \n",
       "1                 TURN     0     0     0     1            0              9   \n",
       "2                 HELX     0     1     0     0            0             10   \n",
       "3                 HELX     0     1     0     0            0             11   \n",
       "4                 HELX     0     1     0     0            0             11   \n",
       "...                ...   ...   ...   ...   ...          ...            ...   \n",
       "52926             HELX     0     1     0     0            0             91   \n",
       "52927             BEND     1     0     0     0            0             78   \n",
       "52928     unstructured     0     0     0     0            1             36   \n",
       "52929     unstructured     0     0     0     0            1             19   \n",
       "52930     unstructured     0     0     0     0            1             11   \n",
       "\n",
       "       nAA_12_70_pae  \n",
       "0                  0  \n",
       "1                  0  \n",
       "2                  0  \n",
       "3                  1  \n",
       "4                  1  \n",
       "...              ...  \n",
       "52926             12  \n",
       "52927              4  \n",
       "52928              0  \n",
       "52929              0  \n",
       "52930              0  \n",
       "\n",
       "[52931 rows x 26 columns]"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alphafold_accessibility = alphafold_accessibility.merge(\n",
    "    part_sphere_exposure, how='left', on=['protein_id','AA','position'])\n",
    "alphafold_accessibility"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 113/113 [00:00<00:00, 176.83it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>AA</th>\n",
       "      <th>position</th>\n",
       "      <th>nAA_6_180_pae</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O35129</td>\n",
       "      <td>M</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O35129</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O35129</td>\n",
       "      <td>Q</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O35129</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O35129</td>\n",
       "      <td>L</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52926</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>S</td>\n",
       "      <td>429</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52927</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>V</td>\n",
       "      <td>430</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52928</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>A</td>\n",
       "      <td>431</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52929</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>K</td>\n",
       "      <td>432</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52930</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>K</td>\n",
       "      <td>433</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52931 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      protein_id AA  position  nAA_6_180_pae\n",
       "0         O35129  M         1              1\n",
       "1         O35129  A         2              2\n",
       "2         O35129  Q         3              2\n",
       "3         O35129  N         4              2\n",
       "4         O35129  L         5              2\n",
       "...          ... ..       ...            ...\n",
       "52926     Q9Z2I8  S       429              2\n",
       "52927     Q9Z2I8  V       430              2\n",
       "52928     Q9Z2I8  A       431              2\n",
       "52929     Q9Z2I8  K       432              2\n",
       "52930     Q9Z2I8  K       433              1\n",
       "\n",
       "[52931 rows x 4 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# calculate full sphere exposure -> radius = 6\n",
    "full_sphere_exposure_6 = annotate_accessibility(\n",
    "    df=alphafold_annotation, \n",
    "    max_dist=6, \n",
    "    max_angle=180, \n",
    "    error_dir=pae_dir)\n",
    "full_sphere_exposure_6"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>protein_number</th>\n",
       "      <th>AA</th>\n",
       "      <th>position</th>\n",
       "      <th>quality</th>\n",
       "      <th>x_coord_c</th>\n",
       "      <th>x_coord_ca</th>\n",
       "      <th>x_coord_cb</th>\n",
       "      <th>x_coord_n</th>\n",
       "      <th>y_coord_c</th>\n",
       "      <th>...</th>\n",
       "      <th>secondary_structure</th>\n",
       "      <th>structure_group</th>\n",
       "      <th>BEND</th>\n",
       "      <th>HELX</th>\n",
       "      <th>STRN</th>\n",
       "      <th>TURN</th>\n",
       "      <th>unstructured</th>\n",
       "      <th>nAA_24_180_pae</th>\n",
       "      <th>nAA_12_70_pae</th>\n",
       "      <th>nAA_6_180_pae</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>1</td>\n",
       "      <td>45.38</td>\n",
       "      <td>-35.172</td>\n",
       "      <td>-33.707</td>\n",
       "      <td>-32.901</td>\n",
       "      <td>-33.031</td>\n",
       "      <td>3.159</td>\n",
       "      <td>...</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>50.78</td>\n",
       "      <td>-38.162</td>\n",
       "      <td>-37.117</td>\n",
       "      <td>-37.283</td>\n",
       "      <td>-35.709</td>\n",
       "      <td>2.229</td>\n",
       "      <td>...</td>\n",
       "      <td>TURN_TY1_P</td>\n",
       "      <td>TURN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>Q</td>\n",
       "      <td>3</td>\n",
       "      <td>52.40</td>\n",
       "      <td>-38.996</td>\n",
       "      <td>-38.671</td>\n",
       "      <td>-38.089</td>\n",
       "      <td>-37.769</td>\n",
       "      <td>4.596</td>\n",
       "      <td>...</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>51.23</td>\n",
       "      <td>-39.258</td>\n",
       "      <td>-38.274</td>\n",
       "      <td>-36.904</td>\n",
       "      <td>-38.080</td>\n",
       "      <td>7.104</td>\n",
       "      <td>...</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>L</td>\n",
       "      <td>5</td>\n",
       "      <td>50.02</td>\n",
       "      <td>-42.031</td>\n",
       "      <td>-40.574</td>\n",
       "      <td>-40.297</td>\n",
       "      <td>-39.604</td>\n",
       "      <td>6.273</td>\n",
       "      <td>...</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52926</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>S</td>\n",
       "      <td>429</td>\n",
       "      <td>94.38</td>\n",
       "      <td>26.176</td>\n",
       "      <td>27.466</td>\n",
       "      <td>27.350</td>\n",
       "      <td>27.774</td>\n",
       "      <td>14.096</td>\n",
       "      <td>...</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52927</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>V</td>\n",
       "      <td>430</td>\n",
       "      <td>91.54</td>\n",
       "      <td>24.398</td>\n",
       "      <td>24.087</td>\n",
       "      <td>23.021</td>\n",
       "      <td>25.313</td>\n",
       "      <td>16.707</td>\n",
       "      <td>...</td>\n",
       "      <td>BEND</td>\n",
       "      <td>BEND</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>78</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52928</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>A</td>\n",
       "      <td>431</td>\n",
       "      <td>73.79</td>\n",
       "      <td>23.429</td>\n",
       "      <td>23.961</td>\n",
       "      <td>23.247</td>\n",
       "      <td>23.808</td>\n",
       "      <td>19.412</td>\n",
       "      <td>...</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52929</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>K</td>\n",
       "      <td>432</td>\n",
       "      <td>62.38</td>\n",
       "      <td>22.744</td>\n",
       "      <td>23.748</td>\n",
       "      <td>24.902</td>\n",
       "      <td>24.238</td>\n",
       "      <td>22.331</td>\n",
       "      <td>...</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52930</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>K</td>\n",
       "      <td>433</td>\n",
       "      <td>51.30</td>\n",
       "      <td>20.532</td>\n",
       "      <td>20.426</td>\n",
       "      <td>19.015</td>\n",
       "      <td>21.457</td>\n",
       "      <td>24.555</td>\n",
       "      <td>...</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52931 rows × 27 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      protein_id  protein_number AA  position  quality  x_coord_c  x_coord_ca  \\\n",
       "0         O35129               1  M         1    45.38    -35.172     -33.707   \n",
       "1         O35129               1  A         2    50.78    -38.162     -37.117   \n",
       "2         O35129               1  Q         3    52.40    -38.996     -38.671   \n",
       "3         O35129               1  N         4    51.23    -39.258     -38.274   \n",
       "4         O35129               1  L         5    50.02    -42.031     -40.574   \n",
       "...          ...             ... ..       ...      ...        ...         ...   \n",
       "52926     Q9Z2I8             113  S       429    94.38     26.176      27.466   \n",
       "52927     Q9Z2I8             113  V       430    91.54     24.398      24.087   \n",
       "52928     Q9Z2I8             113  A       431    73.79     23.429      23.961   \n",
       "52929     Q9Z2I8             113  K       432    62.38     22.744      23.748   \n",
       "52930     Q9Z2I8             113  K       433    51.30     20.532      20.426   \n",
       "\n",
       "       x_coord_cb  x_coord_n  y_coord_c  ...  secondary_structure  \\\n",
       "0         -32.901    -33.031      3.159  ...         unstructured   \n",
       "1         -37.283    -35.709      2.229  ...           TURN_TY1_P   \n",
       "2         -38.089    -37.769      4.596  ...         HELX_RH_AL_P   \n",
       "3         -36.904    -38.080      7.104  ...         HELX_RH_AL_P   \n",
       "4         -40.297    -39.604      6.273  ...         HELX_RH_AL_P   \n",
       "...           ...        ...        ...  ...                  ...   \n",
       "52926      27.350     27.774     14.096  ...         HELX_RH_AL_P   \n",
       "52927      23.021     25.313     16.707  ...                 BEND   \n",
       "52928      23.247     23.808     19.412  ...         unstructured   \n",
       "52929      24.902     24.238     22.331  ...         unstructured   \n",
       "52930      19.015     21.457     24.555  ...         unstructured   \n",
       "\n",
       "       structure_group  BEND  HELX  STRN  TURN  unstructured nAA_24_180_pae  \\\n",
       "0         unstructured     0     0     0     0             1              7   \n",
       "1                 TURN     0     0     0     1             0              9   \n",
       "2                 HELX     0     1     0     0             0             10   \n",
       "3                 HELX     0     1     0     0             0             11   \n",
       "4                 HELX     0     1     0     0             0             11   \n",
       "...                ...   ...   ...   ...   ...           ...            ...   \n",
       "52926             HELX     0     1     0     0             0             91   \n",
       "52927             BEND     1     0     0     0             0             78   \n",
       "52928     unstructured     0     0     0     0             1             36   \n",
       "52929     unstructured     0     0     0     0             1             19   \n",
       "52930     unstructured     0     0     0     0             1             11   \n",
       "\n",
       "      nAA_12_70_pae  nAA_6_180_pae  \n",
       "0                 0              1  \n",
       "1                 0              2  \n",
       "2                 0              2  \n",
       "3                 1              2  \n",
       "4                 1              2  \n",
       "...             ...            ...  \n",
       "52926            12              2  \n",
       "52927             4              2  \n",
       "52928             0              2  \n",
       "52929             0              2  \n",
       "52930             0              1  \n",
       "\n",
       "[52931 rows x 27 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alphafold_accessibility = alphafold_accessibility.merge(\n",
    "    full_sphere_exposure_6, how='left', on=['protein_id','AA','position'])\n",
    "alphafold_accessibility"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 113/113 [00:00<00:00, 137.66it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>AA</th>\n",
       "      <th>position</th>\n",
       "      <th>nAA_12_180_pae</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O35129</td>\n",
       "      <td>M</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O35129</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O35129</td>\n",
       "      <td>Q</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O35129</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O35129</td>\n",
       "      <td>L</td>\n",
       "      <td>5</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52926</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>S</td>\n",
       "      <td>429</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52927</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>V</td>\n",
       "      <td>430</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52928</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>A</td>\n",
       "      <td>431</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52929</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>K</td>\n",
       "      <td>432</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52930</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>K</td>\n",
       "      <td>433</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52931 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      protein_id AA  position  nAA_12_180_pae\n",
       "0         O35129  M         1               3\n",
       "1         O35129  A         2               4\n",
       "2         O35129  Q         3               5\n",
       "3         O35129  N         4               6\n",
       "4         O35129  L         5               8\n",
       "...          ... ..       ...             ...\n",
       "52926     Q9Z2I8  S       429              18\n",
       "52927     Q9Z2I8  V       430              15\n",
       "52928     Q9Z2I8  A       431               5\n",
       "52929     Q9Z2I8  K       432               3\n",
       "52930     Q9Z2I8  K       433               2\n",
       "\n",
       "[52931 rows x 4 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# calculate full sphere exposure -> radius = 12\n",
    "full_sphere_exposure_12 = annotate_accessibility(\n",
    "    df=alphafold_annotation, \n",
    "    max_dist=12, \n",
    "    max_angle=180, \n",
    "    error_dir=pae_dir)\n",
    "full_sphere_exposure_12"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>protein_number</th>\n",
       "      <th>AA</th>\n",
       "      <th>position</th>\n",
       "      <th>quality</th>\n",
       "      <th>x_coord_c</th>\n",
       "      <th>x_coord_ca</th>\n",
       "      <th>x_coord_cb</th>\n",
       "      <th>x_coord_n</th>\n",
       "      <th>y_coord_c</th>\n",
       "      <th>...</th>\n",
       "      <th>structure_group</th>\n",
       "      <th>BEND</th>\n",
       "      <th>HELX</th>\n",
       "      <th>STRN</th>\n",
       "      <th>TURN</th>\n",
       "      <th>unstructured</th>\n",
       "      <th>nAA_24_180_pae</th>\n",
       "      <th>nAA_12_70_pae</th>\n",
       "      <th>nAA_6_180_pae</th>\n",
       "      <th>nAA_12_180_pae</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>1</td>\n",
       "      <td>45.38</td>\n",
       "      <td>-35.172</td>\n",
       "      <td>-33.707</td>\n",
       "      <td>-32.901</td>\n",
       "      <td>-33.031</td>\n",
       "      <td>3.159</td>\n",
       "      <td>...</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>50.78</td>\n",
       "      <td>-38.162</td>\n",
       "      <td>-37.117</td>\n",
       "      <td>-37.283</td>\n",
       "      <td>-35.709</td>\n",
       "      <td>2.229</td>\n",
       "      <td>...</td>\n",
       "      <td>TURN</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>Q</td>\n",
       "      <td>3</td>\n",
       "      <td>52.40</td>\n",
       "      <td>-38.996</td>\n",
       "      <td>-38.671</td>\n",
       "      <td>-38.089</td>\n",
       "      <td>-37.769</td>\n",
       "      <td>4.596</td>\n",
       "      <td>...</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>51.23</td>\n",
       "      <td>-39.258</td>\n",
       "      <td>-38.274</td>\n",
       "      <td>-36.904</td>\n",
       "      <td>-38.080</td>\n",
       "      <td>7.104</td>\n",
       "      <td>...</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>L</td>\n",
       "      <td>5</td>\n",
       "      <td>50.02</td>\n",
       "      <td>-42.031</td>\n",
       "      <td>-40.574</td>\n",
       "      <td>-40.297</td>\n",
       "      <td>-39.604</td>\n",
       "      <td>6.273</td>\n",
       "      <td>...</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52926</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>S</td>\n",
       "      <td>429</td>\n",
       "      <td>94.38</td>\n",
       "      <td>26.176</td>\n",
       "      <td>27.466</td>\n",
       "      <td>27.350</td>\n",
       "      <td>27.774</td>\n",
       "      <td>14.096</td>\n",
       "      <td>...</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>18</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52927</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>V</td>\n",
       "      <td>430</td>\n",
       "      <td>91.54</td>\n",
       "      <td>24.398</td>\n",
       "      <td>24.087</td>\n",
       "      <td>23.021</td>\n",
       "      <td>25.313</td>\n",
       "      <td>16.707</td>\n",
       "      <td>...</td>\n",
       "      <td>BEND</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>78</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52928</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>A</td>\n",
       "      <td>431</td>\n",
       "      <td>73.79</td>\n",
       "      <td>23.429</td>\n",
       "      <td>23.961</td>\n",
       "      <td>23.247</td>\n",
       "      <td>23.808</td>\n",
       "      <td>19.412</td>\n",
       "      <td>...</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52929</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>K</td>\n",
       "      <td>432</td>\n",
       "      <td>62.38</td>\n",
       "      <td>22.744</td>\n",
       "      <td>23.748</td>\n",
       "      <td>24.902</td>\n",
       "      <td>24.238</td>\n",
       "      <td>22.331</td>\n",
       "      <td>...</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52930</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>K</td>\n",
       "      <td>433</td>\n",
       "      <td>51.30</td>\n",
       "      <td>20.532</td>\n",
       "      <td>20.426</td>\n",
       "      <td>19.015</td>\n",
       "      <td>21.457</td>\n",
       "      <td>24.555</td>\n",
       "      <td>...</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52931 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      protein_id  protein_number AA  position  quality  x_coord_c  x_coord_ca  \\\n",
       "0         O35129               1  M         1    45.38    -35.172     -33.707   \n",
       "1         O35129               1  A         2    50.78    -38.162     -37.117   \n",
       "2         O35129               1  Q         3    52.40    -38.996     -38.671   \n",
       "3         O35129               1  N         4    51.23    -39.258     -38.274   \n",
       "4         O35129               1  L         5    50.02    -42.031     -40.574   \n",
       "...          ...             ... ..       ...      ...        ...         ...   \n",
       "52926     Q9Z2I8             113  S       429    94.38     26.176      27.466   \n",
       "52927     Q9Z2I8             113  V       430    91.54     24.398      24.087   \n",
       "52928     Q9Z2I8             113  A       431    73.79     23.429      23.961   \n",
       "52929     Q9Z2I8             113  K       432    62.38     22.744      23.748   \n",
       "52930     Q9Z2I8             113  K       433    51.30     20.532      20.426   \n",
       "\n",
       "       x_coord_cb  x_coord_n  y_coord_c  ...  structure_group  BEND  HELX  \\\n",
       "0         -32.901    -33.031      3.159  ...     unstructured     0     0   \n",
       "1         -37.283    -35.709      2.229  ...             TURN     0     0   \n",
       "2         -38.089    -37.769      4.596  ...             HELX     0     1   \n",
       "3         -36.904    -38.080      7.104  ...             HELX     0     1   \n",
       "4         -40.297    -39.604      6.273  ...             HELX     0     1   \n",
       "...           ...        ...        ...  ...              ...   ...   ...   \n",
       "52926      27.350     27.774     14.096  ...             HELX     0     1   \n",
       "52927      23.021     25.313     16.707  ...             BEND     1     0   \n",
       "52928      23.247     23.808     19.412  ...     unstructured     0     0   \n",
       "52929      24.902     24.238     22.331  ...     unstructured     0     0   \n",
       "52930      19.015     21.457     24.555  ...     unstructured     0     0   \n",
       "\n",
       "       STRN  TURN  unstructured  nAA_24_180_pae nAA_12_70_pae nAA_6_180_pae  \\\n",
       "0         0     0             1               7             0             1   \n",
       "1         0     1             0               9             0             2   \n",
       "2         0     0             0              10             0             2   \n",
       "3         0     0             0              11             1             2   \n",
       "4         0     0             0              11             1             2   \n",
       "...     ...   ...           ...             ...           ...           ...   \n",
       "52926     0     0             0              91            12             2   \n",
       "52927     0     0             0              78             4             2   \n",
       "52928     0     0             1              36             0             2   \n",
       "52929     0     0             1              19             0             2   \n",
       "52930     0     0             1              11             0             1   \n",
       "\n",
       "       nAA_12_180_pae  \n",
       "0                   3  \n",
       "1                   4  \n",
       "2                   5  \n",
       "3                   6  \n",
       "4                   8  \n",
       "...               ...  \n",
       "52926              18  \n",
       "52927              15  \n",
       "52928               5  \n",
       "52929               3  \n",
       "52930               2  \n",
       "\n",
       "[52931 rows x 28 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alphafold_accessibility = alphafold_accessibility.merge(\n",
    "    full_sphere_exposure_12, how='left', on=['protein_id','AA','position'])\n",
    "alphafold_accessibility"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 113/113 [00:01<00:00, 95.15it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>AA</th>\n",
       "      <th>position</th>\n",
       "      <th>nAA_18_180_pae</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O35129</td>\n",
       "      <td>M</td>\n",
       "      <td>1</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O35129</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O35129</td>\n",
       "      <td>Q</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O35129</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O35129</td>\n",
       "      <td>L</td>\n",
       "      <td>5</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52926</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>S</td>\n",
       "      <td>429</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52927</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>V</td>\n",
       "      <td>430</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52928</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>A</td>\n",
       "      <td>431</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52929</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>K</td>\n",
       "      <td>432</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52930</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>K</td>\n",
       "      <td>433</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52931 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      protein_id AA  position  nAA_18_180_pae\n",
       "0         O35129  M         1               5\n",
       "1         O35129  A         2               6\n",
       "2         O35129  Q         3               7\n",
       "3         O35129  N         4               9\n",
       "4         O35129  L         5               9\n",
       "...          ... ..       ...             ...\n",
       "52926     Q9Z2I8  S       429              53\n",
       "52927     Q9Z2I8  V       430              44\n",
       "52928     Q9Z2I8  A       431              15\n",
       "52929     Q9Z2I8  K       432               7\n",
       "52930     Q9Z2I8  K       433               3\n",
       "\n",
       "[52931 rows x 4 columns]"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# calculate full sphere exposure -> radius = 18\n",
    "full_sphere_exposure_18 = annotate_accessibility(\n",
    "    df=alphafold_annotation, \n",
    "    max_dist=18, \n",
    "    max_angle=180, \n",
    "    error_dir=pae_dir)\n",
    "full_sphere_exposure_18"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>protein_number</th>\n",
       "      <th>AA</th>\n",
       "      <th>position</th>\n",
       "      <th>quality</th>\n",
       "      <th>x_coord_c</th>\n",
       "      <th>x_coord_ca</th>\n",
       "      <th>x_coord_cb</th>\n",
       "      <th>x_coord_n</th>\n",
       "      <th>y_coord_c</th>\n",
       "      <th>...</th>\n",
       "      <th>BEND</th>\n",
       "      <th>HELX</th>\n",
       "      <th>STRN</th>\n",
       "      <th>TURN</th>\n",
       "      <th>unstructured</th>\n",
       "      <th>nAA_24_180_pae</th>\n",
       "      <th>nAA_12_70_pae</th>\n",
       "      <th>nAA_6_180_pae</th>\n",
       "      <th>nAA_12_180_pae</th>\n",
       "      <th>nAA_18_180_pae</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>1</td>\n",
       "      <td>45.38</td>\n",
       "      <td>-35.172</td>\n",
       "      <td>-33.707</td>\n",
       "      <td>-32.901</td>\n",
       "      <td>-33.031</td>\n",
       "      <td>3.159</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>50.78</td>\n",
       "      <td>-38.162</td>\n",
       "      <td>-37.117</td>\n",
       "      <td>-37.283</td>\n",
       "      <td>-35.709</td>\n",
       "      <td>2.229</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>Q</td>\n",
       "      <td>3</td>\n",
       "      <td>52.40</td>\n",
       "      <td>-38.996</td>\n",
       "      <td>-38.671</td>\n",
       "      <td>-38.089</td>\n",
       "      <td>-37.769</td>\n",
       "      <td>4.596</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>51.23</td>\n",
       "      <td>-39.258</td>\n",
       "      <td>-38.274</td>\n",
       "      <td>-36.904</td>\n",
       "      <td>-38.080</td>\n",
       "      <td>7.104</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>L</td>\n",
       "      <td>5</td>\n",
       "      <td>50.02</td>\n",
       "      <td>-42.031</td>\n",
       "      <td>-40.574</td>\n",
       "      <td>-40.297</td>\n",
       "      <td>-39.604</td>\n",
       "      <td>6.273</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52926</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>S</td>\n",
       "      <td>429</td>\n",
       "      <td>94.38</td>\n",
       "      <td>26.176</td>\n",
       "      <td>27.466</td>\n",
       "      <td>27.350</td>\n",
       "      <td>27.774</td>\n",
       "      <td>14.096</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>18</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52927</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>V</td>\n",
       "      <td>430</td>\n",
       "      <td>91.54</td>\n",
       "      <td>24.398</td>\n",
       "      <td>24.087</td>\n",
       "      <td>23.021</td>\n",
       "      <td>25.313</td>\n",
       "      <td>16.707</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>78</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52928</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>A</td>\n",
       "      <td>431</td>\n",
       "      <td>73.79</td>\n",
       "      <td>23.429</td>\n",
       "      <td>23.961</td>\n",
       "      <td>23.247</td>\n",
       "      <td>23.808</td>\n",
       "      <td>19.412</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52929</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>K</td>\n",
       "      <td>432</td>\n",
       "      <td>62.38</td>\n",
       "      <td>22.744</td>\n",
       "      <td>23.748</td>\n",
       "      <td>24.902</td>\n",
       "      <td>24.238</td>\n",
       "      <td>22.331</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52930</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>K</td>\n",
       "      <td>433</td>\n",
       "      <td>51.30</td>\n",
       "      <td>20.532</td>\n",
       "      <td>20.426</td>\n",
       "      <td>19.015</td>\n",
       "      <td>21.457</td>\n",
       "      <td>24.555</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52931 rows × 29 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      protein_id  protein_number AA  position  quality  x_coord_c  x_coord_ca  \\\n",
       "0         O35129               1  M         1    45.38    -35.172     -33.707   \n",
       "1         O35129               1  A         2    50.78    -38.162     -37.117   \n",
       "2         O35129               1  Q         3    52.40    -38.996     -38.671   \n",
       "3         O35129               1  N         4    51.23    -39.258     -38.274   \n",
       "4         O35129               1  L         5    50.02    -42.031     -40.574   \n",
       "...          ...             ... ..       ...      ...        ...         ...   \n",
       "52926     Q9Z2I8             113  S       429    94.38     26.176      27.466   \n",
       "52927     Q9Z2I8             113  V       430    91.54     24.398      24.087   \n",
       "52928     Q9Z2I8             113  A       431    73.79     23.429      23.961   \n",
       "52929     Q9Z2I8             113  K       432    62.38     22.744      23.748   \n",
       "52930     Q9Z2I8             113  K       433    51.30     20.532      20.426   \n",
       "\n",
       "       x_coord_cb  x_coord_n  y_coord_c  ...  BEND  HELX  STRN  TURN  \\\n",
       "0         -32.901    -33.031      3.159  ...     0     0     0     0   \n",
       "1         -37.283    -35.709      2.229  ...     0     0     0     1   \n",
       "2         -38.089    -37.769      4.596  ...     0     1     0     0   \n",
       "3         -36.904    -38.080      7.104  ...     0     1     0     0   \n",
       "4         -40.297    -39.604      6.273  ...     0     1     0     0   \n",
       "...           ...        ...        ...  ...   ...   ...   ...   ...   \n",
       "52926      27.350     27.774     14.096  ...     0     1     0     0   \n",
       "52927      23.021     25.313     16.707  ...     1     0     0     0   \n",
       "52928      23.247     23.808     19.412  ...     0     0     0     0   \n",
       "52929      24.902     24.238     22.331  ...     0     0     0     0   \n",
       "52930      19.015     21.457     24.555  ...     0     0     0     0   \n",
       "\n",
       "       unstructured  nAA_24_180_pae  nAA_12_70_pae nAA_6_180_pae  \\\n",
       "0                 1               7              0             1   \n",
       "1                 0               9              0             2   \n",
       "2                 0              10              0             2   \n",
       "3                 0              11              1             2   \n",
       "4                 0              11              1             2   \n",
       "...             ...             ...            ...           ...   \n",
       "52926             0              91             12             2   \n",
       "52927             0              78              4             2   \n",
       "52928             1              36              0             2   \n",
       "52929             1              19              0             2   \n",
       "52930             1              11              0             1   \n",
       "\n",
       "      nAA_12_180_pae  nAA_18_180_pae  \n",
       "0                  3               5  \n",
       "1                  4               6  \n",
       "2                  5               7  \n",
       "3                  6               9  \n",
       "4                  8               9  \n",
       "...              ...             ...  \n",
       "52926             18              53  \n",
       "52927             15              44  \n",
       "52928              5              15  \n",
       "52929              3               7  \n",
       "52930              2               3  \n",
       "\n",
       "[52931 rows x 29 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alphafold_accessibility = alphafold_accessibility.merge(\n",
    "    full_sphere_exposure_18, how='left', on=['protein_id','AA','position'])\n",
    "alphafold_accessibility"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 113/113 [00:00<00:00, 669.87it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>protein_number</th>\n",
       "      <th>AA</th>\n",
       "      <th>position</th>\n",
       "      <th>quality</th>\n",
       "      <th>x_coord_c</th>\n",
       "      <th>x_coord_ca</th>\n",
       "      <th>x_coord_cb</th>\n",
       "      <th>x_coord_n</th>\n",
       "      <th>y_coord_c</th>\n",
       "      <th>...</th>\n",
       "      <th>unstructured</th>\n",
       "      <th>nAA_24_180_pae</th>\n",
       "      <th>nAA_12_70_pae</th>\n",
       "      <th>nAA_6_180_pae</th>\n",
       "      <th>nAA_12_180_pae</th>\n",
       "      <th>nAA_18_180_pae</th>\n",
       "      <th>nAA_6_180_pae_smooth10</th>\n",
       "      <th>nAA_12_180_pae_smooth10</th>\n",
       "      <th>nAA_18_180_pae_smooth10</th>\n",
       "      <th>nAA_24_180_pae_smooth10</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>1</td>\n",
       "      <td>45.38</td>\n",
       "      <td>-35.172</td>\n",
       "      <td>-33.707</td>\n",
       "      <td>-32.901</td>\n",
       "      <td>-33.031</td>\n",
       "      <td>3.159</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>1.909091</td>\n",
       "      <td>6.000000</td>\n",
       "      <td>8.727273</td>\n",
       "      <td>11.181818</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>50.78</td>\n",
       "      <td>-38.162</td>\n",
       "      <td>-37.117</td>\n",
       "      <td>-37.283</td>\n",
       "      <td>-35.709</td>\n",
       "      <td>2.229</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1.916667</td>\n",
       "      <td>5.916667</td>\n",
       "      <td>8.666667</td>\n",
       "      <td>11.416667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>Q</td>\n",
       "      <td>3</td>\n",
       "      <td>52.40</td>\n",
       "      <td>-38.996</td>\n",
       "      <td>-38.671</td>\n",
       "      <td>-38.089</td>\n",
       "      <td>-37.769</td>\n",
       "      <td>4.596</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>1.923077</td>\n",
       "      <td>5.769231</td>\n",
       "      <td>8.538462</td>\n",
       "      <td>11.538462</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>51.23</td>\n",
       "      <td>-39.258</td>\n",
       "      <td>-38.274</td>\n",
       "      <td>-36.904</td>\n",
       "      <td>-38.080</td>\n",
       "      <td>7.104</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "      <td>1.857143</td>\n",
       "      <td>5.642857</td>\n",
       "      <td>8.500000</td>\n",
       "      <td>11.571429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>L</td>\n",
       "      <td>5</td>\n",
       "      <td>50.02</td>\n",
       "      <td>-42.031</td>\n",
       "      <td>-40.574</td>\n",
       "      <td>-40.297</td>\n",
       "      <td>-39.604</td>\n",
       "      <td>6.273</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>1.800000</td>\n",
       "      <td>5.466667</td>\n",
       "      <td>8.466667</td>\n",
       "      <td>11.733333</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>428</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>S</td>\n",
       "      <td>429</td>\n",
       "      <td>94.38</td>\n",
       "      <td>26.176</td>\n",
       "      <td>27.466</td>\n",
       "      <td>27.350</td>\n",
       "      <td>27.774</td>\n",
       "      <td>14.096</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>91</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>18</td>\n",
       "      <td>53</td>\n",
       "      <td>2.133333</td>\n",
       "      <td>20.666667</td>\n",
       "      <td>47.266667</td>\n",
       "      <td>85.200000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>429</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>V</td>\n",
       "      <td>430</td>\n",
       "      <td>91.54</td>\n",
       "      <td>24.398</td>\n",
       "      <td>24.087</td>\n",
       "      <td>23.021</td>\n",
       "      <td>25.313</td>\n",
       "      <td>16.707</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>78</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>44</td>\n",
       "      <td>2.071429</td>\n",
       "      <td>20.285714</td>\n",
       "      <td>46.357143</td>\n",
       "      <td>83.642857</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>A</td>\n",
       "      <td>431</td>\n",
       "      <td>73.79</td>\n",
       "      <td>23.429</td>\n",
       "      <td>23.961</td>\n",
       "      <td>23.247</td>\n",
       "      <td>23.808</td>\n",
       "      <td>19.412</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>2.076923</td>\n",
       "      <td>20.307692</td>\n",
       "      <td>46.153846</td>\n",
       "      <td>83.307692</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>K</td>\n",
       "      <td>432</td>\n",
       "      <td>62.38</td>\n",
       "      <td>22.744</td>\n",
       "      <td>23.748</td>\n",
       "      <td>24.902</td>\n",
       "      <td>24.238</td>\n",
       "      <td>22.331</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>2.083333</td>\n",
       "      <td>20.333333</td>\n",
       "      <td>45.583333</td>\n",
       "      <td>81.916667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>K</td>\n",
       "      <td>433</td>\n",
       "      <td>51.30</td>\n",
       "      <td>20.532</td>\n",
       "      <td>20.426</td>\n",
       "      <td>19.015</td>\n",
       "      <td>21.457</td>\n",
       "      <td>24.555</td>\n",
       "      <td>...</td>\n",
       "      <td>1</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>1.909091</td>\n",
       "      <td>19.636364</td>\n",
       "      <td>43.454545</td>\n",
       "      <td>78.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52931 rows × 33 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    protein_id  protein_number AA  position  quality  x_coord_c  x_coord_ca  \\\n",
       "0       O35129               1  M         1    45.38    -35.172     -33.707   \n",
       "1       O35129               1  A         2    50.78    -38.162     -37.117   \n",
       "2       O35129               1  Q         3    52.40    -38.996     -38.671   \n",
       "3       O35129               1  N         4    51.23    -39.258     -38.274   \n",
       "4       O35129               1  L         5    50.02    -42.031     -40.574   \n",
       "..         ...             ... ..       ...      ...        ...         ...   \n",
       "428     Q9Z2I8             113  S       429    94.38     26.176      27.466   \n",
       "429     Q9Z2I8             113  V       430    91.54     24.398      24.087   \n",
       "430     Q9Z2I8             113  A       431    73.79     23.429      23.961   \n",
       "431     Q9Z2I8             113  K       432    62.38     22.744      23.748   \n",
       "432     Q9Z2I8             113  K       433    51.30     20.532      20.426   \n",
       "\n",
       "     x_coord_cb  x_coord_n  y_coord_c  ...  unstructured  nAA_24_180_pae  \\\n",
       "0       -32.901    -33.031      3.159  ...             1               7   \n",
       "1       -37.283    -35.709      2.229  ...             0               9   \n",
       "2       -38.089    -37.769      4.596  ...             0              10   \n",
       "3       -36.904    -38.080      7.104  ...             0              11   \n",
       "4       -40.297    -39.604      6.273  ...             0              11   \n",
       "..          ...        ...        ...  ...           ...             ...   \n",
       "428      27.350     27.774     14.096  ...             0              91   \n",
       "429      23.021     25.313     16.707  ...             0              78   \n",
       "430      23.247     23.808     19.412  ...             1              36   \n",
       "431      24.902     24.238     22.331  ...             1              19   \n",
       "432      19.015     21.457     24.555  ...             1              11   \n",
       "\n",
       "     nAA_12_70_pae  nAA_6_180_pae  nAA_12_180_pae  nAA_18_180_pae  \\\n",
       "0                0              1               3               5   \n",
       "1                0              2               4               6   \n",
       "2                0              2               5               7   \n",
       "3                1              2               6               9   \n",
       "4                1              2               8               9   \n",
       "..             ...            ...             ...             ...   \n",
       "428             12              2              18              53   \n",
       "429              4              2              15              44   \n",
       "430              0              2               5              15   \n",
       "431              0              2               3               7   \n",
       "432              0              1               2               3   \n",
       "\n",
       "     nAA_6_180_pae_smooth10 nAA_12_180_pae_smooth10 nAA_18_180_pae_smooth10  \\\n",
       "0                  1.909091                6.000000                8.727273   \n",
       "1                  1.916667                5.916667                8.666667   \n",
       "2                  1.923077                5.769231                8.538462   \n",
       "3                  1.857143                5.642857                8.500000   \n",
       "4                  1.800000                5.466667                8.466667   \n",
       "..                      ...                     ...                     ...   \n",
       "428                2.133333               20.666667               47.266667   \n",
       "429                2.071429               20.285714               46.357143   \n",
       "430                2.076923               20.307692               46.153846   \n",
       "431                2.083333               20.333333               45.583333   \n",
       "432                1.909091               19.636364               43.454545   \n",
       "\n",
       "     nAA_24_180_pae_smooth10  \n",
       "0                  11.181818  \n",
       "1                  11.416667  \n",
       "2                  11.538462  \n",
       "3                  11.571429  \n",
       "4                  11.733333  \n",
       "..                       ...  \n",
       "428                85.200000  \n",
       "429                83.642857  \n",
       "430                83.307692  \n",
       "431                81.916667  \n",
       "432                78.000000  \n",
       "\n",
       "[52931 rows x 33 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alphafold_accessibility_smooth = get_smooth_score(\n",
    "    alphafold_accessibility, \n",
    "    np.array(['nAA_6_180_pae', 'nAA_12_180_pae', 'nAA_18_180_pae', 'nAA_24_180_pae']), \n",
    "    [10])\n",
    "alphafold_accessibility_smooth"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>protein_number</th>\n",
       "      <th>AA</th>\n",
       "      <th>position</th>\n",
       "      <th>quality</th>\n",
       "      <th>x_coord_c</th>\n",
       "      <th>x_coord_ca</th>\n",
       "      <th>x_coord_cb</th>\n",
       "      <th>x_coord_n</th>\n",
       "      <th>y_coord_c</th>\n",
       "      <th>...</th>\n",
       "      <th>nAA_24_180_pae</th>\n",
       "      <th>nAA_12_70_pae</th>\n",
       "      <th>nAA_6_180_pae</th>\n",
       "      <th>nAA_12_180_pae</th>\n",
       "      <th>nAA_18_180_pae</th>\n",
       "      <th>nAA_6_180_pae_smooth10</th>\n",
       "      <th>nAA_12_180_pae_smooth10</th>\n",
       "      <th>nAA_18_180_pae_smooth10</th>\n",
       "      <th>nAA_24_180_pae_smooth10</th>\n",
       "      <th>IDR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>1</td>\n",
       "      <td>45.38</td>\n",
       "      <td>-35.172</td>\n",
       "      <td>-33.707</td>\n",
       "      <td>-32.901</td>\n",
       "      <td>-33.031</td>\n",
       "      <td>3.159</td>\n",
       "      <td>...</td>\n",
       "      <td>7</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>5</td>\n",
       "      <td>1.909091</td>\n",
       "      <td>6.000000</td>\n",
       "      <td>8.727273</td>\n",
       "      <td>11.181818</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>A</td>\n",
       "      <td>2</td>\n",
       "      <td>50.78</td>\n",
       "      <td>-38.162</td>\n",
       "      <td>-37.117</td>\n",
       "      <td>-37.283</td>\n",
       "      <td>-35.709</td>\n",
       "      <td>2.229</td>\n",
       "      <td>...</td>\n",
       "      <td>9</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>4</td>\n",
       "      <td>6</td>\n",
       "      <td>1.916667</td>\n",
       "      <td>5.916667</td>\n",
       "      <td>8.666667</td>\n",
       "      <td>11.416667</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>Q</td>\n",
       "      <td>3</td>\n",
       "      <td>52.40</td>\n",
       "      <td>-38.996</td>\n",
       "      <td>-38.671</td>\n",
       "      <td>-38.089</td>\n",
       "      <td>-37.769</td>\n",
       "      <td>4.596</td>\n",
       "      <td>...</td>\n",
       "      <td>10</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>7</td>\n",
       "      <td>1.923077</td>\n",
       "      <td>5.769231</td>\n",
       "      <td>8.538462</td>\n",
       "      <td>11.538462</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>N</td>\n",
       "      <td>4</td>\n",
       "      <td>51.23</td>\n",
       "      <td>-39.258</td>\n",
       "      <td>-38.274</td>\n",
       "      <td>-36.904</td>\n",
       "      <td>-38.080</td>\n",
       "      <td>7.104</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>6</td>\n",
       "      <td>9</td>\n",
       "      <td>1.857143</td>\n",
       "      <td>5.642857</td>\n",
       "      <td>8.500000</td>\n",
       "      <td>11.571429</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>O35129</td>\n",
       "      <td>1</td>\n",
       "      <td>L</td>\n",
       "      <td>5</td>\n",
       "      <td>50.02</td>\n",
       "      <td>-42.031</td>\n",
       "      <td>-40.574</td>\n",
       "      <td>-40.297</td>\n",
       "      <td>-39.604</td>\n",
       "      <td>6.273</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>8</td>\n",
       "      <td>9</td>\n",
       "      <td>1.800000</td>\n",
       "      <td>5.466667</td>\n",
       "      <td>8.466667</td>\n",
       "      <td>11.733333</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>428</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>S</td>\n",
       "      <td>429</td>\n",
       "      <td>94.38</td>\n",
       "      <td>26.176</td>\n",
       "      <td>27.466</td>\n",
       "      <td>27.350</td>\n",
       "      <td>27.774</td>\n",
       "      <td>14.096</td>\n",
       "      <td>...</td>\n",
       "      <td>91</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>18</td>\n",
       "      <td>53</td>\n",
       "      <td>2.133333</td>\n",
       "      <td>20.666667</td>\n",
       "      <td>47.266667</td>\n",
       "      <td>85.200000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>429</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>V</td>\n",
       "      <td>430</td>\n",
       "      <td>91.54</td>\n",
       "      <td>24.398</td>\n",
       "      <td>24.087</td>\n",
       "      <td>23.021</td>\n",
       "      <td>25.313</td>\n",
       "      <td>16.707</td>\n",
       "      <td>...</td>\n",
       "      <td>78</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>44</td>\n",
       "      <td>2.071429</td>\n",
       "      <td>20.285714</td>\n",
       "      <td>46.357143</td>\n",
       "      <td>83.642857</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>430</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>A</td>\n",
       "      <td>431</td>\n",
       "      <td>73.79</td>\n",
       "      <td>23.429</td>\n",
       "      <td>23.961</td>\n",
       "      <td>23.247</td>\n",
       "      <td>23.808</td>\n",
       "      <td>19.412</td>\n",
       "      <td>...</td>\n",
       "      <td>36</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>5</td>\n",
       "      <td>15</td>\n",
       "      <td>2.076923</td>\n",
       "      <td>20.307692</td>\n",
       "      <td>46.153846</td>\n",
       "      <td>83.307692</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>431</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>K</td>\n",
       "      <td>432</td>\n",
       "      <td>62.38</td>\n",
       "      <td>22.744</td>\n",
       "      <td>23.748</td>\n",
       "      <td>24.902</td>\n",
       "      <td>24.238</td>\n",
       "      <td>22.331</td>\n",
       "      <td>...</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>7</td>\n",
       "      <td>2.083333</td>\n",
       "      <td>20.333333</td>\n",
       "      <td>45.583333</td>\n",
       "      <td>81.916667</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>432</th>\n",
       "      <td>Q9Z2I8</td>\n",
       "      <td>113</td>\n",
       "      <td>K</td>\n",
       "      <td>433</td>\n",
       "      <td>51.30</td>\n",
       "      <td>20.532</td>\n",
       "      <td>20.426</td>\n",
       "      <td>19.015</td>\n",
       "      <td>21.457</td>\n",
       "      <td>24.555</td>\n",
       "      <td>...</td>\n",
       "      <td>11</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>1.909091</td>\n",
       "      <td>19.636364</td>\n",
       "      <td>43.454545</td>\n",
       "      <td>78.000000</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52931 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    protein_id  protein_number AA  position  quality  x_coord_c  x_coord_ca  \\\n",
       "0       O35129               1  M         1    45.38    -35.172     -33.707   \n",
       "1       O35129               1  A         2    50.78    -38.162     -37.117   \n",
       "2       O35129               1  Q         3    52.40    -38.996     -38.671   \n",
       "3       O35129               1  N         4    51.23    -39.258     -38.274   \n",
       "4       O35129               1  L         5    50.02    -42.031     -40.574   \n",
       "..         ...             ... ..       ...      ...        ...         ...   \n",
       "428     Q9Z2I8             113  S       429    94.38     26.176      27.466   \n",
       "429     Q9Z2I8             113  V       430    91.54     24.398      24.087   \n",
       "430     Q9Z2I8             113  A       431    73.79     23.429      23.961   \n",
       "431     Q9Z2I8             113  K       432    62.38     22.744      23.748   \n",
       "432     Q9Z2I8             113  K       433    51.30     20.532      20.426   \n",
       "\n",
       "     x_coord_cb  x_coord_n  y_coord_c  ...  nAA_24_180_pae  nAA_12_70_pae  \\\n",
       "0       -32.901    -33.031      3.159  ...               7              0   \n",
       "1       -37.283    -35.709      2.229  ...               9              0   \n",
       "2       -38.089    -37.769      4.596  ...              10              0   \n",
       "3       -36.904    -38.080      7.104  ...              11              1   \n",
       "4       -40.297    -39.604      6.273  ...              11              1   \n",
       "..          ...        ...        ...  ...             ...            ...   \n",
       "428      27.350     27.774     14.096  ...              91             12   \n",
       "429      23.021     25.313     16.707  ...              78              4   \n",
       "430      23.247     23.808     19.412  ...              36              0   \n",
       "431      24.902     24.238     22.331  ...              19              0   \n",
       "432      19.015     21.457     24.555  ...              11              0   \n",
       "\n",
       "     nAA_6_180_pae  nAA_12_180_pae  nAA_18_180_pae  nAA_6_180_pae_smooth10  \\\n",
       "0                1               3               5                1.909091   \n",
       "1                2               4               6                1.916667   \n",
       "2                2               5               7                1.923077   \n",
       "3                2               6               9                1.857143   \n",
       "4                2               8               9                1.800000   \n",
       "..             ...             ...             ...                     ...   \n",
       "428              2              18              53                2.133333   \n",
       "429              2              15              44                2.071429   \n",
       "430              2               5              15                2.076923   \n",
       "431              2               3               7                2.083333   \n",
       "432              1               2               3                1.909091   \n",
       "\n",
       "     nAA_12_180_pae_smooth10 nAA_18_180_pae_smooth10 nAA_24_180_pae_smooth10  \\\n",
       "0                   6.000000                8.727273               11.181818   \n",
       "1                   5.916667                8.666667               11.416667   \n",
       "2                   5.769231                8.538462               11.538462   \n",
       "3                   5.642857                8.500000               11.571429   \n",
       "4                   5.466667                8.466667               11.733333   \n",
       "..                       ...                     ...                     ...   \n",
       "428                20.666667               47.266667               85.200000   \n",
       "429                20.285714               46.357143               83.642857   \n",
       "430                20.307692               46.153846               83.307692   \n",
       "431                20.333333               45.583333               81.916667   \n",
       "432                19.636364               43.454545               78.000000   \n",
       "\n",
       "     IDR  \n",
       "0      1  \n",
       "1      1  \n",
       "2      1  \n",
       "3      1  \n",
       "4      1  \n",
       "..   ...  \n",
       "428    0  \n",
       "429    0  \n",
       "430    0  \n",
       "431    0  \n",
       "432    0  \n",
       "\n",
       "[52931 rows x 34 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alphafold_accessibility_smooth['IDR'] = np.where(\n",
    "    alphafold_accessibility_smooth['nAA_24_180_pae_smooth10']<=34.27, 1, 0)\n",
    "alphafold_accessibility_smooth"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge Dataframes into Full Dataset (Includes Alphafold) - Labeled Methionines"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide Sequence</th>\n",
       "      <th>Modified Peptide</th>\n",
       "      <th>Light Modified Peptide</th>\n",
       "      <th>Heavy Modified Peptide</th>\n",
       "      <th>1 Log2 Ratio HL</th>\n",
       "      <th>10 Log2 Ratio HL</th>\n",
       "      <th>11 Log2 Ratio HL</th>\n",
       "      <th>12 Log2 Ratio HL</th>\n",
       "      <th>2 Log2 Ratio HL</th>\n",
       "      <th>3 Log2 Ratio HL</th>\n",
       "      <th>...</th>\n",
       "      <th>nAA_24_180_pae</th>\n",
       "      <th>nAA_12_70_pae</th>\n",
       "      <th>nAA_6_180_pae</th>\n",
       "      <th>nAA_12_180_pae</th>\n",
       "      <th>nAA_18_180_pae</th>\n",
       "      <th>nAA_6_180_pae_smooth10</th>\n",
       "      <th>nAA_12_180_pae_smooth10</th>\n",
       "      <th>nAA_18_180_pae_smooth10</th>\n",
       "      <th>nAA_24_180_pae_smooth10</th>\n",
       "      <th>IDR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVM[649.3660]IR</td>\n",
       "      <td>AADTIGYPVM[655.3735]IR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.269016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>144</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>31</td>\n",
       "      <td>85</td>\n",
       "      <td>2.380952</td>\n",
       "      <td>22.190476</td>\n",
       "      <td>59.952381</td>\n",
       "      <td>120.857143</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAM[649.3660]QTLDMGR</td>\n",
       "      <td>IAM[655.3735]QTLDMGR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.783695</td>\n",
       "      <td>3.114945</td>\n",
       "      <td>2.697822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>173</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>19</td>\n",
       "      <td>90</td>\n",
       "      <td>2.761905</td>\n",
       "      <td>30.571429</td>\n",
       "      <td>90.857143</td>\n",
       "      <td>176.571429</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIM[649.3660]EK</td>\n",
       "      <td>FVGAVDPIM[655.3735]EK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.383482</td>\n",
       "      <td>...</td>\n",
       "      <td>52</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>24</td>\n",
       "      <td>2.238095</td>\n",
       "      <td>11.238095</td>\n",
       "      <td>29.571429</td>\n",
       "      <td>61.952381</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[649.3660]PINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[655.3735]PINGPFKPDHYRY</td>\n",
       "      <td>2.394458</td>\n",
       "      <td>2.380664</td>\n",
       "      <td>2.682897</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.435014</td>\n",
       "      <td>2.412394</td>\n",
       "      <td>...</td>\n",
       "      <td>39</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>24</td>\n",
       "      <td>2.095238</td>\n",
       "      <td>13.380952</td>\n",
       "      <td>27.047619</td>\n",
       "      <td>48.190476</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPM[649.3660]NLPHR</td>\n",
       "      <td>FADVIPM[655.3735]NLPHR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.580647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.446614</td>\n",
       "      <td>2.450718</td>\n",
       "      <td>...</td>\n",
       "      <td>207</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>27</td>\n",
       "      <td>104</td>\n",
       "      <td>2.857143</td>\n",
       "      <td>27.523810</td>\n",
       "      <td>85.285714</td>\n",
       "      <td>171.619048</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[649.3660]VTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[655.3735]VTLDYTVQVPGTGR</td>\n",
       "      <td>-1.822067</td>\n",
       "      <td>-1.687501</td>\n",
       "      <td>-1.706146</td>\n",
       "      <td>-1.647231</td>\n",
       "      <td>-1.793966</td>\n",
       "      <td>-1.407199</td>\n",
       "      <td>...</td>\n",
       "      <td>104</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>25</td>\n",
       "      <td>60</td>\n",
       "      <td>2.523810</td>\n",
       "      <td>16.476190</td>\n",
       "      <td>42.238095</td>\n",
       "      <td>79.904762</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDM[649.3660]K</td>\n",
       "      <td>KEQESEVDM[655.3735]K</td>\n",
       "      <td>-1.759694</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.821773</td>\n",
       "      <td>-1.960428</td>\n",
       "      <td>-1.781592</td>\n",
       "      <td>-1.988603</td>\n",
       "      <td>...</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>11</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>7.190476</td>\n",
       "      <td>12.523810</td>\n",
       "      <td>19.428571</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVM[649.3660]LAVDAVIAELK</td>\n",
       "      <td>RGVM[655.3735]LAVDAVIAELK</td>\n",
       "      <td>-1.812349</td>\n",
       "      <td>-1.980371</td>\n",
       "      <td>-2.307386</td>\n",
       "      <td>-2.492858</td>\n",
       "      <td>-1.405837</td>\n",
       "      <td>-2.094407</td>\n",
       "      <td>...</td>\n",
       "      <td>128</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>26</td>\n",
       "      <td>71</td>\n",
       "      <td>2.857143</td>\n",
       "      <td>26.095238</td>\n",
       "      <td>62.571429</td>\n",
       "      <td>115.380952</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>M[649.3660]QLLEIITTDK</td>\n",
       "      <td>M[655.3735]QLLEIITTDK</td>\n",
       "      <td>-2.286954</td>\n",
       "      <td>-2.700322</td>\n",
       "      <td>-1.941018</td>\n",
       "      <td>-1.860306</td>\n",
       "      <td>-2.069488</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>158</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>55</td>\n",
       "      <td>2.571429</td>\n",
       "      <td>25.952381</td>\n",
       "      <td>80.380952</td>\n",
       "      <td>173.428571</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAM[649.3660]R</td>\n",
       "      <td>DHFEEAM[655.3735]R</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.891986</td>\n",
       "      <td>-2.129614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>79</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>46</td>\n",
       "      <td>2.285714</td>\n",
       "      <td>13.952381</td>\n",
       "      <td>39.285714</td>\n",
       "      <td>68.047619</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>203 rows × 67 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Peptide Sequence      Modified Peptide  \\\n",
       "0            AADTIGYPVMIR          AADTIGYPVMIR   \n",
       "1              IAMQTLDMGR            IAMQTLDMGR   \n",
       "2             FVGAVDPIMEK           FVGAVDPIMEK   \n",
       "3    QAQYLGMPINGPFKPDHYRY  QAQYLGMPINGPFKPDHYRY   \n",
       "4            FADVIPMNLPHR          FADVIPMNLPHR   \n",
       "..                    ...                   ...   \n",
       "198     AHMVTLDYTVQVPGTGR     AHMVTLDYTVQVPGTGR   \n",
       "199            KEQESEVDMK            KEQESEVDMK   \n",
       "200       RGVMLAVDAVIAELK       RGVMLAVDAVIAELK   \n",
       "201           MQLLEIITTDK           MQLLEIITTDK   \n",
       "202              DHFEEAMR              DHFEEAMR   \n",
       "\n",
       "             Light Modified Peptide          Heavy Modified Peptide  \\\n",
       "0            AADTIGYPVM[649.3660]IR          AADTIGYPVM[655.3735]IR   \n",
       "1              IAM[649.3660]QTLDMGR            IAM[655.3735]QTLDMGR   \n",
       "2             FVGAVDPIM[649.3660]EK           FVGAVDPIM[655.3735]EK   \n",
       "3    QAQYLGM[649.3660]PINGPFKPDHYRY  QAQYLGM[655.3735]PINGPFKPDHYRY   \n",
       "4            FADVIPM[649.3660]NLPHR          FADVIPM[655.3735]NLPHR   \n",
       "..                              ...                             ...   \n",
       "198     AHM[649.3660]VTLDYTVQVPGTGR     AHM[655.3735]VTLDYTVQVPGTGR   \n",
       "199            KEQESEVDM[649.3660]K            KEQESEVDM[655.3735]K   \n",
       "200       RGVM[649.3660]LAVDAVIAELK       RGVM[655.3735]LAVDAVIAELK   \n",
       "201           M[649.3660]QLLEIITTDK           M[655.3735]QLLEIITTDK   \n",
       "202              DHFEEAM[649.3660]R              DHFEEAM[655.3735]R   \n",
       "\n",
       "     1 Log2 Ratio HL  10 Log2 Ratio HL  11 Log2 Ratio HL  12 Log2 Ratio HL  \\\n",
       "0                NaN               NaN          3.269016               NaN   \n",
       "1                NaN          2.783695          3.114945          2.697822   \n",
       "2                NaN               NaN               NaN               NaN   \n",
       "3           2.394458          2.380664          2.682897               NaN   \n",
       "4                NaN               NaN          2.580647               NaN   \n",
       "..               ...               ...               ...               ...   \n",
       "198        -1.822067         -1.687501         -1.706146         -1.647231   \n",
       "199        -1.759694               NaN         -1.821773         -1.960428   \n",
       "200        -1.812349         -1.980371         -2.307386         -2.492858   \n",
       "201        -2.286954         -2.700322         -1.941018         -1.860306   \n",
       "202              NaN         -2.891986         -2.129614               NaN   \n",
       "\n",
       "     2 Log2 Ratio HL  3 Log2 Ratio HL  ...  nAA_24_180_pae  nAA_12_70_pae  \\\n",
       "0                NaN              NaN  ...             144              5   \n",
       "1                NaN              NaN  ...             173              1   \n",
       "2                NaN         2.383482  ...              52              2   \n",
       "3           2.435014         2.412394  ...              39              5   \n",
       "4           2.446614         2.450718  ...             207              6   \n",
       "..               ...              ...  ...             ...            ...   \n",
       "198        -1.793966        -1.407199  ...             104              2   \n",
       "199        -1.781592        -1.988603  ...              19              0   \n",
       "200        -1.405837        -2.094407  ...             128              2   \n",
       "201        -2.069488              NaN  ...             158              8   \n",
       "202              NaN              NaN  ...              79              6   \n",
       "\n",
       "     nAA_6_180_pae  nAA_12_180_pae  nAA_18_180_pae  nAA_6_180_pae_smooth10  \\\n",
       "0                3              31              85                2.380952   \n",
       "1                2              19              90                2.761905   \n",
       "2                3               9              24                2.238095   \n",
       "3                2              15              24                2.095238   \n",
       "4                3              27             104                2.857143   \n",
       "..             ...             ...             ...                     ...   \n",
       "198              3              25              60                2.523810   \n",
       "199              2               7              11                2.000000   \n",
       "200              3              26              71                2.857143   \n",
       "201              2              15              55                2.571429   \n",
       "202              3              19              46                2.285714   \n",
       "\n",
       "    nAA_12_180_pae_smooth10 nAA_18_180_pae_smooth10 nAA_24_180_pae_smooth10  \\\n",
       "0                 22.190476               59.952381              120.857143   \n",
       "1                 30.571429               90.857143              176.571429   \n",
       "2                 11.238095               29.571429               61.952381   \n",
       "3                 13.380952               27.047619               48.190476   \n",
       "4                 27.523810               85.285714              171.619048   \n",
       "..                      ...                     ...                     ...   \n",
       "198               16.476190               42.238095               79.904762   \n",
       "199                7.190476               12.523810               19.428571   \n",
       "200               26.095238               62.571429              115.380952   \n",
       "201               25.952381               80.380952              173.428571   \n",
       "202               13.952381               39.285714               68.047619   \n",
       "\n",
       "    IDR  \n",
       "0     0  \n",
       "1     0  \n",
       "2     0  \n",
       "3     0  \n",
       "4     0  \n",
       "..   ..  \n",
       "198   0  \n",
       "199   1  \n",
       "200   0  \n",
       "201   0  \n",
       "202   0  \n",
       "\n",
       "[203 rows x 67 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "alphafold_accessibility_smooth[\"position\"] -= 1 # zero-index the positions to match initial dataframe\n",
    "\n",
    "peptides_with_alphafold = peptides_completed_sequence.merge(\n",
    "    alphafold_accessibility_smooth, \n",
    "    how=\"left\", \n",
    "    left_on=[\"Protein ID\", \"Methionine Location\"], \n",
    "    right_on=[\"protein_id\", \"position\"]\n",
    ")\n",
    "peptides_with_alphafold"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "#peptides_with_alphafold.to_csv(os.path.join(datasets_path, \"RvsS_peptides_with_alphafold.csv\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Peptide Sequence</th>\n",
       "      <th>Modified Peptide</th>\n",
       "      <th>Light Modified Peptide</th>\n",
       "      <th>Heavy Modified Peptide</th>\n",
       "      <th>1 Log2 Ratio HL</th>\n",
       "      <th>10 Log2 Ratio HL</th>\n",
       "      <th>11 Log2 Ratio HL</th>\n",
       "      <th>12 Log2 Ratio HL</th>\n",
       "      <th>2 Log2 Ratio HL</th>\n",
       "      <th>3 Log2 Ratio HL</th>\n",
       "      <th>...</th>\n",
       "      <th>nAA_24_180_pae</th>\n",
       "      <th>nAA_12_70_pae</th>\n",
       "      <th>nAA_6_180_pae</th>\n",
       "      <th>nAA_12_180_pae</th>\n",
       "      <th>nAA_18_180_pae</th>\n",
       "      <th>nAA_6_180_pae_smooth10</th>\n",
       "      <th>nAA_12_180_pae_smooth10</th>\n",
       "      <th>nAA_18_180_pae_smooth10</th>\n",
       "      <th>nAA_24_180_pae_smooth10</th>\n",
       "      <th>IDR</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVMIR</td>\n",
       "      <td>AADTIGYPVM[649.3660]IR</td>\n",
       "      <td>AADTIGYPVM[655.3735]IR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.269016</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>144</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>31</td>\n",
       "      <td>85</td>\n",
       "      <td>2.380952</td>\n",
       "      <td>22.190476</td>\n",
       "      <td>59.952381</td>\n",
       "      <td>120.857143</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAMQTLDMGR</td>\n",
       "      <td>IAM[649.3660]QTLDMGR</td>\n",
       "      <td>IAM[655.3735]QTLDMGR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.783695</td>\n",
       "      <td>3.114945</td>\n",
       "      <td>2.697822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>173</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>19</td>\n",
       "      <td>90</td>\n",
       "      <td>2.761905</td>\n",
       "      <td>30.571429</td>\n",
       "      <td>90.857143</td>\n",
       "      <td>176.571429</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIMEK</td>\n",
       "      <td>FVGAVDPIM[649.3660]EK</td>\n",
       "      <td>FVGAVDPIM[655.3735]EK</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.383482</td>\n",
       "      <td>...</td>\n",
       "      <td>52</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>9</td>\n",
       "      <td>24</td>\n",
       "      <td>2.238095</td>\n",
       "      <td>11.238095</td>\n",
       "      <td>29.571429</td>\n",
       "      <td>61.952381</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGMPINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[649.3660]PINGPFKPDHYRY</td>\n",
       "      <td>QAQYLGM[655.3735]PINGPFKPDHYRY</td>\n",
       "      <td>2.394458</td>\n",
       "      <td>2.380664</td>\n",
       "      <td>2.682897</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.435014</td>\n",
       "      <td>2.412394</td>\n",
       "      <td>...</td>\n",
       "      <td>39</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>24</td>\n",
       "      <td>2.095238</td>\n",
       "      <td>13.380952</td>\n",
       "      <td>27.047619</td>\n",
       "      <td>48.190476</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPMNLPHR</td>\n",
       "      <td>FADVIPM[649.3660]NLPHR</td>\n",
       "      <td>FADVIPM[655.3735]NLPHR</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.580647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.446614</td>\n",
       "      <td>2.450718</td>\n",
       "      <td>...</td>\n",
       "      <td>207</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>27</td>\n",
       "      <td>104</td>\n",
       "      <td>2.857143</td>\n",
       "      <td>27.523810</td>\n",
       "      <td>85.285714</td>\n",
       "      <td>171.619048</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>198</th>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHMVTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[649.3660]VTLDYTVQVPGTGR</td>\n",
       "      <td>AHM[655.3735]VTLDYTVQVPGTGR</td>\n",
       "      <td>-1.822067</td>\n",
       "      <td>-1.687501</td>\n",
       "      <td>-1.706146</td>\n",
       "      <td>-1.647231</td>\n",
       "      <td>-1.793966</td>\n",
       "      <td>-1.407199</td>\n",
       "      <td>...</td>\n",
       "      <td>104</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>25</td>\n",
       "      <td>60</td>\n",
       "      <td>2.523810</td>\n",
       "      <td>16.476190</td>\n",
       "      <td>42.238095</td>\n",
       "      <td>79.904762</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDMK</td>\n",
       "      <td>KEQESEVDM[649.3660]K</td>\n",
       "      <td>KEQESEVDM[655.3735]K</td>\n",
       "      <td>-1.759694</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1.821773</td>\n",
       "      <td>-1.960428</td>\n",
       "      <td>-1.781592</td>\n",
       "      <td>-1.988603</td>\n",
       "      <td>...</td>\n",
       "      <td>19</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>7</td>\n",
       "      <td>11</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>7.190476</td>\n",
       "      <td>12.523810</td>\n",
       "      <td>19.428571</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVMLAVDAVIAELK</td>\n",
       "      <td>RGVM[649.3660]LAVDAVIAELK</td>\n",
       "      <td>RGVM[655.3735]LAVDAVIAELK</td>\n",
       "      <td>-1.812349</td>\n",
       "      <td>-1.980371</td>\n",
       "      <td>-2.307386</td>\n",
       "      <td>-2.492858</td>\n",
       "      <td>-1.405837</td>\n",
       "      <td>-2.094407</td>\n",
       "      <td>...</td>\n",
       "      <td>128</td>\n",
       "      <td>2</td>\n",
       "      <td>3</td>\n",
       "      <td>26</td>\n",
       "      <td>71</td>\n",
       "      <td>2.857143</td>\n",
       "      <td>26.095238</td>\n",
       "      <td>62.571429</td>\n",
       "      <td>115.380952</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>MQLLEIITTDK</td>\n",
       "      <td>M[649.3660]QLLEIITTDK</td>\n",
       "      <td>M[655.3735]QLLEIITTDK</td>\n",
       "      <td>-2.286954</td>\n",
       "      <td>-2.700322</td>\n",
       "      <td>-1.941018</td>\n",
       "      <td>-1.860306</td>\n",
       "      <td>-2.069488</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>158</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>15</td>\n",
       "      <td>55</td>\n",
       "      <td>2.571429</td>\n",
       "      <td>25.952381</td>\n",
       "      <td>80.380952</td>\n",
       "      <td>173.428571</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAMR</td>\n",
       "      <td>DHFEEAM[649.3660]R</td>\n",
       "      <td>DHFEEAM[655.3735]R</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-2.891986</td>\n",
       "      <td>-2.129614</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>79</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "      <td>19</td>\n",
       "      <td>46</td>\n",
       "      <td>2.285714</td>\n",
       "      <td>13.952381</td>\n",
       "      <td>39.285714</td>\n",
       "      <td>68.047619</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>203 rows × 67 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Peptide Sequence      Modified Peptide  \\\n",
       "0            AADTIGYPVMIR          AADTIGYPVMIR   \n",
       "1              IAMQTLDMGR            IAMQTLDMGR   \n",
       "2             FVGAVDPIMEK           FVGAVDPIMEK   \n",
       "3    QAQYLGMPINGPFKPDHYRY  QAQYLGMPINGPFKPDHYRY   \n",
       "4            FADVIPMNLPHR          FADVIPMNLPHR   \n",
       "..                    ...                   ...   \n",
       "198     AHMVTLDYTVQVPGTGR     AHMVTLDYTVQVPGTGR   \n",
       "199            KEQESEVDMK            KEQESEVDMK   \n",
       "200       RGVMLAVDAVIAELK       RGVMLAVDAVIAELK   \n",
       "201           MQLLEIITTDK           MQLLEIITTDK   \n",
       "202              DHFEEAMR              DHFEEAMR   \n",
       "\n",
       "             Light Modified Peptide          Heavy Modified Peptide  \\\n",
       "0            AADTIGYPVM[649.3660]IR          AADTIGYPVM[655.3735]IR   \n",
       "1              IAM[649.3660]QTLDMGR            IAM[655.3735]QTLDMGR   \n",
       "2             FVGAVDPIM[649.3660]EK           FVGAVDPIM[655.3735]EK   \n",
       "3    QAQYLGM[649.3660]PINGPFKPDHYRY  QAQYLGM[655.3735]PINGPFKPDHYRY   \n",
       "4            FADVIPM[649.3660]NLPHR          FADVIPM[655.3735]NLPHR   \n",
       "..                              ...                             ...   \n",
       "198     AHM[649.3660]VTLDYTVQVPGTGR     AHM[655.3735]VTLDYTVQVPGTGR   \n",
       "199            KEQESEVDM[649.3660]K            KEQESEVDM[655.3735]K   \n",
       "200       RGVM[649.3660]LAVDAVIAELK       RGVM[655.3735]LAVDAVIAELK   \n",
       "201           M[649.3660]QLLEIITTDK           M[655.3735]QLLEIITTDK   \n",
       "202              DHFEEAM[649.3660]R              DHFEEAM[655.3735]R   \n",
       "\n",
       "     1 Log2 Ratio HL  10 Log2 Ratio HL  11 Log2 Ratio HL  12 Log2 Ratio HL  \\\n",
       "0                NaN               NaN          3.269016               NaN   \n",
       "1                NaN          2.783695          3.114945          2.697822   \n",
       "2                NaN               NaN               NaN               NaN   \n",
       "3           2.394458          2.380664          2.682897               NaN   \n",
       "4                NaN               NaN          2.580647               NaN   \n",
       "..               ...               ...               ...               ...   \n",
       "198        -1.822067         -1.687501         -1.706146         -1.647231   \n",
       "199        -1.759694               NaN         -1.821773         -1.960428   \n",
       "200        -1.812349         -1.980371         -2.307386         -2.492858   \n",
       "201        -2.286954         -2.700322         -1.941018         -1.860306   \n",
       "202              NaN         -2.891986         -2.129614               NaN   \n",
       "\n",
       "     2 Log2 Ratio HL  3 Log2 Ratio HL  ...  nAA_24_180_pae  nAA_12_70_pae  \\\n",
       "0                NaN              NaN  ...             144              5   \n",
       "1                NaN              NaN  ...             173              1   \n",
       "2                NaN         2.383482  ...              52              2   \n",
       "3           2.435014         2.412394  ...              39              5   \n",
       "4           2.446614         2.450718  ...             207              6   \n",
       "..               ...              ...  ...             ...            ...   \n",
       "198        -1.793966        -1.407199  ...             104              2   \n",
       "199        -1.781592        -1.988603  ...              19              0   \n",
       "200        -1.405837        -2.094407  ...             128              2   \n",
       "201        -2.069488              NaN  ...             158              8   \n",
       "202              NaN              NaN  ...              79              6   \n",
       "\n",
       "     nAA_6_180_pae  nAA_12_180_pae  nAA_18_180_pae  nAA_6_180_pae_smooth10  \\\n",
       "0                3              31              85                2.380952   \n",
       "1                2              19              90                2.761905   \n",
       "2                3               9              24                2.238095   \n",
       "3                2              15              24                2.095238   \n",
       "4                3              27             104                2.857143   \n",
       "..             ...             ...             ...                     ...   \n",
       "198              3              25              60                2.523810   \n",
       "199              2               7              11                2.000000   \n",
       "200              3              26              71                2.857143   \n",
       "201              2              15              55                2.571429   \n",
       "202              3              19              46                2.285714   \n",
       "\n",
       "    nAA_12_180_pae_smooth10 nAA_18_180_pae_smooth10 nAA_24_180_pae_smooth10  \\\n",
       "0                 22.190476               59.952381              120.857143   \n",
       "1                 30.571429               90.857143              176.571429   \n",
       "2                 11.238095               29.571429               61.952381   \n",
       "3                 13.380952               27.047619               48.190476   \n",
       "4                 27.523810               85.285714              171.619048   \n",
       "..                      ...                     ...                     ...   \n",
       "198               16.476190               42.238095               79.904762   \n",
       "199                7.190476               12.523810               19.428571   \n",
       "200               26.095238               62.571429              115.380952   \n",
       "201               25.952381               80.380952              173.428571   \n",
       "202               13.952381               39.285714               68.047619   \n",
       "\n",
       "    IDR  \n",
       "0     0  \n",
       "1     0  \n",
       "2     0  \n",
       "3     0  \n",
       "4     0  \n",
       "..   ..  \n",
       "198   0  \n",
       "199   1  \n",
       "200   0  \n",
       "201   0  \n",
       "202   0  \n",
       "\n",
       "[203 rows x 67 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "path = os.path.join(datasets_path, \"RvsS_peptides_with_alphafold.csv\")\n",
    "peptides_with_alphafold = pd.read_csv(path)\n",
    "peptides_with_alphafold.set_index(\"Unnamed: 0\", inplace=True)\n",
    "peptides_with_alphafold.index.name = None\n",
    "peptides_with_alphafold"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load Dataset (MitoCarta3.0) - Full Mitochondrial Proteome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MouseGeneID</th>\n",
       "      <th>HumanOrthologGeneID</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>Synonyms</th>\n",
       "      <th>Description</th>\n",
       "      <th>MitoCarta3.0_List</th>\n",
       "      <th>MitoCarta3.0_Evidence</th>\n",
       "      <th>MitoCarta3.0_SubMitoLocalization</th>\n",
       "      <th>MitoCarta3.0_MitoPathways</th>\n",
       "      <th>TrainingDataset</th>\n",
       "      <th>...</th>\n",
       "      <th>liver_total_peak_intensity_log10</th>\n",
       "      <th>heart_total_peak_intensity_log10</th>\n",
       "      <th>skeletalmuscle_total_peak_intensity_log10</th>\n",
       "      <th>adipose_total_peak_intensity_log10</th>\n",
       "      <th>smallintestine_total_peak_intensity_log10</th>\n",
       "      <th>largeintestine_total_peak_intensity_log10</th>\n",
       "      <th>stomach_total_peak_intensity_log10</th>\n",
       "      <th>placenta_total_peak_intensity_log10</th>\n",
       "      <th>testis_total_peak_intensity_log10</th>\n",
       "      <th>HPA_Main_Location_2020 (Reliability)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>66445</td>\n",
       "      <td>1537.0</td>\n",
       "      <td>Cyc1</td>\n",
       "      <td>2610002H19Rik|AA408921|Cyct1</td>\n",
       "      <td>cytochrome c-1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, APEX_IMS, APEX_matrix, targetP sig...</td>\n",
       "      <td>MIM</td>\n",
       "      <td>OXPHOS &gt; Complex III &gt; CIII subunits | Metabol...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.4</td>\n",
       "      <td>9.8</td>\n",
       "      <td>10.1</td>\n",
       "      <td>10.2</td>\n",
       "      <td>10.2</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>9.8</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>18597</td>\n",
       "      <td>5160.0</td>\n",
       "      <td>Pdha1</td>\n",
       "      <td>Pdha|Pdha-1</td>\n",
       "      <td>pyruvate dehydrogenase E1 alpha 1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, APEX_matrix, targetP signal+, yeas...</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Metabolism &gt; Carbohydrate metabolism &gt; Pyruvat...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>9.6</td>\n",
       "      <td>10.5</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.1</td>\n",
       "      <td>9.9</td>\n",
       "      <td>9.6</td>\n",
       "      <td>10.0</td>\n",
       "      <td>9.9</td>\n",
       "      <td>8.9</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>66043</td>\n",
       "      <td>513.0</td>\n",
       "      <td>Atp5d</td>\n",
       "      <td>0610008F14Rik|1500000I11Rik|AA960090|AI876556|...</td>\n",
       "      <td>ATP synthase, H+ transporting, mitochondrial F...</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, APEX_matrix, targetP signal, yeast...</td>\n",
       "      <td>MIM</td>\n",
       "      <td>OXPHOS &gt; Complex V &gt; CV subunits | OXPHOS &gt; OX...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>9.8</td>\n",
       "      <td>9.8</td>\n",
       "      <td>9.5</td>\n",
       "      <td>9.8</td>\n",
       "      <td>10.1</td>\n",
       "      <td>10.0</td>\n",
       "      <td>9.8</td>\n",
       "      <td>9.8</td>\n",
       "      <td>9.6</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>74316</td>\n",
       "      <td>122961.0</td>\n",
       "      <td>Isca2</td>\n",
       "      <td>0710001C05Rik|5730594E03Rik|Hbld|Hbld1</td>\n",
       "      <td>iron-sulfur cluster assembly 2</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>APEX_matrix, targetP signal+, yeast mito homol...</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Metabolism &gt; Metals and cofactors &gt; Fe-S clust...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>8.2</td>\n",
       "      <td>7.9</td>\n",
       "      <td>7.2</td>\n",
       "      <td>8.7</td>\n",
       "      <td>8.5</td>\n",
       "      <td>8.4</td>\n",
       "      <td>8.0</td>\n",
       "      <td>8.5</td>\n",
       "      <td>8.5</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>68263</td>\n",
       "      <td>5162.0</td>\n",
       "      <td>Pdhb</td>\n",
       "      <td>2610103L06Rik|AL024199|C81408</td>\n",
       "      <td>pyruvate dehydrogenase (lipoamide) beta</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, targetP signal+, yeast mito homolo...</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Metabolism &gt; Carbohydrate metabolism &gt; Pyruvat...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>9.6</td>\n",
       "      <td>10.5</td>\n",
       "      <td>9.8</td>\n",
       "      <td>10.4</td>\n",
       "      <td>9.8</td>\n",
       "      <td>9.9</td>\n",
       "      <td>9.9</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.5</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1135</th>\n",
       "      <td>71703</td>\n",
       "      <td>51566.0</td>\n",
       "      <td>Armcx3</td>\n",
       "      <td>1200004E24Rik|AI450003|ALEX3</td>\n",
       "      <td>armadillo repeat containing, X-linked 3</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>MOM</td>\n",
       "      <td>Mitochondrial dynamics and surveillance &gt; Traf...</td>\n",
       "      <td>Tpossible_mito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Nucleoplasm (Approved)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1136</th>\n",
       "      <td>67474</td>\n",
       "      <td>9342.0</td>\n",
       "      <td>Snap29</td>\n",
       "      <td>1300018G05Rik|AI891940|AU020222|BB131856|Gs32</td>\n",
       "      <td>synaptosomal-associated protein 29</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>APEX_IMS</td>\n",
       "      <td>IMS</td>\n",
       "      <td>Mitochondrial dynamics and surveillance &gt; Auto...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Cytosol (Supported)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1137</th>\n",
       "      <td>18970</td>\n",
       "      <td>5423.0</td>\n",
       "      <td>Polb</td>\n",
       "      <td>A430088C08Rik</td>\n",
       "      <td>polymerase (DNA directed), beta</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Mitochondrial central dogma &gt; mtDNA maintenanc...</td>\n",
       "      <td>Tnon_mito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Vesicles (Uncertain)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1138</th>\n",
       "      <td>109729085</td>\n",
       "      <td>109703458.0</td>\n",
       "      <td>Htd2</td>\n",
       "      <td>-</td>\n",
       "      <td>hydroxyacyl-thioester dehydratase type 2</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Metabolism &gt; Lipid metabolism &gt; Type II fatty ...</td>\n",
       "      <td>NA - newly added to NCBI Entrez Gene</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1139</th>\n",
       "      <td>66364</td>\n",
       "      <td>101928527.0</td>\n",
       "      <td>Pigbos1</td>\n",
       "      <td>2310009A05Rik|5031420N21Rik</td>\n",
       "      <td>Pigb opposite strand 1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>MOM</td>\n",
       "      <td>Mitochondrial dynamics and surveillance &gt; Orga...</td>\n",
       "      <td>NA - newly added to NCBI Entrez Gene</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1140 rows × 48 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      MouseGeneID  HumanOrthologGeneID   Symbol  \\\n",
       "0           66445               1537.0     Cyc1   \n",
       "1           18597               5160.0    Pdha1   \n",
       "2           66043                513.0    Atp5d   \n",
       "3           74316             122961.0    Isca2   \n",
       "4           68263               5162.0     Pdhb   \n",
       "...           ...                  ...      ...   \n",
       "1135        71703              51566.0   Armcx3   \n",
       "1136        67474               9342.0   Snap29   \n",
       "1137        18970               5423.0     Polb   \n",
       "1138    109729085          109703458.0     Htd2   \n",
       "1139        66364          101928527.0  Pigbos1   \n",
       "\n",
       "                                               Synonyms  \\\n",
       "0                          2610002H19Rik|AA408921|Cyct1   \n",
       "1                                           Pdha|Pdha-1   \n",
       "2     0610008F14Rik|1500000I11Rik|AA960090|AI876556|...   \n",
       "3                0710001C05Rik|5730594E03Rik|Hbld|Hbld1   \n",
       "4                         2610103L06Rik|AL024199|C81408   \n",
       "...                                                 ...   \n",
       "1135                       1200004E24Rik|AI450003|ALEX3   \n",
       "1136      1300018G05Rik|AI891940|AU020222|BB131856|Gs32   \n",
       "1137                                      A430088C08Rik   \n",
       "1138                                                  -   \n",
       "1139                        2310009A05Rik|5031420N21Rik   \n",
       "\n",
       "                                            Description MitoCarta3.0_List  \\\n",
       "0                                        cytochrome c-1      MitoCarta3.0   \n",
       "1                     pyruvate dehydrogenase E1 alpha 1      MitoCarta3.0   \n",
       "2     ATP synthase, H+ transporting, mitochondrial F...      MitoCarta3.0   \n",
       "3                        iron-sulfur cluster assembly 2      MitoCarta3.0   \n",
       "4               pyruvate dehydrogenase (lipoamide) beta      MitoCarta3.0   \n",
       "...                                                 ...               ...   \n",
       "1135            armadillo repeat containing, X-linked 3      MitoCarta3.0   \n",
       "1136                 synaptosomal-associated protein 29      MitoCarta3.0   \n",
       "1137                    polymerase (DNA directed), beta      MitoCarta3.0   \n",
       "1138           hydroxyacyl-thioester dehydratase type 2      MitoCarta3.0   \n",
       "1139                             Pigb opposite strand 1      MitoCarta3.0   \n",
       "\n",
       "                                  MitoCarta3.0_Evidence  \\\n",
       "0     literature, APEX_IMS, APEX_matrix, targetP sig...   \n",
       "1     literature, APEX_matrix, targetP signal+, yeas...   \n",
       "2     literature, APEX_matrix, targetP signal, yeast...   \n",
       "3     APEX_matrix, targetP signal+, yeast mito homol...   \n",
       "4     literature, targetP signal+, yeast mito homolo...   \n",
       "...                                                 ...   \n",
       "1135                                         literature   \n",
       "1136                                           APEX_IMS   \n",
       "1137                                         literature   \n",
       "1138                                         literature   \n",
       "1139                                         literature   \n",
       "\n",
       "     MitoCarta3.0_SubMitoLocalization  \\\n",
       "0                                 MIM   \n",
       "1                              Matrix   \n",
       "2                                 MIM   \n",
       "3                              Matrix   \n",
       "4                              Matrix   \n",
       "...                               ...   \n",
       "1135                              MOM   \n",
       "1136                              IMS   \n",
       "1137                           Matrix   \n",
       "1138                           Matrix   \n",
       "1139                              MOM   \n",
       "\n",
       "                              MitoCarta3.0_MitoPathways  \\\n",
       "0     OXPHOS > Complex III > CIII subunits | Metabol...   \n",
       "1     Metabolism > Carbohydrate metabolism > Pyruvat...   \n",
       "2     OXPHOS > Complex V > CV subunits | OXPHOS > OX...   \n",
       "3     Metabolism > Metals and cofactors > Fe-S clust...   \n",
       "4     Metabolism > Carbohydrate metabolism > Pyruvat...   \n",
       "...                                                 ...   \n",
       "1135  Mitochondrial dynamics and surveillance > Traf...   \n",
       "1136  Mitochondrial dynamics and surveillance > Auto...   \n",
       "1137  Mitochondrial central dogma > mtDNA maintenanc...   \n",
       "1138  Metabolism > Lipid metabolism > Type II fatty ...   \n",
       "1139  Mitochondrial dynamics and surveillance > Orga...   \n",
       "\n",
       "                           TrainingDataset  ...  \\\n",
       "0                                    Tmito  ...   \n",
       "1                                    Tmito  ...   \n",
       "2                                    Tmito  ...   \n",
       "3                                    Tmito  ...   \n",
       "4                                    Tmito  ...   \n",
       "...                                    ...  ...   \n",
       "1135                        Tpossible_mito  ...   \n",
       "1136                                 Tmito  ...   \n",
       "1137                             Tnon_mito  ...   \n",
       "1138  NA - newly added to NCBI Entrez Gene  ...   \n",
       "1139  NA - newly added to NCBI Entrez Gene  ...   \n",
       "\n",
       "      liver_total_peak_intensity_log10  heart_total_peak_intensity_log10  \\\n",
       "0                                 10.0                              10.4   \n",
       "1                                  9.6                              10.5   \n",
       "2                                  9.8                               9.8   \n",
       "3                                  8.2                               7.9   \n",
       "4                                  9.6                              10.5   \n",
       "...                                ...                               ...   \n",
       "1135                               NaN                               NaN   \n",
       "1136                               NaN                               NaN   \n",
       "1137                               NaN                               NaN   \n",
       "1138                               NaN                               NaN   \n",
       "1139                               NaN                               NaN   \n",
       "\n",
       "      skeletalmuscle_total_peak_intensity_log10  \\\n",
       "0                                           9.8   \n",
       "1                                          10.0   \n",
       "2                                           9.5   \n",
       "3                                           7.2   \n",
       "4                                           9.8   \n",
       "...                                         ...   \n",
       "1135                                        NaN   \n",
       "1136                                        NaN   \n",
       "1137                                        NaN   \n",
       "1138                                        NaN   \n",
       "1139                                        NaN   \n",
       "\n",
       "     adipose_total_peak_intensity_log10  \\\n",
       "0                                  10.1   \n",
       "1                                  10.1   \n",
       "2                                   9.8   \n",
       "3                                   8.7   \n",
       "4                                  10.4   \n",
       "...                                 ...   \n",
       "1135                                NaN   \n",
       "1136                                NaN   \n",
       "1137                                NaN   \n",
       "1138                                NaN   \n",
       "1139                                NaN   \n",
       "\n",
       "     smallintestine_total_peak_intensity_log10  \\\n",
       "0                                         10.2   \n",
       "1                                          9.9   \n",
       "2                                         10.1   \n",
       "3                                          8.5   \n",
       "4                                          9.8   \n",
       "...                                        ...   \n",
       "1135                                       NaN   \n",
       "1136                                       NaN   \n",
       "1137                                       NaN   \n",
       "1138                                       NaN   \n",
       "1139                                       NaN   \n",
       "\n",
       "     largeintestine_total_peak_intensity_log10  \\\n",
       "0                                         10.2   \n",
       "1                                          9.6   \n",
       "2                                         10.0   \n",
       "3                                          8.4   \n",
       "4                                          9.9   \n",
       "...                                        ...   \n",
       "1135                                       NaN   \n",
       "1136                                       NaN   \n",
       "1137                                       NaN   \n",
       "1138                                       NaN   \n",
       "1139                                       NaN   \n",
       "\n",
       "      stomach_total_peak_intensity_log10  placenta_total_peak_intensity_log10  \\\n",
       "0                                   10.0                                 10.0   \n",
       "1                                   10.0                                  9.9   \n",
       "2                                    9.8                                  9.8   \n",
       "3                                    8.0                                  8.5   \n",
       "4                                    9.9                                 10.0   \n",
       "...                                  ...                                  ...   \n",
       "1135                                 NaN                                  NaN   \n",
       "1136                                 NaN                                  NaN   \n",
       "1137                                 NaN                                  NaN   \n",
       "1138                                 NaN                                  NaN   \n",
       "1139                                 NaN                                  NaN   \n",
       "\n",
       "     testis_total_peak_intensity_log10  HPA_Main_Location_2020 (Reliability)  \n",
       "0                                  9.8              Mitochondria (Supported)  \n",
       "1                                  8.9              Mitochondria (Supported)  \n",
       "2                                  9.6                                   NaN  \n",
       "3                                  8.5                                   NaN  \n",
       "4                                 10.5              Mitochondria (Supported)  \n",
       "...                                ...                                   ...  \n",
       "1135                               NaN                Nucleoplasm (Approved)  \n",
       "1136                               NaN                   Cytosol (Supported)  \n",
       "1137                               NaN                  Vesicles (Uncertain)  \n",
       "1138                               NaN              Mitochondria (Supported)  \n",
       "1139                               NaN                                   NaN  \n",
       "\n",
       "[1140 rows x 48 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "data_loc = os.path.join(datasets_path, \"Mouse.MitoCarta3.0.xls\")\n",
    "mitocarta3_0 = pd.read_excel(data_loc, sheet_name=\"A Mouse MitoCarta3.0\")\n",
    "mitocarta3_0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "HPA_Main_Location_2020 (Reliability)\n",
       "False    609\n",
       "True     531\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# calculate number of proteins in the mitochondrial proteome\n",
    "is_mitochondrial = (mitocarta3_0[\"HPA_Main_Location_2020 (Reliability)\"].str.contains(\"mitoch\", case=False))\n",
    "(is_mitochondrial == True).value_counts(dropna=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "HPA_Main_Location_2020 (Reliability)\n",
       "Mitochondria (Supported)                               266\n",
       "Mitochondria (Enhanced)                                108\n",
       "Mitochondria (Approved)                                 87\n",
       "Mitochondria;Nucleoplasm (Supported)                    17\n",
       "Mitochondria;Nucleoplasm (Approved)                     16\n",
       "Cytosol;Mitochondria (Approved)                          6\n",
       "Cytosol;Mitochondria (Supported)                         5\n",
       "Mitochondria;Nucleoli (Approved)                         3\n",
       "Mitochondria (Uncertain)                                 3\n",
       "Mitochondria;Nuclear bodies (Supported)                  2\n",
       "Mitochondria;Plasma membrane (Supported)                 2\n",
       "Cytosol;Mitochondria (Enhanced)                          2\n",
       "Mitochondria;Nuclear bodies (Approved)                   1\n",
       "Mitochondria;Nucleoplasm;Vesicles (Approved)             1\n",
       "Mitochondria;Nucleoli fibrillar center (Approved)        1\n",
       "Cytosol;Mitochondria;Nucleoplasm (Approved)              1\n",
       "Mitochondria;Peroxisomes (Enhanced)                      1\n",
       "Cytosol;Mitochondria;Nucleoplasm (Supported)             1\n",
       "Golgi apparatus;Mitochondria;Nucleoplasm (Approved)      1\n",
       "Mitochondria;Nucleoplasm (Uncertain)                     1\n",
       "Mitochondria;Vesicles (Supported)                        1\n",
       "Mitochondria;Nucleoli (Supported)                        1\n",
       "Mitochondria;Vesicles (Approved)                         1\n",
       "Mitochondria;Nuclear membrane (Supported)                1\n",
       "Mitochondria;Nucleoli;Nucleoplasm (Supported)            1\n",
       "Intermediate filaments;Mitochondria (Approved)           1\n",
       "Name: count, dtype: int64"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# ensure protein split was done correctly (correct mitoch... string matching)\n",
    "pd.set_option('display.max_rows', None)\n",
    "display(mitocarta3_0[is_mitochondrial == True][\"HPA_Main_Location_2020 (Reliability)\"].value_counts())\n",
    "pd.reset_option('display.max_rows')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MouseGeneID</th>\n",
       "      <th>HumanOrthologGeneID</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>Synonyms</th>\n",
       "      <th>Description</th>\n",
       "      <th>MitoCarta3.0_List</th>\n",
       "      <th>MitoCarta3.0_Evidence</th>\n",
       "      <th>MitoCarta3.0_SubMitoLocalization</th>\n",
       "      <th>MitoCarta3.0_MitoPathways</th>\n",
       "      <th>TrainingDataset</th>\n",
       "      <th>...</th>\n",
       "      <th>liver_total_peak_intensity_log10</th>\n",
       "      <th>heart_total_peak_intensity_log10</th>\n",
       "      <th>skeletalmuscle_total_peak_intensity_log10</th>\n",
       "      <th>adipose_total_peak_intensity_log10</th>\n",
       "      <th>smallintestine_total_peak_intensity_log10</th>\n",
       "      <th>largeintestine_total_peak_intensity_log10</th>\n",
       "      <th>stomach_total_peak_intensity_log10</th>\n",
       "      <th>placenta_total_peak_intensity_log10</th>\n",
       "      <th>testis_total_peak_intensity_log10</th>\n",
       "      <th>HPA_Main_Location_2020 (Reliability)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>66445</td>\n",
       "      <td>1537.0</td>\n",
       "      <td>Cyc1</td>\n",
       "      <td>2610002H19Rik|AA408921|Cyct1</td>\n",
       "      <td>cytochrome c-1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, APEX_IMS, APEX_matrix, targetP sig...</td>\n",
       "      <td>MIM</td>\n",
       "      <td>OXPHOS &gt; Complex III &gt; CIII subunits | Metabol...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.4</td>\n",
       "      <td>9.8</td>\n",
       "      <td>10.1</td>\n",
       "      <td>10.2</td>\n",
       "      <td>10.2</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>9.8</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>18597</td>\n",
       "      <td>5160.0</td>\n",
       "      <td>Pdha1</td>\n",
       "      <td>Pdha|Pdha-1</td>\n",
       "      <td>pyruvate dehydrogenase E1 alpha 1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, APEX_matrix, targetP signal+, yeas...</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Metabolism &gt; Carbohydrate metabolism &gt; Pyruvat...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>9.6</td>\n",
       "      <td>10.5</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.1</td>\n",
       "      <td>9.9</td>\n",
       "      <td>9.6</td>\n",
       "      <td>10.0</td>\n",
       "      <td>9.9</td>\n",
       "      <td>8.9</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>68263</td>\n",
       "      <td>5162.0</td>\n",
       "      <td>Pdhb</td>\n",
       "      <td>2610103L06Rik|AL024199|C81408</td>\n",
       "      <td>pyruvate dehydrogenase (lipoamide) beta</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, targetP signal+, yeast mito homolo...</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Metabolism &gt; Carbohydrate metabolism &gt; Pyruvat...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>9.6</td>\n",
       "      <td>10.5</td>\n",
       "      <td>9.8</td>\n",
       "      <td>10.4</td>\n",
       "      <td>9.8</td>\n",
       "      <td>9.9</td>\n",
       "      <td>9.9</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.5</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>22273</td>\n",
       "      <td>7384.0</td>\n",
       "      <td>Uqcrc1</td>\n",
       "      <td>1110032G10Rik</td>\n",
       "      <td>ubiquinol-cytochrome c reductase core protein 1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, APEX_matrix, targetP signal+, yeas...</td>\n",
       "      <td>MIM</td>\n",
       "      <td>Protein import, sorting and homeostasis &gt; Prot...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.8</td>\n",
       "      <td>10.3</td>\n",
       "      <td>10.1</td>\n",
       "      <td>10.6</td>\n",
       "      <td>10.2</td>\n",
       "      <td>10.3</td>\n",
       "      <td>10.2</td>\n",
       "      <td>9.9</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>56282</td>\n",
       "      <td>6182.0</td>\n",
       "      <td>Mrpl12</td>\n",
       "      <td>0610034O11Rik|1500031N16Rik|L12mt|MRP-|MRP-L12...</td>\n",
       "      <td>mitochondrial ribosomal protein L12</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, targetP signal+, yeast mito homolo...</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Mitochondrial central dogma &gt; mtRNA metabolism...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>9.3</td>\n",
       "      <td>8.9</td>\n",
       "      <td>8.6</td>\n",
       "      <td>9.3</td>\n",
       "      <td>9.2</td>\n",
       "      <td>9.8</td>\n",
       "      <td>9.0</td>\n",
       "      <td>9.4</td>\n",
       "      <td>9.0</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1129</th>\n",
       "      <td>78248</td>\n",
       "      <td>51309.0</td>\n",
       "      <td>Armcx1</td>\n",
       "      <td>3010033I09Rik|ALEX1</td>\n",
       "      <td>armadillo repeat containing, X-linked 1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>MOM</td>\n",
       "      <td>Mitochondrial dynamics and surveillance &gt; Traf...</td>\n",
       "      <td>Tnon_mito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mitochondria;Nucleoplasm (Approved)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1130</th>\n",
       "      <td>67416</td>\n",
       "      <td>9823.0</td>\n",
       "      <td>Armcx2</td>\n",
       "      <td>3230401N03Rik|AI043003|ALEX2</td>\n",
       "      <td>armadillo repeat containing, X-linked 2</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>Membrane</td>\n",
       "      <td>0</td>\n",
       "      <td>Tnon_mito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mitochondria (Approved)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1132</th>\n",
       "      <td>269642</td>\n",
       "      <td>339983.0</td>\n",
       "      <td>Nat8l</td>\n",
       "      <td>1110038O08Rik|Sh|Shati</td>\n",
       "      <td>N-acetyltransferase 8-like</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>Membrane</td>\n",
       "      <td>Metabolism &gt; Amino acid metabolism &gt; Glutamate...</td>\n",
       "      <td>Tpossible_mito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mitochondria (Approved)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1134</th>\n",
       "      <td>76781</td>\n",
       "      <td>64863.0</td>\n",
       "      <td>Mettl4</td>\n",
       "      <td>2410198H06Rik|A730091E08Rik|AV296509|HsT66|HsT661</td>\n",
       "      <td>methyltransferase like 4</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Mitochondrial central dogma &gt; mtDNA maintenanc...</td>\n",
       "      <td>Tnon_mito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mitochondria (Approved)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1138</th>\n",
       "      <td>109729085</td>\n",
       "      <td>109703458.0</td>\n",
       "      <td>Htd2</td>\n",
       "      <td>-</td>\n",
       "      <td>hydroxyacyl-thioester dehydratase type 2</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Metabolism &gt; Lipid metabolism &gt; Type II fatty ...</td>\n",
       "      <td>NA - newly added to NCBI Entrez Gene</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>531 rows × 48 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      MouseGeneID  HumanOrthologGeneID  Symbol  \\\n",
       "0           66445               1537.0    Cyc1   \n",
       "1           18597               5160.0   Pdha1   \n",
       "4           68263               5162.0    Pdhb   \n",
       "5           22273               7384.0  Uqcrc1   \n",
       "7           56282               6182.0  Mrpl12   \n",
       "...           ...                  ...     ...   \n",
       "1129        78248              51309.0  Armcx1   \n",
       "1130        67416               9823.0  Armcx2   \n",
       "1132       269642             339983.0   Nat8l   \n",
       "1134        76781              64863.0  Mettl4   \n",
       "1138    109729085          109703458.0    Htd2   \n",
       "\n",
       "                                               Synonyms  \\\n",
       "0                          2610002H19Rik|AA408921|Cyct1   \n",
       "1                                           Pdha|Pdha-1   \n",
       "4                         2610103L06Rik|AL024199|C81408   \n",
       "5                                         1110032G10Rik   \n",
       "7     0610034O11Rik|1500031N16Rik|L12mt|MRP-|MRP-L12...   \n",
       "...                                                 ...   \n",
       "1129                                3010033I09Rik|ALEX1   \n",
       "1130                       3230401N03Rik|AI043003|ALEX2   \n",
       "1132                             1110038O08Rik|Sh|Shati   \n",
       "1134  2410198H06Rik|A730091E08Rik|AV296509|HsT66|HsT661   \n",
       "1138                                                  -   \n",
       "\n",
       "                                          Description MitoCarta3.0_List  \\\n",
       "0                                      cytochrome c-1      MitoCarta3.0   \n",
       "1                   pyruvate dehydrogenase E1 alpha 1      MitoCarta3.0   \n",
       "4             pyruvate dehydrogenase (lipoamide) beta      MitoCarta3.0   \n",
       "5     ubiquinol-cytochrome c reductase core protein 1      MitoCarta3.0   \n",
       "7                 mitochondrial ribosomal protein L12      MitoCarta3.0   \n",
       "...                                               ...               ...   \n",
       "1129          armadillo repeat containing, X-linked 1      MitoCarta3.0   \n",
       "1130          armadillo repeat containing, X-linked 2      MitoCarta3.0   \n",
       "1132                       N-acetyltransferase 8-like      MitoCarta3.0   \n",
       "1134                         methyltransferase like 4      MitoCarta3.0   \n",
       "1138         hydroxyacyl-thioester dehydratase type 2      MitoCarta3.0   \n",
       "\n",
       "                                  MitoCarta3.0_Evidence  \\\n",
       "0     literature, APEX_IMS, APEX_matrix, targetP sig...   \n",
       "1     literature, APEX_matrix, targetP signal+, yeas...   \n",
       "4     literature, targetP signal+, yeast mito homolo...   \n",
       "5     literature, APEX_matrix, targetP signal+, yeas...   \n",
       "7     literature, targetP signal+, yeast mito homolo...   \n",
       "...                                                 ...   \n",
       "1129                                         literature   \n",
       "1130                                         literature   \n",
       "1132                                         literature   \n",
       "1134                                         literature   \n",
       "1138                                         literature   \n",
       "\n",
       "     MitoCarta3.0_SubMitoLocalization  \\\n",
       "0                                 MIM   \n",
       "1                              Matrix   \n",
       "4                              Matrix   \n",
       "5                                 MIM   \n",
       "7                              Matrix   \n",
       "...                               ...   \n",
       "1129                              MOM   \n",
       "1130                         Membrane   \n",
       "1132                         Membrane   \n",
       "1134                           Matrix   \n",
       "1138                           Matrix   \n",
       "\n",
       "                              MitoCarta3.0_MitoPathways  \\\n",
       "0     OXPHOS > Complex III > CIII subunits | Metabol...   \n",
       "1     Metabolism > Carbohydrate metabolism > Pyruvat...   \n",
       "4     Metabolism > Carbohydrate metabolism > Pyruvat...   \n",
       "5     Protein import, sorting and homeostasis > Prot...   \n",
       "7     Mitochondrial central dogma > mtRNA metabolism...   \n",
       "...                                                 ...   \n",
       "1129  Mitochondrial dynamics and surveillance > Traf...   \n",
       "1130                                                  0   \n",
       "1132  Metabolism > Amino acid metabolism > Glutamate...   \n",
       "1134  Mitochondrial central dogma > mtDNA maintenanc...   \n",
       "1138  Metabolism > Lipid metabolism > Type II fatty ...   \n",
       "\n",
       "                           TrainingDataset  ...  \\\n",
       "0                                    Tmito  ...   \n",
       "1                                    Tmito  ...   \n",
       "4                                    Tmito  ...   \n",
       "5                                    Tmito  ...   \n",
       "7                                    Tmito  ...   \n",
       "...                                    ...  ...   \n",
       "1129                             Tnon_mito  ...   \n",
       "1130                             Tnon_mito  ...   \n",
       "1132                        Tpossible_mito  ...   \n",
       "1134                             Tnon_mito  ...   \n",
       "1138  NA - newly added to NCBI Entrez Gene  ...   \n",
       "\n",
       "      liver_total_peak_intensity_log10  heart_total_peak_intensity_log10  \\\n",
       "0                                 10.0                              10.4   \n",
       "1                                  9.6                              10.5   \n",
       "4                                  9.6                              10.5   \n",
       "5                                 10.0                              10.8   \n",
       "7                                  9.3                               8.9   \n",
       "...                                ...                               ...   \n",
       "1129                               NaN                               NaN   \n",
       "1130                               NaN                               NaN   \n",
       "1132                               NaN                               NaN   \n",
       "1134                               NaN                               NaN   \n",
       "1138                               NaN                               NaN   \n",
       "\n",
       "      skeletalmuscle_total_peak_intensity_log10  \\\n",
       "0                                           9.8   \n",
       "1                                          10.0   \n",
       "4                                           9.8   \n",
       "5                                          10.3   \n",
       "7                                           8.6   \n",
       "...                                         ...   \n",
       "1129                                        NaN   \n",
       "1130                                        NaN   \n",
       "1132                                        NaN   \n",
       "1134                                        NaN   \n",
       "1138                                        NaN   \n",
       "\n",
       "     adipose_total_peak_intensity_log10  \\\n",
       "0                                  10.1   \n",
       "1                                  10.1   \n",
       "4                                  10.4   \n",
       "5                                  10.1   \n",
       "7                                   9.3   \n",
       "...                                 ...   \n",
       "1129                                NaN   \n",
       "1130                                NaN   \n",
       "1132                                NaN   \n",
       "1134                                NaN   \n",
       "1138                                NaN   \n",
       "\n",
       "     smallintestine_total_peak_intensity_log10  \\\n",
       "0                                         10.2   \n",
       "1                                          9.9   \n",
       "4                                          9.8   \n",
       "5                                         10.6   \n",
       "7                                          9.2   \n",
       "...                                        ...   \n",
       "1129                                       NaN   \n",
       "1130                                       NaN   \n",
       "1132                                       NaN   \n",
       "1134                                       NaN   \n",
       "1138                                       NaN   \n",
       "\n",
       "     largeintestine_total_peak_intensity_log10  \\\n",
       "0                                         10.2   \n",
       "1                                          9.6   \n",
       "4                                          9.9   \n",
       "5                                         10.2   \n",
       "7                                          9.8   \n",
       "...                                        ...   \n",
       "1129                                       NaN   \n",
       "1130                                       NaN   \n",
       "1132                                       NaN   \n",
       "1134                                       NaN   \n",
       "1138                                       NaN   \n",
       "\n",
       "      stomach_total_peak_intensity_log10  placenta_total_peak_intensity_log10  \\\n",
       "0                                   10.0                                 10.0   \n",
       "1                                   10.0                                  9.9   \n",
       "4                                    9.9                                 10.0   \n",
       "5                                   10.3                                 10.2   \n",
       "7                                    9.0                                  9.4   \n",
       "...                                  ...                                  ...   \n",
       "1129                                 NaN                                  NaN   \n",
       "1130                                 NaN                                  NaN   \n",
       "1132                                 NaN                                  NaN   \n",
       "1134                                 NaN                                  NaN   \n",
       "1138                                 NaN                                  NaN   \n",
       "\n",
       "     testis_total_peak_intensity_log10  HPA_Main_Location_2020 (Reliability)  \n",
       "0                                  9.8              Mitochondria (Supported)  \n",
       "1                                  8.9              Mitochondria (Supported)  \n",
       "4                                 10.5              Mitochondria (Supported)  \n",
       "5                                  9.9              Mitochondria (Supported)  \n",
       "7                                  9.0              Mitochondria (Supported)  \n",
       "...                                ...                                   ...  \n",
       "1129                               NaN   Mitochondria;Nucleoplasm (Approved)  \n",
       "1130                               NaN               Mitochondria (Approved)  \n",
       "1132                               NaN               Mitochondria (Approved)  \n",
       "1134                               NaN               Mitochondria (Approved)  \n",
       "1138                               NaN              Mitochondria (Supported)  \n",
       "\n",
       "[531 rows x 48 columns]"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# filter MitoCarta3.0 dataset to only include mitochondrial proteins\n",
    "mitocarta3_0_mitochondrial = mitocarta3_0[is_mitochondrial == True]\n",
    "mitocarta3_0_mitochondrial"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "UniProt\n",
       "False    530\n",
       "True       1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mitocarta3_0_mitochondrial[\"UniProt\"].isna().value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/var/folders/75/pkq34r597jg472ny6_np6tdc0000gn/T/ipykernel_69527/1050142315.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  mitocarta3_0_mitochondrial.dropna(subset=[\"UniProt\"], inplace=True)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MouseGeneID</th>\n",
       "      <th>HumanOrthologGeneID</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>Synonyms</th>\n",
       "      <th>Description</th>\n",
       "      <th>MitoCarta3.0_List</th>\n",
       "      <th>MitoCarta3.0_Evidence</th>\n",
       "      <th>MitoCarta3.0_SubMitoLocalization</th>\n",
       "      <th>MitoCarta3.0_MitoPathways</th>\n",
       "      <th>TrainingDataset</th>\n",
       "      <th>...</th>\n",
       "      <th>liver_total_peak_intensity_log10</th>\n",
       "      <th>heart_total_peak_intensity_log10</th>\n",
       "      <th>skeletalmuscle_total_peak_intensity_log10</th>\n",
       "      <th>adipose_total_peak_intensity_log10</th>\n",
       "      <th>smallintestine_total_peak_intensity_log10</th>\n",
       "      <th>largeintestine_total_peak_intensity_log10</th>\n",
       "      <th>stomach_total_peak_intensity_log10</th>\n",
       "      <th>placenta_total_peak_intensity_log10</th>\n",
       "      <th>testis_total_peak_intensity_log10</th>\n",
       "      <th>HPA_Main_Location_2020 (Reliability)</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>66445</td>\n",
       "      <td>1537.0</td>\n",
       "      <td>Cyc1</td>\n",
       "      <td>2610002H19Rik|AA408921|Cyct1</td>\n",
       "      <td>cytochrome c-1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, APEX_IMS, APEX_matrix, targetP sig...</td>\n",
       "      <td>MIM</td>\n",
       "      <td>OXPHOS &gt; Complex III &gt; CIII subunits | Metabol...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.4</td>\n",
       "      <td>9.8</td>\n",
       "      <td>10.1</td>\n",
       "      <td>10.2</td>\n",
       "      <td>10.2</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>9.8</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>18597</td>\n",
       "      <td>5160.0</td>\n",
       "      <td>Pdha1</td>\n",
       "      <td>Pdha|Pdha-1</td>\n",
       "      <td>pyruvate dehydrogenase E1 alpha 1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, APEX_matrix, targetP signal+, yeas...</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Metabolism &gt; Carbohydrate metabolism &gt; Pyruvat...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>9.6</td>\n",
       "      <td>10.5</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.1</td>\n",
       "      <td>9.9</td>\n",
       "      <td>9.6</td>\n",
       "      <td>10.0</td>\n",
       "      <td>9.9</td>\n",
       "      <td>8.9</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>68263</td>\n",
       "      <td>5162.0</td>\n",
       "      <td>Pdhb</td>\n",
       "      <td>2610103L06Rik|AL024199|C81408</td>\n",
       "      <td>pyruvate dehydrogenase (lipoamide) beta</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, targetP signal+, yeast mito homolo...</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Metabolism &gt; Carbohydrate metabolism &gt; Pyruvat...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>9.6</td>\n",
       "      <td>10.5</td>\n",
       "      <td>9.8</td>\n",
       "      <td>10.4</td>\n",
       "      <td>9.8</td>\n",
       "      <td>9.9</td>\n",
       "      <td>9.9</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.5</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>22273</td>\n",
       "      <td>7384.0</td>\n",
       "      <td>Uqcrc1</td>\n",
       "      <td>1110032G10Rik</td>\n",
       "      <td>ubiquinol-cytochrome c reductase core protein 1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, APEX_matrix, targetP signal+, yeas...</td>\n",
       "      <td>MIM</td>\n",
       "      <td>Protein import, sorting and homeostasis &gt; Prot...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.8</td>\n",
       "      <td>10.3</td>\n",
       "      <td>10.1</td>\n",
       "      <td>10.6</td>\n",
       "      <td>10.2</td>\n",
       "      <td>10.3</td>\n",
       "      <td>10.2</td>\n",
       "      <td>9.9</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>56282</td>\n",
       "      <td>6182.0</td>\n",
       "      <td>Mrpl12</td>\n",
       "      <td>0610034O11Rik|1500031N16Rik|L12mt|MRP-|MRP-L12...</td>\n",
       "      <td>mitochondrial ribosomal protein L12</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, targetP signal+, yeast mito homolo...</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Mitochondrial central dogma &gt; mtRNA metabolism...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>9.3</td>\n",
       "      <td>8.9</td>\n",
       "      <td>8.6</td>\n",
       "      <td>9.3</td>\n",
       "      <td>9.2</td>\n",
       "      <td>9.8</td>\n",
       "      <td>9.0</td>\n",
       "      <td>9.4</td>\n",
       "      <td>9.0</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1123</th>\n",
       "      <td>67759</td>\n",
       "      <td>55848.0</td>\n",
       "      <td>Plgrkt</td>\n",
       "      <td>1110007H22Rik|5033414D02Rik|AI852040|Plg-|Plg-...</td>\n",
       "      <td>plasminogen receptor, C-terminal lysine transm...</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>GFP</td>\n",
       "      <td>Membrane</td>\n",
       "      <td>0</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.9</td>\n",
       "      <td>8.6</td>\n",
       "      <td>9.3</td>\n",
       "      <td>9.5</td>\n",
       "      <td>8.1</td>\n",
       "      <td>9.5</td>\n",
       "      <td>8.7</td>\n",
       "      <td>Mitochondria (Uncertain)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1129</th>\n",
       "      <td>78248</td>\n",
       "      <td>51309.0</td>\n",
       "      <td>Armcx1</td>\n",
       "      <td>3010033I09Rik|ALEX1</td>\n",
       "      <td>armadillo repeat containing, X-linked 1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>MOM</td>\n",
       "      <td>Mitochondrial dynamics and surveillance &gt; Traf...</td>\n",
       "      <td>Tnon_mito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mitochondria;Nucleoplasm (Approved)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1130</th>\n",
       "      <td>67416</td>\n",
       "      <td>9823.0</td>\n",
       "      <td>Armcx2</td>\n",
       "      <td>3230401N03Rik|AI043003|ALEX2</td>\n",
       "      <td>armadillo repeat containing, X-linked 2</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>Membrane</td>\n",
       "      <td>0</td>\n",
       "      <td>Tnon_mito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mitochondria (Approved)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1132</th>\n",
       "      <td>269642</td>\n",
       "      <td>339983.0</td>\n",
       "      <td>Nat8l</td>\n",
       "      <td>1110038O08Rik|Sh|Shati</td>\n",
       "      <td>N-acetyltransferase 8-like</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>Membrane</td>\n",
       "      <td>Metabolism &gt; Amino acid metabolism &gt; Glutamate...</td>\n",
       "      <td>Tpossible_mito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mitochondria (Approved)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1134</th>\n",
       "      <td>76781</td>\n",
       "      <td>64863.0</td>\n",
       "      <td>Mettl4</td>\n",
       "      <td>2410198H06Rik|A730091E08Rik|AV296509|HsT66|HsT661</td>\n",
       "      <td>methyltransferase like 4</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Mitochondrial central dogma &gt; mtDNA maintenanc...</td>\n",
       "      <td>Tnon_mito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mitochondria (Approved)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>530 rows × 48 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      MouseGeneID  HumanOrthologGeneID  Symbol  \\\n",
       "0           66445               1537.0    Cyc1   \n",
       "1           18597               5160.0   Pdha1   \n",
       "4           68263               5162.0    Pdhb   \n",
       "5           22273               7384.0  Uqcrc1   \n",
       "7           56282               6182.0  Mrpl12   \n",
       "...           ...                  ...     ...   \n",
       "1123        67759              55848.0  Plgrkt   \n",
       "1129        78248              51309.0  Armcx1   \n",
       "1130        67416               9823.0  Armcx2   \n",
       "1132       269642             339983.0   Nat8l   \n",
       "1134        76781              64863.0  Mettl4   \n",
       "\n",
       "                                               Synonyms  \\\n",
       "0                          2610002H19Rik|AA408921|Cyct1   \n",
       "1                                           Pdha|Pdha-1   \n",
       "4                         2610103L06Rik|AL024199|C81408   \n",
       "5                                         1110032G10Rik   \n",
       "7     0610034O11Rik|1500031N16Rik|L12mt|MRP-|MRP-L12...   \n",
       "...                                                 ...   \n",
       "1123  1110007H22Rik|5033414D02Rik|AI852040|Plg-|Plg-...   \n",
       "1129                                3010033I09Rik|ALEX1   \n",
       "1130                       3230401N03Rik|AI043003|ALEX2   \n",
       "1132                             1110038O08Rik|Sh|Shati   \n",
       "1134  2410198H06Rik|A730091E08Rik|AV296509|HsT66|HsT661   \n",
       "\n",
       "                                            Description MitoCarta3.0_List  \\\n",
       "0                                        cytochrome c-1      MitoCarta3.0   \n",
       "1                     pyruvate dehydrogenase E1 alpha 1      MitoCarta3.0   \n",
       "4               pyruvate dehydrogenase (lipoamide) beta      MitoCarta3.0   \n",
       "5       ubiquinol-cytochrome c reductase core protein 1      MitoCarta3.0   \n",
       "7                   mitochondrial ribosomal protein L12      MitoCarta3.0   \n",
       "...                                                 ...               ...   \n",
       "1123  plasminogen receptor, C-terminal lysine transm...      MitoCarta3.0   \n",
       "1129            armadillo repeat containing, X-linked 1      MitoCarta3.0   \n",
       "1130            armadillo repeat containing, X-linked 2      MitoCarta3.0   \n",
       "1132                         N-acetyltransferase 8-like      MitoCarta3.0   \n",
       "1134                           methyltransferase like 4      MitoCarta3.0   \n",
       "\n",
       "                                  MitoCarta3.0_Evidence  \\\n",
       "0     literature, APEX_IMS, APEX_matrix, targetP sig...   \n",
       "1     literature, APEX_matrix, targetP signal+, yeas...   \n",
       "4     literature, targetP signal+, yeast mito homolo...   \n",
       "5     literature, APEX_matrix, targetP signal+, yeas...   \n",
       "7     literature, targetP signal+, yeast mito homolo...   \n",
       "...                                                 ...   \n",
       "1123                                                GFP   \n",
       "1129                                         literature   \n",
       "1130                                         literature   \n",
       "1132                                         literature   \n",
       "1134                                         literature   \n",
       "\n",
       "     MitoCarta3.0_SubMitoLocalization  \\\n",
       "0                                 MIM   \n",
       "1                              Matrix   \n",
       "4                              Matrix   \n",
       "5                                 MIM   \n",
       "7                              Matrix   \n",
       "...                               ...   \n",
       "1123                         Membrane   \n",
       "1129                              MOM   \n",
       "1130                         Membrane   \n",
       "1132                         Membrane   \n",
       "1134                           Matrix   \n",
       "\n",
       "                              MitoCarta3.0_MitoPathways TrainingDataset  ...  \\\n",
       "0     OXPHOS > Complex III > CIII subunits | Metabol...           Tmito  ...   \n",
       "1     Metabolism > Carbohydrate metabolism > Pyruvat...           Tmito  ...   \n",
       "4     Metabolism > Carbohydrate metabolism > Pyruvat...           Tmito  ...   \n",
       "5     Protein import, sorting and homeostasis > Prot...           Tmito  ...   \n",
       "7     Mitochondrial central dogma > mtRNA metabolism...           Tmito  ...   \n",
       "...                                                 ...             ...  ...   \n",
       "1123                                                  0           Tmito  ...   \n",
       "1129  Mitochondrial dynamics and surveillance > Traf...       Tnon_mito  ...   \n",
       "1130                                                  0       Tnon_mito  ...   \n",
       "1132  Metabolism > Amino acid metabolism > Glutamate...  Tpossible_mito  ...   \n",
       "1134  Mitochondrial central dogma > mtDNA maintenanc...       Tnon_mito  ...   \n",
       "\n",
       "      liver_total_peak_intensity_log10  heart_total_peak_intensity_log10  \\\n",
       "0                                 10.0                              10.4   \n",
       "1                                  9.6                              10.5   \n",
       "4                                  9.6                              10.5   \n",
       "5                                 10.0                              10.8   \n",
       "7                                  9.3                               8.9   \n",
       "...                                ...                               ...   \n",
       "1123                               NaN                               NaN   \n",
       "1129                               NaN                               NaN   \n",
       "1130                               NaN                               NaN   \n",
       "1132                               NaN                               NaN   \n",
       "1134                               NaN                               NaN   \n",
       "\n",
       "      skeletalmuscle_total_peak_intensity_log10  \\\n",
       "0                                           9.8   \n",
       "1                                          10.0   \n",
       "4                                           9.8   \n",
       "5                                          10.3   \n",
       "7                                           8.6   \n",
       "...                                         ...   \n",
       "1123                                        6.9   \n",
       "1129                                        NaN   \n",
       "1130                                        NaN   \n",
       "1132                                        NaN   \n",
       "1134                                        NaN   \n",
       "\n",
       "     adipose_total_peak_intensity_log10  \\\n",
       "0                                  10.1   \n",
       "1                                  10.1   \n",
       "4                                  10.4   \n",
       "5                                  10.1   \n",
       "7                                   9.3   \n",
       "...                                 ...   \n",
       "1123                                8.6   \n",
       "1129                                NaN   \n",
       "1130                                NaN   \n",
       "1132                                NaN   \n",
       "1134                                NaN   \n",
       "\n",
       "     smallintestine_total_peak_intensity_log10  \\\n",
       "0                                         10.2   \n",
       "1                                          9.9   \n",
       "4                                          9.8   \n",
       "5                                         10.6   \n",
       "7                                          9.2   \n",
       "...                                        ...   \n",
       "1123                                       9.3   \n",
       "1129                                       NaN   \n",
       "1130                                       NaN   \n",
       "1132                                       NaN   \n",
       "1134                                       NaN   \n",
       "\n",
       "     largeintestine_total_peak_intensity_log10  \\\n",
       "0                                         10.2   \n",
       "1                                          9.6   \n",
       "4                                          9.9   \n",
       "5                                         10.2   \n",
       "7                                          9.8   \n",
       "...                                        ...   \n",
       "1123                                       9.5   \n",
       "1129                                       NaN   \n",
       "1130                                       NaN   \n",
       "1132                                       NaN   \n",
       "1134                                       NaN   \n",
       "\n",
       "      stomach_total_peak_intensity_log10  placenta_total_peak_intensity_log10  \\\n",
       "0                                   10.0                                 10.0   \n",
       "1                                   10.0                                  9.9   \n",
       "4                                    9.9                                 10.0   \n",
       "5                                   10.3                                 10.2   \n",
       "7                                    9.0                                  9.4   \n",
       "...                                  ...                                  ...   \n",
       "1123                                 8.1                                  9.5   \n",
       "1129                                 NaN                                  NaN   \n",
       "1130                                 NaN                                  NaN   \n",
       "1132                                 NaN                                  NaN   \n",
       "1134                                 NaN                                  NaN   \n",
       "\n",
       "     testis_total_peak_intensity_log10  HPA_Main_Location_2020 (Reliability)  \n",
       "0                                  9.8              Mitochondria (Supported)  \n",
       "1                                  8.9              Mitochondria (Supported)  \n",
       "4                                 10.5              Mitochondria (Supported)  \n",
       "5                                  9.9              Mitochondria (Supported)  \n",
       "7                                  9.0              Mitochondria (Supported)  \n",
       "...                                ...                                   ...  \n",
       "1123                               8.7              Mitochondria (Uncertain)  \n",
       "1129                               NaN   Mitochondria;Nucleoplasm (Approved)  \n",
       "1130                               NaN               Mitochondria (Approved)  \n",
       "1132                               NaN               Mitochondria (Approved)  \n",
       "1134                               NaN               Mitochondria (Approved)  \n",
       "\n",
       "[530 rows x 48 columns]"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# drop rows with NaN UniProt IDs (just one)\n",
    "mitocarta3_0_mitochondrial.dropna(subset=[\"UniProt\"], inplace=True)\n",
    "mitocarta3_0_mitochondrial"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 530/530 [17:02<00:00,  1.93s/it]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MouseGeneID</th>\n",
       "      <th>HumanOrthologGeneID</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>Synonyms</th>\n",
       "      <th>Description</th>\n",
       "      <th>MitoCarta3.0_List</th>\n",
       "      <th>MitoCarta3.0_Evidence</th>\n",
       "      <th>MitoCarta3.0_SubMitoLocalization</th>\n",
       "      <th>MitoCarta3.0_MitoPathways</th>\n",
       "      <th>TrainingDataset</th>\n",
       "      <th>...</th>\n",
       "      <th>heart_total_peak_intensity_log10</th>\n",
       "      <th>skeletalmuscle_total_peak_intensity_log10</th>\n",
       "      <th>adipose_total_peak_intensity_log10</th>\n",
       "      <th>smallintestine_total_peak_intensity_log10</th>\n",
       "      <th>largeintestine_total_peak_intensity_log10</th>\n",
       "      <th>stomach_total_peak_intensity_log10</th>\n",
       "      <th>placenta_total_peak_intensity_log10</th>\n",
       "      <th>testis_total_peak_intensity_log10</th>\n",
       "      <th>HPA_Main_Location_2020 (Reliability)</th>\n",
       "      <th>Complete Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>66445</td>\n",
       "      <td>1537.0</td>\n",
       "      <td>Cyc1</td>\n",
       "      <td>2610002H19Rik|AA408921|Cyct1</td>\n",
       "      <td>cytochrome c-1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, APEX_IMS, APEX_matrix, targetP sig...</td>\n",
       "      <td>MIM</td>\n",
       "      <td>OXPHOS &gt; Complex III &gt; CIII subunits | Metabol...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>10.4</td>\n",
       "      <td>9.8</td>\n",
       "      <td>10.1</td>\n",
       "      <td>10.2</td>\n",
       "      <td>10.2</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>9.8</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "      <td>MAAAAASLRRTVLGPRGVGLPGASAPGLLGGARSRQLPLRTPQAVS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>18597</td>\n",
       "      <td>5160.0</td>\n",
       "      <td>Pdha1</td>\n",
       "      <td>Pdha|Pdha-1</td>\n",
       "      <td>pyruvate dehydrogenase E1 alpha 1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, APEX_matrix, targetP signal+, yeas...</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Metabolism &gt; Carbohydrate metabolism &gt; Pyruvat...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>10.5</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.1</td>\n",
       "      <td>9.9</td>\n",
       "      <td>9.6</td>\n",
       "      <td>10.0</td>\n",
       "      <td>9.9</td>\n",
       "      <td>8.9</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "      <td>MRKMLAAVSRVLAGSAQKPASRVLVASRNFANDATFEIKKCDLHRL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>68263</td>\n",
       "      <td>5162.0</td>\n",
       "      <td>Pdhb</td>\n",
       "      <td>2610103L06Rik|AL024199|C81408</td>\n",
       "      <td>pyruvate dehydrogenase (lipoamide) beta</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, targetP signal+, yeast mito homolo...</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Metabolism &gt; Carbohydrate metabolism &gt; Pyruvat...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>10.5</td>\n",
       "      <td>9.8</td>\n",
       "      <td>10.4</td>\n",
       "      <td>9.8</td>\n",
       "      <td>9.9</td>\n",
       "      <td>9.9</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.5</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "      <td>MAVVAGLVRGPLRQASGLLKRRFHRSAPAAVQLTVREAINQGMDEE...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>22273</td>\n",
       "      <td>7384.0</td>\n",
       "      <td>Uqcrc1</td>\n",
       "      <td>1110032G10Rik</td>\n",
       "      <td>ubiquinol-cytochrome c reductase core protein 1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, APEX_matrix, targetP signal+, yeas...</td>\n",
       "      <td>MIM</td>\n",
       "      <td>Protein import, sorting and homeostasis &gt; Prot...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>10.8</td>\n",
       "      <td>10.3</td>\n",
       "      <td>10.1</td>\n",
       "      <td>10.6</td>\n",
       "      <td>10.2</td>\n",
       "      <td>10.3</td>\n",
       "      <td>10.2</td>\n",
       "      <td>9.9</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "      <td>MAASAVCRAACSGTQVLLRTRRSPALLRLPALRGTATFAQALQSVP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>56282</td>\n",
       "      <td>6182.0</td>\n",
       "      <td>Mrpl12</td>\n",
       "      <td>0610034O11Rik|1500031N16Rik|L12mt|MRP-|MRP-L12...</td>\n",
       "      <td>mitochondrial ribosomal protein L12</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, targetP signal+, yeast mito homolo...</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Mitochondrial central dogma &gt; mtRNA metabolism...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>8.9</td>\n",
       "      <td>8.6</td>\n",
       "      <td>9.3</td>\n",
       "      <td>9.2</td>\n",
       "      <td>9.8</td>\n",
       "      <td>9.0</td>\n",
       "      <td>9.4</td>\n",
       "      <td>9.0</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "      <td>MLPVAASRCLWGPRLGLRGAALRLARQQMPSVCAARQLRSSSHRRS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1123</th>\n",
       "      <td>67759</td>\n",
       "      <td>55848.0</td>\n",
       "      <td>Plgrkt</td>\n",
       "      <td>1110007H22Rik|5033414D02Rik|AI852040|Plg-|Plg-...</td>\n",
       "      <td>plasminogen receptor, C-terminal lysine transm...</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>GFP</td>\n",
       "      <td>Membrane</td>\n",
       "      <td>0</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.9</td>\n",
       "      <td>8.6</td>\n",
       "      <td>9.3</td>\n",
       "      <td>9.5</td>\n",
       "      <td>8.1</td>\n",
       "      <td>9.5</td>\n",
       "      <td>8.7</td>\n",
       "      <td>Mitochondria (Uncertain)</td>\n",
       "      <td>MGFIFSKSMNENMKNQQEFMVTHARLQLERHLTMQNEMRERQMAMQ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1129</th>\n",
       "      <td>78248</td>\n",
       "      <td>51309.0</td>\n",
       "      <td>Armcx1</td>\n",
       "      <td>3010033I09Rik|ALEX1</td>\n",
       "      <td>armadillo repeat containing, X-linked 1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>MOM</td>\n",
       "      <td>Mitochondrial dynamics and surveillance &gt; Traf...</td>\n",
       "      <td>Tnon_mito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mitochondria;Nucleoplasm (Approved)</td>\n",
       "      <td>MGRTREAGCVAAGMVIGAGACYCVYRLTWGKDENEKLWDEEEEEEE...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1130</th>\n",
       "      <td>67416</td>\n",
       "      <td>9823.0</td>\n",
       "      <td>Armcx2</td>\n",
       "      <td>3230401N03Rik|AI043003|ALEX2</td>\n",
       "      <td>armadillo repeat containing, X-linked 2</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>Membrane</td>\n",
       "      <td>0</td>\n",
       "      <td>Tnon_mito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mitochondria (Approved)</td>\n",
       "      <td>MSRARDAGCVAAGIVIGASAWYCVYKYTRGKDQKKKRLTKPKNRAS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1132</th>\n",
       "      <td>269642</td>\n",
       "      <td>339983.0</td>\n",
       "      <td>Nat8l</td>\n",
       "      <td>1110038O08Rik|Sh|Shati</td>\n",
       "      <td>N-acetyltransferase 8-like</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>Membrane</td>\n",
       "      <td>Metabolism &gt; Amino acid metabolism &gt; Glutamate...</td>\n",
       "      <td>Tpossible_mito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mitochondria (Approved)</td>\n",
       "      <td>MHCGPPDMVCETKIVATEDHEALPGAKKDALLVAAGAMWPPLPAAP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1134</th>\n",
       "      <td>76781</td>\n",
       "      <td>64863.0</td>\n",
       "      <td>Mettl4</td>\n",
       "      <td>2410198H06Rik|A730091E08Rik|AV296509|HsT66|HsT661</td>\n",
       "      <td>methyltransferase like 4</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Mitochondrial central dogma &gt; mtDNA maintenanc...</td>\n",
       "      <td>Tnon_mito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mitochondria (Approved)</td>\n",
       "      <td>MSVVHHLPPGWLLDHLSFINKVNYQLCQHQESFCSKNNPTSSVYMD...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>530 rows × 49 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      MouseGeneID  HumanOrthologGeneID  Symbol  \\\n",
       "0           66445               1537.0    Cyc1   \n",
       "1           18597               5160.0   Pdha1   \n",
       "4           68263               5162.0    Pdhb   \n",
       "5           22273               7384.0  Uqcrc1   \n",
       "7           56282               6182.0  Mrpl12   \n",
       "...           ...                  ...     ...   \n",
       "1123        67759              55848.0  Plgrkt   \n",
       "1129        78248              51309.0  Armcx1   \n",
       "1130        67416               9823.0  Armcx2   \n",
       "1132       269642             339983.0   Nat8l   \n",
       "1134        76781              64863.0  Mettl4   \n",
       "\n",
       "                                               Synonyms  \\\n",
       "0                          2610002H19Rik|AA408921|Cyct1   \n",
       "1                                           Pdha|Pdha-1   \n",
       "4                         2610103L06Rik|AL024199|C81408   \n",
       "5                                         1110032G10Rik   \n",
       "7     0610034O11Rik|1500031N16Rik|L12mt|MRP-|MRP-L12...   \n",
       "...                                                 ...   \n",
       "1123  1110007H22Rik|5033414D02Rik|AI852040|Plg-|Plg-...   \n",
       "1129                                3010033I09Rik|ALEX1   \n",
       "1130                       3230401N03Rik|AI043003|ALEX2   \n",
       "1132                             1110038O08Rik|Sh|Shati   \n",
       "1134  2410198H06Rik|A730091E08Rik|AV296509|HsT66|HsT661   \n",
       "\n",
       "                                            Description MitoCarta3.0_List  \\\n",
       "0                                        cytochrome c-1      MitoCarta3.0   \n",
       "1                     pyruvate dehydrogenase E1 alpha 1      MitoCarta3.0   \n",
       "4               pyruvate dehydrogenase (lipoamide) beta      MitoCarta3.0   \n",
       "5       ubiquinol-cytochrome c reductase core protein 1      MitoCarta3.0   \n",
       "7                   mitochondrial ribosomal protein L12      MitoCarta3.0   \n",
       "...                                                 ...               ...   \n",
       "1123  plasminogen receptor, C-terminal lysine transm...      MitoCarta3.0   \n",
       "1129            armadillo repeat containing, X-linked 1      MitoCarta3.0   \n",
       "1130            armadillo repeat containing, X-linked 2      MitoCarta3.0   \n",
       "1132                         N-acetyltransferase 8-like      MitoCarta3.0   \n",
       "1134                           methyltransferase like 4      MitoCarta3.0   \n",
       "\n",
       "                                  MitoCarta3.0_Evidence  \\\n",
       "0     literature, APEX_IMS, APEX_matrix, targetP sig...   \n",
       "1     literature, APEX_matrix, targetP signal+, yeas...   \n",
       "4     literature, targetP signal+, yeast mito homolo...   \n",
       "5     literature, APEX_matrix, targetP signal+, yeas...   \n",
       "7     literature, targetP signal+, yeast mito homolo...   \n",
       "...                                                 ...   \n",
       "1123                                                GFP   \n",
       "1129                                         literature   \n",
       "1130                                         literature   \n",
       "1132                                         literature   \n",
       "1134                                         literature   \n",
       "\n",
       "     MitoCarta3.0_SubMitoLocalization  \\\n",
       "0                                 MIM   \n",
       "1                              Matrix   \n",
       "4                              Matrix   \n",
       "5                                 MIM   \n",
       "7                              Matrix   \n",
       "...                               ...   \n",
       "1123                         Membrane   \n",
       "1129                              MOM   \n",
       "1130                         Membrane   \n",
       "1132                         Membrane   \n",
       "1134                           Matrix   \n",
       "\n",
       "                              MitoCarta3.0_MitoPathways TrainingDataset  ...  \\\n",
       "0     OXPHOS > Complex III > CIII subunits | Metabol...           Tmito  ...   \n",
       "1     Metabolism > Carbohydrate metabolism > Pyruvat...           Tmito  ...   \n",
       "4     Metabolism > Carbohydrate metabolism > Pyruvat...           Tmito  ...   \n",
       "5     Protein import, sorting and homeostasis > Prot...           Tmito  ...   \n",
       "7     Mitochondrial central dogma > mtRNA metabolism...           Tmito  ...   \n",
       "...                                                 ...             ...  ...   \n",
       "1123                                                  0           Tmito  ...   \n",
       "1129  Mitochondrial dynamics and surveillance > Traf...       Tnon_mito  ...   \n",
       "1130                                                  0       Tnon_mito  ...   \n",
       "1132  Metabolism > Amino acid metabolism > Glutamate...  Tpossible_mito  ...   \n",
       "1134  Mitochondrial central dogma > mtDNA maintenanc...       Tnon_mito  ...   \n",
       "\n",
       "      heart_total_peak_intensity_log10  \\\n",
       "0                                 10.4   \n",
       "1                                 10.5   \n",
       "4                                 10.5   \n",
       "5                                 10.8   \n",
       "7                                  8.9   \n",
       "...                                ...   \n",
       "1123                               NaN   \n",
       "1129                               NaN   \n",
       "1130                               NaN   \n",
       "1132                               NaN   \n",
       "1134                               NaN   \n",
       "\n",
       "      skeletalmuscle_total_peak_intensity_log10  \\\n",
       "0                                           9.8   \n",
       "1                                          10.0   \n",
       "4                                           9.8   \n",
       "5                                          10.3   \n",
       "7                                           8.6   \n",
       "...                                         ...   \n",
       "1123                                        6.9   \n",
       "1129                                        NaN   \n",
       "1130                                        NaN   \n",
       "1132                                        NaN   \n",
       "1134                                        NaN   \n",
       "\n",
       "      adipose_total_peak_intensity_log10  \\\n",
       "0                                   10.1   \n",
       "1                                   10.1   \n",
       "4                                   10.4   \n",
       "5                                   10.1   \n",
       "7                                    9.3   \n",
       "...                                  ...   \n",
       "1123                                 8.6   \n",
       "1129                                 NaN   \n",
       "1130                                 NaN   \n",
       "1132                                 NaN   \n",
       "1134                                 NaN   \n",
       "\n",
       "     smallintestine_total_peak_intensity_log10  \\\n",
       "0                                         10.2   \n",
       "1                                          9.9   \n",
       "4                                          9.8   \n",
       "5                                         10.6   \n",
       "7                                          9.2   \n",
       "...                                        ...   \n",
       "1123                                       9.3   \n",
       "1129                                       NaN   \n",
       "1130                                       NaN   \n",
       "1132                                       NaN   \n",
       "1134                                       NaN   \n",
       "\n",
       "     largeintestine_total_peak_intensity_log10  \\\n",
       "0                                         10.2   \n",
       "1                                          9.6   \n",
       "4                                          9.9   \n",
       "5                                         10.2   \n",
       "7                                          9.8   \n",
       "...                                        ...   \n",
       "1123                                       9.5   \n",
       "1129                                       NaN   \n",
       "1130                                       NaN   \n",
       "1132                                       NaN   \n",
       "1134                                       NaN   \n",
       "\n",
       "     stomach_total_peak_intensity_log10  placenta_total_peak_intensity_log10  \\\n",
       "0                                  10.0                                 10.0   \n",
       "1                                  10.0                                  9.9   \n",
       "4                                   9.9                                 10.0   \n",
       "5                                  10.3                                 10.2   \n",
       "7                                   9.0                                  9.4   \n",
       "...                                 ...                                  ...   \n",
       "1123                                8.1                                  9.5   \n",
       "1129                                NaN                                  NaN   \n",
       "1130                                NaN                                  NaN   \n",
       "1132                                NaN                                  NaN   \n",
       "1134                                NaN                                  NaN   \n",
       "\n",
       "      testis_total_peak_intensity_log10 HPA_Main_Location_2020 (Reliability)  \\\n",
       "0                                   9.8             Mitochondria (Supported)   \n",
       "1                                   8.9             Mitochondria (Supported)   \n",
       "4                                  10.5             Mitochondria (Supported)   \n",
       "5                                   9.9             Mitochondria (Supported)   \n",
       "7                                   9.0             Mitochondria (Supported)   \n",
       "...                                 ...                                  ...   \n",
       "1123                                8.7             Mitochondria (Uncertain)   \n",
       "1129                                NaN  Mitochondria;Nucleoplasm (Approved)   \n",
       "1130                                NaN              Mitochondria (Approved)   \n",
       "1132                                NaN              Mitochondria (Approved)   \n",
       "1134                                NaN              Mitochondria (Approved)   \n",
       "\n",
       "                                      Complete Sequence  \n",
       "0     MAAAAASLRRTVLGPRGVGLPGASAPGLLGGARSRQLPLRTPQAVS...  \n",
       "1     MRKMLAAVSRVLAGSAQKPASRVLVASRNFANDATFEIKKCDLHRL...  \n",
       "4     MAVVAGLVRGPLRQASGLLKRRFHRSAPAAVQLTVREAINQGMDEE...  \n",
       "5     MAASAVCRAACSGTQVLLRTRRSPALLRLPALRGTATFAQALQSVP...  \n",
       "7     MLPVAASRCLWGPRLGLRGAALRLARQQMPSVCAARQLRSSSHRRS...  \n",
       "...                                                 ...  \n",
       "1123  MGFIFSKSMNENMKNQQEFMVTHARLQLERHLTMQNEMRERQMAMQ...  \n",
       "1129  MGRTREAGCVAAGMVIGAGACYCVYRLTWGKDENEKLWDEEEEEEE...  \n",
       "1130  MSRARDAGCVAAGIVIGASAWYCVYKYTRGKDQKKKRLTKPKNRAS...  \n",
       "1132  MHCGPPDMVCETKIVATEDHEALPGAKKDALLVAAGAMWPPLPAAP...  \n",
       "1134  MSVVHHLPPGWLLDHLSFINKVNYQLCQHQESFCSKNNPTSSVYMD...  \n",
       "\n",
       "[530 rows x 49 columns]"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# get whole amino acid sequences for mitochondrial proteome\n",
    "# SLOW - ONLY DO THIS ONCE - CONVERT TO CSV FILE, THEN RE-LOAD FROM THAT FILE\n",
    "\n",
    "#tqdm.pandas()\n",
    "#mitocarta3_0_mitochondrial_completed_sequence = mitocarta3_0_mitochondrial.copy()\n",
    "#mitocarta3_0_mitochondrial_completed_sequence[\"Complete Sequence\"] = mitocarta3_0_mitochondrial_completed_sequence[\"UniProt\"].progress_apply(get_full_protein_seq)\n",
    "#mitocarta3_0_mitochondrial_completed_sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "#mitocarta3_0_mitochondrial_completed_sequence.to_csv(os.path.join(datasets_path, \"RvsS_full_mitochondrial_completed_sequence.csv\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MouseGeneID</th>\n",
       "      <th>HumanOrthologGeneID</th>\n",
       "      <th>Symbol</th>\n",
       "      <th>Synonyms</th>\n",
       "      <th>Description</th>\n",
       "      <th>MitoCarta3.0_List</th>\n",
       "      <th>MitoCarta3.0_Evidence</th>\n",
       "      <th>MitoCarta3.0_SubMitoLocalization</th>\n",
       "      <th>MitoCarta3.0_MitoPathways</th>\n",
       "      <th>TrainingDataset</th>\n",
       "      <th>...</th>\n",
       "      <th>heart_total_peak_intensity_log10</th>\n",
       "      <th>skeletalmuscle_total_peak_intensity_log10</th>\n",
       "      <th>adipose_total_peak_intensity_log10</th>\n",
       "      <th>smallintestine_total_peak_intensity_log10</th>\n",
       "      <th>largeintestine_total_peak_intensity_log10</th>\n",
       "      <th>stomach_total_peak_intensity_log10</th>\n",
       "      <th>placenta_total_peak_intensity_log10</th>\n",
       "      <th>testis_total_peak_intensity_log10</th>\n",
       "      <th>HPA_Main_Location_2020 (Reliability)</th>\n",
       "      <th>Complete Sequence</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>66445</td>\n",
       "      <td>1537.0</td>\n",
       "      <td>Cyc1</td>\n",
       "      <td>2610002H19Rik|AA408921|Cyct1</td>\n",
       "      <td>cytochrome c-1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, APEX_IMS, APEX_matrix, targetP sig...</td>\n",
       "      <td>MIM</td>\n",
       "      <td>OXPHOS &gt; Complex III &gt; CIII subunits | Metabol...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>10.4</td>\n",
       "      <td>9.8</td>\n",
       "      <td>10.1</td>\n",
       "      <td>10.2</td>\n",
       "      <td>10.2</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>9.8</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "      <td>MAAAAASLRRTVLGPRGVGLPGASAPGLLGGARSRQLPLRTPQAVS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>18597</td>\n",
       "      <td>5160.0</td>\n",
       "      <td>Pdha1</td>\n",
       "      <td>Pdha|Pdha-1</td>\n",
       "      <td>pyruvate dehydrogenase E1 alpha 1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, APEX_matrix, targetP signal+, yeas...</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Metabolism &gt; Carbohydrate metabolism &gt; Pyruvat...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>10.5</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.1</td>\n",
       "      <td>9.9</td>\n",
       "      <td>9.6</td>\n",
       "      <td>10.0</td>\n",
       "      <td>9.9</td>\n",
       "      <td>8.9</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "      <td>MRKMLAAVSRVLAGSAQKPASRVLVASRNFANDATFEIKKCDLHRL...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>68263</td>\n",
       "      <td>5162.0</td>\n",
       "      <td>Pdhb</td>\n",
       "      <td>2610103L06Rik|AL024199|C81408</td>\n",
       "      <td>pyruvate dehydrogenase (lipoamide) beta</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, targetP signal+, yeast mito homolo...</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Metabolism &gt; Carbohydrate metabolism &gt; Pyruvat...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>10.5</td>\n",
       "      <td>9.8</td>\n",
       "      <td>10.4</td>\n",
       "      <td>9.8</td>\n",
       "      <td>9.9</td>\n",
       "      <td>9.9</td>\n",
       "      <td>10.0</td>\n",
       "      <td>10.5</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "      <td>MAVVAGLVRGPLRQASGLLKRRFHRSAPAAVQLTVREAINQGMDEE...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>22273</td>\n",
       "      <td>7384.0</td>\n",
       "      <td>Uqcrc1</td>\n",
       "      <td>1110032G10Rik</td>\n",
       "      <td>ubiquinol-cytochrome c reductase core protein 1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, APEX_matrix, targetP signal+, yeas...</td>\n",
       "      <td>MIM</td>\n",
       "      <td>Protein import, sorting and homeostasis &gt; Prot...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>10.8</td>\n",
       "      <td>10.3</td>\n",
       "      <td>10.1</td>\n",
       "      <td>10.6</td>\n",
       "      <td>10.2</td>\n",
       "      <td>10.3</td>\n",
       "      <td>10.2</td>\n",
       "      <td>9.9</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "      <td>MAASAVCRAACSGTQVLLRTRRSPALLRLPALRGTATFAQALQSVP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>56282</td>\n",
       "      <td>6182.0</td>\n",
       "      <td>Mrpl12</td>\n",
       "      <td>0610034O11Rik|1500031N16Rik|L12mt|MRP-|MRP-L12...</td>\n",
       "      <td>mitochondrial ribosomal protein L12</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature, targetP signal+, yeast mito homolo...</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Mitochondrial central dogma &gt; mtRNA metabolism...</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>8.9</td>\n",
       "      <td>8.6</td>\n",
       "      <td>9.3</td>\n",
       "      <td>9.2</td>\n",
       "      <td>9.8</td>\n",
       "      <td>9.0</td>\n",
       "      <td>9.4</td>\n",
       "      <td>9.0</td>\n",
       "      <td>Mitochondria (Supported)</td>\n",
       "      <td>MLPVAASRCLWGPRLGLRGAALRLARQQMPSVCAARQLRSSSHRRS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1123</th>\n",
       "      <td>67759</td>\n",
       "      <td>55848.0</td>\n",
       "      <td>Plgrkt</td>\n",
       "      <td>1110007H22Rik|5033414D02Rik|AI852040|Plg-|Plg-...</td>\n",
       "      <td>plasminogen receptor, C-terminal lysine transm...</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>GFP</td>\n",
       "      <td>Membrane</td>\n",
       "      <td>0</td>\n",
       "      <td>Tmito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6.9</td>\n",
       "      <td>8.6</td>\n",
       "      <td>9.3</td>\n",
       "      <td>9.5</td>\n",
       "      <td>8.1</td>\n",
       "      <td>9.5</td>\n",
       "      <td>8.7</td>\n",
       "      <td>Mitochondria (Uncertain)</td>\n",
       "      <td>MGFIFSKSMNENMKNQQEFMVTHARLQLERHLTMQNEMRERQMAMQ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1129</th>\n",
       "      <td>78248</td>\n",
       "      <td>51309.0</td>\n",
       "      <td>Armcx1</td>\n",
       "      <td>3010033I09Rik|ALEX1</td>\n",
       "      <td>armadillo repeat containing, X-linked 1</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>MOM</td>\n",
       "      <td>Mitochondrial dynamics and surveillance &gt; Traf...</td>\n",
       "      <td>Tnon_mito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mitochondria;Nucleoplasm (Approved)</td>\n",
       "      <td>MGRTREAGCVAAGMVIGAGACYCVYRLTWGKDENEKLWDEEEEEEE...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1130</th>\n",
       "      <td>67416</td>\n",
       "      <td>9823.0</td>\n",
       "      <td>Armcx2</td>\n",
       "      <td>3230401N03Rik|AI043003|ALEX2</td>\n",
       "      <td>armadillo repeat containing, X-linked 2</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>Membrane</td>\n",
       "      <td>0</td>\n",
       "      <td>Tnon_mito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mitochondria (Approved)</td>\n",
       "      <td>MSRARDAGCVAAGIVIGASAWYCVYKYTRGKDQKKKRLTKPKNRAS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1132</th>\n",
       "      <td>269642</td>\n",
       "      <td>339983.0</td>\n",
       "      <td>Nat8l</td>\n",
       "      <td>1110038O08Rik|Sh|Shati</td>\n",
       "      <td>N-acetyltransferase 8-like</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>Membrane</td>\n",
       "      <td>Metabolism &gt; Amino acid metabolism &gt; Glutamate...</td>\n",
       "      <td>Tpossible_mito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mitochondria (Approved)</td>\n",
       "      <td>MHCGPPDMVCETKIVATEDHEALPGAKKDALLVAAGAMWPPLPAAP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1134</th>\n",
       "      <td>76781</td>\n",
       "      <td>64863.0</td>\n",
       "      <td>Mettl4</td>\n",
       "      <td>2410198H06Rik|A730091E08Rik|AV296509|HsT66|HsT661</td>\n",
       "      <td>methyltransferase like 4</td>\n",
       "      <td>MitoCarta3.0</td>\n",
       "      <td>literature</td>\n",
       "      <td>Matrix</td>\n",
       "      <td>Mitochondrial central dogma &gt; mtDNA maintenanc...</td>\n",
       "      <td>Tnon_mito</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mitochondria (Approved)</td>\n",
       "      <td>MSVVHHLPPGWLLDHLSFINKVNYQLCQHQESFCSKNNPTSSVYMD...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>530 rows × 49 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      MouseGeneID  HumanOrthologGeneID  Symbol  \\\n",
       "0           66445               1537.0    Cyc1   \n",
       "1           18597               5160.0   Pdha1   \n",
       "4           68263               5162.0    Pdhb   \n",
       "5           22273               7384.0  Uqcrc1   \n",
       "7           56282               6182.0  Mrpl12   \n",
       "...           ...                  ...     ...   \n",
       "1123        67759              55848.0  Plgrkt   \n",
       "1129        78248              51309.0  Armcx1   \n",
       "1130        67416               9823.0  Armcx2   \n",
       "1132       269642             339983.0   Nat8l   \n",
       "1134        76781              64863.0  Mettl4   \n",
       "\n",
       "                                               Synonyms  \\\n",
       "0                          2610002H19Rik|AA408921|Cyct1   \n",
       "1                                           Pdha|Pdha-1   \n",
       "4                         2610103L06Rik|AL024199|C81408   \n",
       "5                                         1110032G10Rik   \n",
       "7     0610034O11Rik|1500031N16Rik|L12mt|MRP-|MRP-L12...   \n",
       "...                                                 ...   \n",
       "1123  1110007H22Rik|5033414D02Rik|AI852040|Plg-|Plg-...   \n",
       "1129                                3010033I09Rik|ALEX1   \n",
       "1130                       3230401N03Rik|AI043003|ALEX2   \n",
       "1132                             1110038O08Rik|Sh|Shati   \n",
       "1134  2410198H06Rik|A730091E08Rik|AV296509|HsT66|HsT661   \n",
       "\n",
       "                                            Description MitoCarta3.0_List  \\\n",
       "0                                        cytochrome c-1      MitoCarta3.0   \n",
       "1                     pyruvate dehydrogenase E1 alpha 1      MitoCarta3.0   \n",
       "4               pyruvate dehydrogenase (lipoamide) beta      MitoCarta3.0   \n",
       "5       ubiquinol-cytochrome c reductase core protein 1      MitoCarta3.0   \n",
       "7                   mitochondrial ribosomal protein L12      MitoCarta3.0   \n",
       "...                                                 ...               ...   \n",
       "1123  plasminogen receptor, C-terminal lysine transm...      MitoCarta3.0   \n",
       "1129            armadillo repeat containing, X-linked 1      MitoCarta3.0   \n",
       "1130            armadillo repeat containing, X-linked 2      MitoCarta3.0   \n",
       "1132                         N-acetyltransferase 8-like      MitoCarta3.0   \n",
       "1134                           methyltransferase like 4      MitoCarta3.0   \n",
       "\n",
       "                                  MitoCarta3.0_Evidence  \\\n",
       "0     literature, APEX_IMS, APEX_matrix, targetP sig...   \n",
       "1     literature, APEX_matrix, targetP signal+, yeas...   \n",
       "4     literature, targetP signal+, yeast mito homolo...   \n",
       "5     literature, APEX_matrix, targetP signal+, yeas...   \n",
       "7     literature, targetP signal+, yeast mito homolo...   \n",
       "...                                                 ...   \n",
       "1123                                                GFP   \n",
       "1129                                         literature   \n",
       "1130                                         literature   \n",
       "1132                                         literature   \n",
       "1134                                         literature   \n",
       "\n",
       "     MitoCarta3.0_SubMitoLocalization  \\\n",
       "0                                 MIM   \n",
       "1                              Matrix   \n",
       "4                              Matrix   \n",
       "5                                 MIM   \n",
       "7                              Matrix   \n",
       "...                               ...   \n",
       "1123                         Membrane   \n",
       "1129                              MOM   \n",
       "1130                         Membrane   \n",
       "1132                         Membrane   \n",
       "1134                           Matrix   \n",
       "\n",
       "                              MitoCarta3.0_MitoPathways TrainingDataset  ...  \\\n",
       "0     OXPHOS > Complex III > CIII subunits | Metabol...           Tmito  ...   \n",
       "1     Metabolism > Carbohydrate metabolism > Pyruvat...           Tmito  ...   \n",
       "4     Metabolism > Carbohydrate metabolism > Pyruvat...           Tmito  ...   \n",
       "5     Protein import, sorting and homeostasis > Prot...           Tmito  ...   \n",
       "7     Mitochondrial central dogma > mtRNA metabolism...           Tmito  ...   \n",
       "...                                                 ...             ...  ...   \n",
       "1123                                                  0           Tmito  ...   \n",
       "1129  Mitochondrial dynamics and surveillance > Traf...       Tnon_mito  ...   \n",
       "1130                                                  0       Tnon_mito  ...   \n",
       "1132  Metabolism > Amino acid metabolism > Glutamate...  Tpossible_mito  ...   \n",
       "1134  Mitochondrial central dogma > mtDNA maintenanc...       Tnon_mito  ...   \n",
       "\n",
       "      heart_total_peak_intensity_log10  \\\n",
       "0                                 10.4   \n",
       "1                                 10.5   \n",
       "4                                 10.5   \n",
       "5                                 10.8   \n",
       "7                                  8.9   \n",
       "...                                ...   \n",
       "1123                               NaN   \n",
       "1129                               NaN   \n",
       "1130                               NaN   \n",
       "1132                               NaN   \n",
       "1134                               NaN   \n",
       "\n",
       "      skeletalmuscle_total_peak_intensity_log10  \\\n",
       "0                                           9.8   \n",
       "1                                          10.0   \n",
       "4                                           9.8   \n",
       "5                                          10.3   \n",
       "7                                           8.6   \n",
       "...                                         ...   \n",
       "1123                                        6.9   \n",
       "1129                                        NaN   \n",
       "1130                                        NaN   \n",
       "1132                                        NaN   \n",
       "1134                                        NaN   \n",
       "\n",
       "      adipose_total_peak_intensity_log10  \\\n",
       "0                                   10.1   \n",
       "1                                   10.1   \n",
       "4                                   10.4   \n",
       "5                                   10.1   \n",
       "7                                    9.3   \n",
       "...                                  ...   \n",
       "1123                                 8.6   \n",
       "1129                                 NaN   \n",
       "1130                                 NaN   \n",
       "1132                                 NaN   \n",
       "1134                                 NaN   \n",
       "\n",
       "     smallintestine_total_peak_intensity_log10  \\\n",
       "0                                         10.2   \n",
       "1                                          9.9   \n",
       "4                                          9.8   \n",
       "5                                         10.6   \n",
       "7                                          9.2   \n",
       "...                                        ...   \n",
       "1123                                       9.3   \n",
       "1129                                       NaN   \n",
       "1130                                       NaN   \n",
       "1132                                       NaN   \n",
       "1134                                       NaN   \n",
       "\n",
       "     largeintestine_total_peak_intensity_log10  \\\n",
       "0                                         10.2   \n",
       "1                                          9.6   \n",
       "4                                          9.9   \n",
       "5                                         10.2   \n",
       "7                                          9.8   \n",
       "...                                        ...   \n",
       "1123                                       9.5   \n",
       "1129                                       NaN   \n",
       "1130                                       NaN   \n",
       "1132                                       NaN   \n",
       "1134                                       NaN   \n",
       "\n",
       "     stomach_total_peak_intensity_log10  placenta_total_peak_intensity_log10  \\\n",
       "0                                  10.0                                 10.0   \n",
       "1                                  10.0                                  9.9   \n",
       "4                                   9.9                                 10.0   \n",
       "5                                  10.3                                 10.2   \n",
       "7                                   9.0                                  9.4   \n",
       "...                                 ...                                  ...   \n",
       "1123                                8.1                                  9.5   \n",
       "1129                                NaN                                  NaN   \n",
       "1130                                NaN                                  NaN   \n",
       "1132                                NaN                                  NaN   \n",
       "1134                                NaN                                  NaN   \n",
       "\n",
       "      testis_total_peak_intensity_log10 HPA_Main_Location_2020 (Reliability)  \\\n",
       "0                                   9.8             Mitochondria (Supported)   \n",
       "1                                   8.9             Mitochondria (Supported)   \n",
       "4                                  10.5             Mitochondria (Supported)   \n",
       "5                                   9.9             Mitochondria (Supported)   \n",
       "7                                   9.0             Mitochondria (Supported)   \n",
       "...                                 ...                                  ...   \n",
       "1123                                8.7             Mitochondria (Uncertain)   \n",
       "1129                                NaN  Mitochondria;Nucleoplasm (Approved)   \n",
       "1130                                NaN              Mitochondria (Approved)   \n",
       "1132                                NaN              Mitochondria (Approved)   \n",
       "1134                                NaN              Mitochondria (Approved)   \n",
       "\n",
       "                                      Complete Sequence  \n",
       "0     MAAAAASLRRTVLGPRGVGLPGASAPGLLGGARSRQLPLRTPQAVS...  \n",
       "1     MRKMLAAVSRVLAGSAQKPASRVLVASRNFANDATFEIKKCDLHRL...  \n",
       "4     MAVVAGLVRGPLRQASGLLKRRFHRSAPAAVQLTVREAINQGMDEE...  \n",
       "5     MAASAVCRAACSGTQVLLRTRRSPALLRLPALRGTATFAQALQSVP...  \n",
       "7     MLPVAASRCLWGPRLGLRGAALRLARQQMPSVCAARQLRSSSHRRS...  \n",
       "...                                                 ...  \n",
       "1123  MGFIFSKSMNENMKNQQEFMVTHARLQLERHLTMQNEMRERQMAMQ...  \n",
       "1129  MGRTREAGCVAAGMVIGAGACYCVYRLTWGKDENEKLWDEEEEEEE...  \n",
       "1130  MSRARDAGCVAAGIVIGASAWYCVYKYTRGKDQKKKRLTKPKNRAS...  \n",
       "1132  MHCGPPDMVCETKIVATEDHEALPGAKKDALLVAAGAMWPPLPAAP...  \n",
       "1134  MSVVHHLPPGWLLDHLSFINKVNYQLCQHQESFCSKNNPTSSVYMD...  \n",
       "\n",
       "[530 rows x 49 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "path = os.path.join(datasets_path, \"RvsS_full_mitochondrial_completed_sequence.csv\")\n",
    "mitocarta3_0_mitochondrial_completed_sequence = pd.read_csv(path)\n",
    "mitocarta3_0_mitochondrial_completed_sequence.set_index(\"Unnamed: 0\", inplace=True)\n",
    "mitocarta3_0_mitochondrial_completed_sequence.index.name = None\n",
    "mitocarta3_0_mitochondrial_completed_sequence"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       " 0      491\n",
       " 1        6\n",
       " 2        3\n",
       " 6        3\n",
       " 18       2\n",
       " 19       2\n",
       " 27       1\n",
       " 56       1\n",
       "-7        1\n",
       "-1        1\n",
       " 23       1\n",
       " 68       1\n",
       " 36       1\n",
       " 10       1\n",
       " 24       1\n",
       " 14       1\n",
       "-29       1\n",
       "-107      1\n",
       " 15       1\n",
       " 45       1\n",
       " 3        1\n",
       " 4        1\n",
       " 144      1\n",
       " 9        1\n",
       " 364      1\n",
       " 7        1\n",
       " 5        1\n",
       "-68       1\n",
       "-114      1\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# NOTE: sequence length from database doesn't exactly match up with length of sequence as determined by UniProtID - weird\n",
    "(mitocarta3_0_mitochondrial_completed_sequence[\"ProteinLength\"] - mitocarta3_0_mitochondrial_completed_sequence[\"Complete Sequence\"].str.len()).value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Download Alphafold Data - Labeled Methionines"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/ritwiksrinivas/Desktop/Projects/methionine-analysis/alphafold_data\n",
      "/Users/ritwiksrinivas/Desktop/Projects/methionine-analysis/alphafold_data/cif\n",
      "/Users/ritwiksrinivas/Desktop/Projects/methionine-analysis/alphafold_data/pae\n"
     ]
    }
   ],
   "source": [
    "# path for alphafold protein data\n",
    "alphafold_path_str = \"../alphafold_data/\"\n",
    "alphafold_path = os.path.abspath(alphafold_path_str)\n",
    "\n",
    "cif_dir = os.path.join(alphafold_path, \"cif\")\n",
    "pae_dir = os.path.join(alphafold_path, \"pae\")\n",
    "\n",
    "print(alphafold_path)\n",
    "print(cif_dir)\n",
    "print(pae_dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Q9D0M3', 'P35486', 'Q9D051', 'Q9CZ13', 'Q9DB15', 'Q8BJZ4',\n",
       "       'Q9CR68', 'Q8K2B3', 'Q9CQA3', 'Q03265', 'Q8BTE0', 'Q9CXT8',\n",
       "       'Q9DB77', 'Q60597', 'Q9CQ69', 'P08249', 'P53395', 'Q8BMF4',\n",
       "       'Q6PB66', 'Q9D1H8', 'Q8K0D5', 'Q99LP6', 'Q8BMS4', 'Q8VEM8',\n",
       "       'Q9DC71', 'P56480', 'Q8VE22', 'Q99LC5', 'P97807', 'Q8BK72',\n",
       "       'Q9D6R2', 'Q91VD9', 'Q8JZQ2', 'Q9DC61', 'Q9Z2I9', 'Q8VDC0',\n",
       "       'Q9DB20', 'Q9EQI8', 'Q8BJ03', 'Q9CQJ1', 'P62073', 'Q8BH95',\n",
       "       'Q8BMS1', 'Q9WUM5', 'Q9CZU6', 'P43024', 'Q07417', 'Q9D6J6',\n",
       "       'Q8BIJ6', 'Q06185', 'Q91VA7', 'Q9Z2I0', 'Q5RL20', 'Q8K4F5',\n",
       "       'Q9D880', 'Q9D1B9', 'Q8C6I2', 'Q9D6J5', 'O08749', 'Q91WD5',\n",
       "       'Q91YT0', 'P58281', 'Q8CAQ8', 'Q78IK4', 'Q8CGK3', 'Q924L1',\n",
       "       'Q91YJ5', 'Q8K3J1', 'Q9D773', 'Q8R2Q4', 'Q9WV98', 'Q8K411',\n",
       "       'P45952', 'Q80YD1', 'Q99N96', 'Q99KI0', 'Q8BGH2', 'Q924T2',\n",
       "       'Q9CY73', 'P67778', 'Q8C3X4', 'Q8BKF1', 'Q91VC9', 'P62075',\n",
       "       'P52825', 'Q3TBW2', 'P97450', 'Q8CAK1', 'Q9Z2Q5', 'Q8QZS1',\n",
       "       'Q9D0K2', 'Q9D172', 'Q9CQJ8', 'Q9CPQ1', 'P09671', 'Q9CQH3',\n",
       "       'Q61425', 'Q99M04', 'P46656', 'Q99J25', 'Q9CWV0', 'Q99JT1',\n",
       "       'Q8BYL4', 'O35943', 'P70404', 'Q9CY16', 'O35857', 'Q9JHS4',\n",
       "       'Q9DCS3', 'Q9Z2Z6', 'P48962', 'Q791V5', 'Q9JK42', 'P56391',\n",
       "       'Q9CR88', 'P38647', 'Q9DC69', 'Q7TNS2', 'P56394', 'Q9CXZ1',\n",
       "       'O35129', 'Q6P3A8', 'Q8R3F5', 'Q91ZA3', 'P19783', 'Q9D1H6',\n",
       "       'Q5SWT3', 'Q9ER88', 'Q3U276', 'Q8VC19', 'Q8K199', 'O55028',\n",
       "       'P97493', 'Q9Z2I8', 'Q811U4', 'Q3UIU2', 'O88441', 'Q8QZT1',\n",
       "       'Q9D1N9', 'Q8K0Z7', 'Q8BQU3', 'Q9CQF0', 'Q3TLP5', 'Q9D881',\n",
       "       'Q9DBL1', 'Q9CZU4', 'Q2TPA8', 'P63038', 'Q9CZR8', 'Q9DCM0',\n",
       "       'Q60930', 'Q8BGD8', 'Q8BGB8', 'Q9JHI5', 'P50544', 'Q8R0Y8',\n",
       "       'Q9CXJ4', 'Q80X85', 'Q99MR8', 'Q9D7N3', 'Q99M87', 'Q9D338',\n",
       "       'A6H611', 'Q60936', 'Q80XN0', 'P53702', 'Q99LC3', 'O88696',\n",
       "       'Q9WTP7', 'O35143', 'Q9CRB9', 'Q9EQ20', 'Q8BFP9', 'Q99KR7',\n",
       "       'O35459', 'Q8BWT1', 'Q9Z1P6', 'Q9CQL4', 'Q9Z0V8', 'Q9CPW3',\n",
       "       'Q9CPQ3', 'Q99N87', 'A2ATU0', 'Q921N7', 'Q9D6Z0', 'Q9CXW2',\n",
       "       'Q9CYK1', 'Q9JIK9', 'Q8BK30', 'Q9CPU2', 'P51881', 'P58059',\n",
       "       'Q80Y14', 'Q7TNL9', 'Q9QXX4', 'Q9QZD8', 'Q9DB10', 'Q8BU88',\n",
       "       'G5E8U5', 'Q61578', 'Q9D8T7', 'Q91VA6', 'Q8BGA9', 'P54071',\n",
       "       'Q9CPQ8', 'Q9CQ54', 'Q9DCJ5', 'Q3V3R1', 'Q922H2', 'Q8BH51',\n",
       "       'Q60759', 'Q9CPY1', 'F7C846', 'Q9CQN1', 'Q61102', 'P26443',\n",
       "       'Q8BFR5', 'Q8BGV0', 'Q80ZS3', 'Q8BYM8', 'Q8CFY5', 'O09111',\n",
       "       'Q9CYR0', 'Q8K3A0', 'Q9D0S9', 'Q9DCW4', 'P62897', 'O35972',\n",
       "       'Q99N94', 'Q9CQA6', 'Q925I1', 'Q91WS0', 'Q8BIP0', 'Q9Z110',\n",
       "       'P58064', 'Q91VM9', 'E9QPD7', 'Q91W43', 'Q8K2M0', 'Q7TMF3',\n",
       "       'S4R2K0', 'Q8BWM0', 'Q8R3Q6', 'Q3V132', 'Q9DCX2', 'Q9CQQ7',\n",
       "       'Q9CQN7', 'Q9JLT4', 'Q3ULD5', 'Q9WUR9', 'A2ASZ8', 'Q9QYA2',\n",
       "       'Q9CQ62', 'Q8CI78', 'Q5U680', 'Q62425', 'Q9D8S4', 'Q99KE1',\n",
       "       'Q9JJL8', 'Q9D113', 'Q8R164', 'Q8VEA4', 'Q921S7', 'Q5HZI9',\n",
       "       'Q8BGY7', 'Q9CR63', 'P42125', 'Q9DCT2', 'P54869', 'Q91YP0',\n",
       "       'Q9DCJ7', 'Q99M07', 'Q8R404', 'Q7TMG8', 'Q8K2Y7', 'Q8JZN5',\n",
       "       'Q8R4H7', 'Q99LB7', 'Q9CXU4', 'Q8R2R6', 'Q8K1J6', 'Q9D125',\n",
       "       'Q9D1I6', 'Q9CPX7', 'Q9CZN7', 'Q8C3X2', 'Q99N90', 'Q9CR13',\n",
       "       'Q8R035', 'P97742', 'D3Z7P3', 'Q3U8Y1', 'Q9CQY9', 'Q8BMD8',\n",
       "       'Q9D6M3', 'P63030', 'Q9WVA2', 'Q60931', 'O08715', 'Q9WUR2',\n",
       "       'Q9CWD8', 'Q99JR1', 'Q9D964', 'Q9D8Z2', 'P36552', 'Q9CR59',\n",
       "       'P56213', 'Q99KS2', 'Q9CQF8', 'Q14C51', 'Q9D0Q7', 'Q8C0K5',\n",
       "       'Q9CR21', 'F6W8I0', 'Q9EP89', 'P52196', 'Q9CZS1', 'Q9DAT5',\n",
       "       'Q3U2A8', 'Q99K67', 'G5E814', 'Q9D273', 'Q8C1W8', 'Q8BK08',\n",
       "       'Q5ND52', 'Q8VCL2', 'Q9DBT9', 'Q8CEE7', 'P99029', 'Q8BHE8',\n",
       "       'Q5SSK3', 'Q9D2R6', 'Q8BH04', 'Q6PE15', 'Q9QYR9', 'Q8JZN7',\n",
       "       'Q9JLR9', 'Q91YM4', 'Q9DCZ4', 'O35855', 'P56379', 'P29758',\n",
       "       'Q99JY0', 'Q9QZA0', 'O88396', 'Q8BGF9', 'Q8BPT6', 'Q8K009',\n",
       "       'Q9D1L0', 'Q9JJG9', 'A2AMZ4', 'O88967', 'Q91WM2', 'Q9DBF1',\n",
       "       'Q7TSQ8', 'Q91YP2', 'Q59J78', 'Q922E6', 'Q9CXJ1', 'H3BLL2',\n",
       "       'Q91V61', 'Q9R112', 'Q8K2C6', 'Q8CIM3', 'Q9JIQ3', 'Q8CC21',\n",
       "       'P00397', 'Q9D6U8', 'P56135', 'Q8BUY5', 'P40630', 'Q9CWE0',\n",
       "       'Q8BGT5', 'Q9CQN3', 'Q9CQX8', 'Q9CZW5', 'Q6PEB4', 'Q3UG70',\n",
       "       'Q9Z2B2', 'Q9CQS4', 'Q8R107', 'Q8BMG8', 'Q3UMR5', 'Q9WU56',\n",
       "       'Q504M2', 'E9QLB8', 'Q9DBG1', 'Q9D0Y8', 'Q9CQ92', 'Q8CFA2',\n",
       "       'Q8C163', 'Q3UW66', 'Q922G0', 'P61922', 'Q8BWF0', 'Q923K4',\n",
       "       'Q9CWB7', 'Q5XJY4', 'Q925N0', 'Q9D6D0', 'Q6GQS1', 'Q9JIY5',\n",
       "       'P00158', 'Q9CQP3', 'Q61733', 'O35683', 'Q9CQV1', 'B1AXP6',\n",
       "       'Q9CQC7', 'Q9CR98', 'Q8R0F8', 'Q8C5Q4', 'Q99JR6', 'Q9CPS6',\n",
       "       'Q9CPV3', 'Q9WV84', 'Q9D8H7', 'Q8CFI5', 'Q91VN4', 'Q810S1',\n",
       "       'Q9D404', 'Q91VS7', 'Q8R1R3', 'Q8C2E4', 'Q9CRB8', 'Q8VCF0',\n",
       "       'Q9D0D3', 'Q3U5Q7', 'Q8BVN4', 'Q8R4U6', 'O08691', 'Q8R2Y8',\n",
       "       'Q14CH7', 'Q91ZE0', 'Q9D7J4', 'Q925N2', 'P11725', 'Q3UFY8',\n",
       "       'Q8BX10', 'Q66GT5', 'Q8BTX9', 'Q3TQB2', 'Q99PU8', 'P59017',\n",
       "       'Q61387', 'Q8BVI4', 'Q9CZD5', 'P0DJE0', 'Q99MB2', 'P50637',\n",
       "       'Q6P3B9', 'Q14DH7', 'Q3URS9', 'Q8R3J4', 'Q8BVU5', 'Q922Q4',\n",
       "       'A2ADF7', 'Q920G8', 'P70406', 'Q9CW42', 'Q8VCM5', 'Q9D7K5',\n",
       "       'Q8VCX5', 'Q3UHB1', 'Q69ZP3', 'Q9D6K5', 'B9EJ57', 'Q9CRD0',\n",
       "       'Q8C725', 'D3YTP3', 'Q9D1R1', 'O35465', 'Q8CEI1', 'Q8K1R3',\n",
       "       'Q3TL26', 'Q99N84', 'O35435', 'O88741', 'Q9D8W7', 'Q3KNM2',\n",
       "       'Q9CWT6', 'Q9D1E8', 'Q61586', 'Q9JMA2', 'Q8K248', 'P85094',\n",
       "       'Q149L0', 'Q64133', 'Q6PCP5', 'Q3TL44', 'Q8BGV8', 'Q921H9',\n",
       "       'Q9CRA7', 'Q3UJU9', 'Q78IK2', 'Q9DCC8', 'Q9Z2F7', 'Q3TY86',\n",
       "       'Q8BGX2', 'Q8BXN7', 'P03903', 'Q9CQY6', 'Q3U6U5', 'Q8VI64',\n",
       "       'P97287', 'Q9CXC3', 'Q91XR9', 'Q3U5F4', 'Q5SV80', 'Q64441',\n",
       "       'P97363', 'Q9ESW4', 'Q14DK4', 'Q9JIZ9', 'Q8VEG4', 'Q64373',\n",
       "       'Q9D0L7', 'Q8BSN9', 'Q4QQM5', 'Q8JZY4', 'Q6ZQJ5', 'Q8C1W2',\n",
       "       'Q8C3Q9', 'P70677', 'Q61337', 'Q9D3P8', 'Q9CX83', 'Q6A058',\n",
       "       'Q3UGX3', 'Q3U034'], dtype=object)"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# set uniprot IDs to use\n",
    "uniprotIDs_fullproteome = mitocarta3_0_mitochondrial_completed_sequence[\"UniProt\"].values\n",
    "uniprotIDs_fullproteome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 530/530 [20:17<00:00,  2.30s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-04-23 19:20:10> Valid proteins: 444\n",
      "2024-04-23 19:20:10> Invalid proteins: 4\n",
      "2024-04-23 19:20:10> Existing proteins: 82\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "# download cif data for proteins\n",
    "# SLOW THE FIRST TIME\n",
    "valid_proteins_cif, invalid_proteins_cif, existing_proteins_cif = download_alphafold_cif(\n",
    "    proteins=uniprotIDs_fullproteome,\n",
    "    out_folder=cif_dir\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████| 530/530 [20:33<00:00,  2.33s/it]"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2024-04-23 19:41:39> Valid proteins: 444\n",
      "2024-04-23 19:41:39> Invalid proteins: 4\n",
      "2024-04-23 19:41:39> Existing proteins: 82\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "# download pae data for proteins\n",
    "# SLOW THE FIRST TIME\n",
    "valid_proteins_pae, invalid_proteins_pae, existing_proteins_pae = download_alphafold_pae(\n",
    "    proteins=uniprotIDs_fullproteome,\n",
    "    out_folder=pae_dir, \n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "  0%|          | 0/654 [00:00<?, ?it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "  0%|          | 3/654 [00:00<00:30, 21.61it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "  1%|          | 6/654 [00:00<00:45, 14.17it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "  1%|          | 8/654 [00:00<00:41, 15.47it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "  2%|▏         | 10/654 [00:00<00:47, 13.57it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "  2%|▏         | 12/654 [00:00<00:54, 11.69it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "  2%|▏         | 16/654 [00:01<00:36, 17.39it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "  3%|▎         | 19/654 [00:01<00:36, 17.42it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "  3%|▎         | 22/654 [00:01<00:31, 20.01it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "  4%|▍         | 28/654 [00:01<00:22, 27.76it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "  5%|▍         | 32/654 [00:01<00:21, 28.93it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "  6%|▌         | 36/654 [00:01<00:23, 25.85it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "  6%|▋         | 42/654 [00:01<00:20, 30.58it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "  7%|▋         | 46/654 [00:02<00:23, 26.43it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "  8%|▊         | 50/654 [00:02<00:22, 27.22it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "  9%|▉         | 59/654 [00:02<00:14, 40.23it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 11%|█         | 71/654 [00:02<00:10, 58.25it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 12%|█▏        | 80/654 [00:02<00:09, 58.13it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 13%|█▎        | 87/654 [00:02<00:14, 39.70it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 14%|█▍        | 93/654 [00:03<00:14, 39.06it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 15%|█▌        | 101/654 [00:03<00:13, 41.33it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 16%|█▌        | 106/654 [00:03<00:13, 40.85it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 17%|█▋        | 111/654 [00:03<00:17, 30.86it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 18%|█▊        | 115/654 [00:03<00:17, 30.91it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 18%|█▊        | 119/654 [00:03<00:16, 31.59it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 19%|█▉        | 123/654 [00:04<00:18, 28.45it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 19%|█▉        | 127/654 [00:04<00:18, 28.11it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 20%|██        | 134/654 [00:04<00:16, 32.47it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 21%|██        | 138/654 [00:04<00:16, 30.78it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 22%|██▏       | 142/654 [00:04<00:19, 26.39it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 22%|██▏       | 146/654 [00:04<00:19, 25.67it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 23%|██▎       | 149/654 [00:05<00:19, 26.02it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 24%|██▎       | 154/654 [00:05<00:18, 27.67it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 24%|██▍       | 157/654 [00:05<00:19, 25.65it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 24%|██▍       | 160/654 [00:05<00:29, 16.89it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 25%|██▍       | 163/654 [00:05<00:28, 17.10it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 25%|██▌       | 165/654 [00:06<00:33, 14.69it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 26%|██▌       | 167/654 [00:06<00:33, 14.69it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 26%|██▌       | 169/654 [00:06<00:35, 13.79it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 26%|██▌       | 171/654 [00:06<00:37, 12.85it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 26%|██▋       | 173/654 [00:06<00:35, 13.66it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 27%|██▋       | 175/654 [00:06<00:32, 14.61it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 27%|██▋       | 177/654 [00:07<00:40, 11.73it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 27%|██▋       | 179/654 [00:07<00:38, 12.38it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 28%|██▊       | 182/654 [00:07<00:34, 13.77it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 28%|██▊       | 184/654 [00:07<00:32, 14.62it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 28%|██▊       | 186/654 [00:07<00:35, 13.18it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 29%|██▊       | 188/654 [00:07<00:36, 12.80it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 30%|██▉       | 193/654 [00:08<00:24, 18.73it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 30%|██▉       | 196/654 [00:08<00:27, 16.71it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 31%|███       | 200/654 [00:08<00:23, 19.59it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 31%|███       | 203/654 [00:08<00:25, 17.40it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 31%|███▏      | 206/654 [00:08<00:26, 17.20it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 32%|███▏      | 208/654 [00:08<00:25, 17.51it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 33%|███▎      | 214/654 [00:09<00:19, 22.83it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 33%|███▎      | 217/654 [00:09<00:21, 20.28it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 34%|███▍      | 221/654 [00:09<00:18, 23.20it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 34%|███▍      | 225/654 [00:09<00:16, 26.06it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 35%|███▍      | 228/654 [00:09<00:20, 20.83it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 35%|███▌      | 231/654 [00:10<00:26, 15.87it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 36%|███▌      | 235/654 [00:10<00:22, 18.23it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 36%|███▋      | 238/654 [00:10<00:27, 14.88it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 37%|███▋      | 241/654 [00:10<00:24, 16.71it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 37%|███▋      | 245/654 [00:10<00:21, 18.60it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 38%|███▊      | 250/654 [00:10<00:17, 23.43it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 39%|███▊      | 253/654 [00:11<00:19, 20.30it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 39%|███▉      | 256/654 [00:11<00:23, 17.26it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 40%|███▉      | 259/654 [00:11<00:20, 19.21it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 40%|████      | 262/654 [00:11<00:19, 19.89it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 41%|████      | 265/654 [00:11<00:23, 16.40it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 41%|████      | 268/654 [00:12<00:23, 16.32it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 41%|████▏     | 271/654 [00:12<00:20, 18.66it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 42%|████▏     | 274/654 [00:12<00:23, 15.88it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 42%|████▏     | 277/654 [00:12<00:20, 18.05it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 43%|████▎     | 280/654 [00:12<00:18, 19.96it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 43%|████▎     | 283/654 [00:12<00:22, 16.68it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 44%|████▎     | 285/654 [00:13<00:23, 15.98it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 44%|████▍     | 287/654 [00:13<00:25, 14.67it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 44%|████▍     | 290/654 [00:13<00:23, 15.73it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 45%|████▍     | 293/654 [00:13<00:19, 18.18it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 45%|████▌     | 296/654 [00:13<00:18, 19.18it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 46%|████▌     | 299/654 [00:13<00:19, 17.78it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 46%|████▌     | 301/654 [00:13<00:19, 17.91it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 46%|████▋     | 303/654 [00:14<00:20, 17.25it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 47%|████▋     | 305/654 [00:14<00:21, 16.44it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 47%|████▋     | 308/654 [00:14<00:19, 18.15it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 47%|████▋     | 310/654 [00:14<00:25, 13.59it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 48%|████▊     | 312/654 [00:14<00:23, 14.54it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 48%|████▊     | 314/654 [00:14<00:24, 13.71it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 48%|████▊     | 316/654 [00:14<00:23, 14.65it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 49%|████▊     | 318/654 [00:15<00:24, 13.91it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 49%|████▉     | 321/654 [00:15<00:19, 17.20it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 50%|████▉     | 324/654 [00:15<00:18, 17.77it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 50%|████▉     | 326/654 [00:15<00:22, 14.76it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 50%|█████     | 328/654 [00:15<00:22, 14.25it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 51%|█████     | 331/654 [00:15<00:21, 14.80it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 51%|█████     | 333/654 [00:16<00:24, 13.04it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 51%|█████     | 335/654 [00:16<00:28, 11.11it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 52%|█████▏    | 338/654 [00:16<00:23, 13.64it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 52%|█████▏    | 340/654 [00:16<00:21, 14.50it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 52%|█████▏    | 343/654 [00:16<00:19, 15.83it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 53%|█████▎    | 345/654 [00:16<00:19, 16.25it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 54%|█████▎    | 350/654 [00:17<00:13, 23.08it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 54%|█████▍    | 353/654 [00:17<00:17, 17.70it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 54%|█████▍    | 356/654 [00:17<00:14, 19.99it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 55%|█████▍    | 359/654 [00:17<00:14, 21.03it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 55%|█████▌    | 362/654 [00:17<00:16, 17.59it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 56%|█████▌    | 365/654 [00:17<00:15, 18.46it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 56%|█████▋    | 368/654 [00:18<00:13, 20.48it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 57%|█████▋    | 371/654 [00:18<00:17, 16.48it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 57%|█████▋    | 375/654 [00:18<00:13, 20.51it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 58%|█████▊    | 378/654 [00:18<00:13, 21.16it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 58%|█████▊    | 381/654 [00:18<00:13, 19.70it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 59%|█████▊    | 384/654 [00:18<00:14, 18.89it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 59%|█████▉    | 389/654 [00:19<00:11, 23.00it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 60%|█████▉    | 392/654 [00:19<00:13, 19.49it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 60%|██████    | 395/654 [00:19<00:12, 21.53it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 61%|██████    | 398/654 [00:19<00:14, 17.39it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 62%|██████▏   | 406/654 [00:19<00:10, 24.75it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 63%|██████▎   | 409/654 [00:20<00:12, 19.54it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 63%|██████▎   | 412/654 [00:20<00:14, 17.09it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 64%|██████▍   | 417/654 [00:20<00:11, 21.48it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 64%|██████▍   | 420/654 [00:20<00:12, 19.16it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 65%|██████▍   | 423/654 [00:20<00:12, 18.27it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 65%|██████▌   | 426/654 [00:21<00:12, 17.62it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 65%|██████▌   | 428/654 [00:21<00:12, 17.92it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 66%|██████▌   | 430/654 [00:21<00:12, 17.74it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 66%|██████▌   | 432/654 [00:21<00:12, 17.15it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 66%|██████▋   | 434/654 [00:21<00:15, 13.83it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 67%|██████▋   | 436/654 [00:21<00:18, 12.08it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 67%|██████▋   | 441/654 [00:22<00:13, 15.56it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 68%|██████▊   | 443/654 [00:22<00:13, 16.05it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 68%|██████▊   | 445/654 [00:22<00:12, 16.23it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 69%|██████▊   | 448/654 [00:22<00:11, 17.64it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 69%|██████▉   | 450/654 [00:22<00:12, 16.36it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 69%|██████▉   | 453/654 [00:22<00:10, 18.30it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 70%|██████▉   | 455/654 [00:22<00:14, 14.19it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 70%|███████   | 461/654 [00:23<00:08, 22.72it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 71%|███████   | 465/654 [00:23<00:07, 26.41it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 72%|███████▏  | 469/654 [00:23<00:06, 28.45it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 72%|███████▏  | 473/654 [00:23<00:06, 29.57it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 73%|███████▎  | 478/654 [00:23<00:05, 32.57it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 74%|███████▎  | 482/654 [00:23<00:05, 29.36it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 74%|███████▍  | 486/654 [00:23<00:05, 30.27it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 75%|███████▌  | 492/654 [00:23<00:04, 36.74it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 76%|███████▌  | 497/654 [00:24<00:04, 37.89it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 77%|███████▋  | 501/654 [00:24<00:04, 36.81it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 77%|███████▋  | 505/654 [00:24<00:04, 31.90it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 78%|███████▊  | 509/654 [00:24<00:05, 25.55it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 78%|███████▊  | 512/654 [00:24<00:06, 20.65it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 79%|███████▊  | 515/654 [00:25<00:08, 16.46it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 79%|███████▉  | 518/654 [00:25<00:07, 18.21it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 80%|███████▉  | 522/654 [00:25<00:06, 20.26it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 80%|████████  | 525/654 [00:25<00:06, 19.41it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 81%|████████  | 528/654 [00:25<00:07, 17.42it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 81%|████████  | 530/654 [00:25<00:07, 16.50it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 81%|████████▏ | 533/654 [00:26<00:07, 16.79it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 82%|████████▏ | 536/654 [00:26<00:06, 17.74it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 83%|████████▎ | 540/654 [00:26<00:05, 21.83it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 83%|████████▎ | 543/654 [00:26<00:05, 21.17it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 84%|████████▍ | 548/654 [00:26<00:03, 27.32it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 84%|████████▍ | 552/654 [00:26<00:03, 28.37it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 85%|████████▌ | 556/654 [00:26<00:03, 25.29it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 85%|████████▌ | 559/654 [00:27<00:03, 26.15it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 86%|████████▌ | 562/654 [00:27<00:03, 25.28it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 87%|████████▋ | 566/654 [00:27<00:03, 28.22it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 87%|████████▋ | 569/654 [00:27<00:03, 24.90it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 87%|████████▋ | 572/654 [00:27<00:03, 23.80it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 88%|████████▊ | 576/654 [00:27<00:03, 25.83it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 89%|████████▊ | 579/654 [00:27<00:02, 25.73it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 89%|████████▉ | 582/654 [00:27<00:03, 22.13it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 89%|████████▉ | 585/654 [00:28<00:03, 21.46it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 90%|████████▉ | 588/654 [00:28<00:03, 16.68it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 90%|█████████ | 590/654 [00:28<00:03, 17.09it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 91%|█████████ | 594/654 [00:28<00:02, 20.75it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 91%|█████████▏| 598/654 [00:28<00:02, 24.96it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 92%|█████████▏| 601/654 [00:28<00:02, 26.16it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 92%|█████████▏| 604/654 [00:29<00:02, 21.07it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 93%|█████████▎| 607/654 [00:29<00:02, 18.97it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 93%|█████████▎| 610/654 [00:29<00:02, 18.57it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 94%|█████████▎| 613/654 [00:29<00:02, 16.71it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 94%|█████████▍| 616/654 [00:29<00:02, 17.64it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 95%|█████████▌| 624/654 [00:29<00:01, 25.83it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 96%|█████████▌| 627/654 [00:30<00:01, 25.37it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 96%|█████████▋| 631/654 [00:30<00:00, 24.73it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 97%|█████████▋| 634/654 [00:30<00:00, 25.42it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 97%|█████████▋| 637/654 [00:30<00:00, 25.68it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 98%|█████████▊| 641/654 [00:30<00:00, 28.85it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 98%|█████████▊| 644/654 [00:30<00:00, 26.01it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      " 99%|█████████▉| 648/654 [00:30<00:00, 28.95it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "100%|█████████▉| 652/654 [00:31<00:00, 22.06it/s]/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "/Users/ritwiksrinivas/anaconda3/envs/methionine-analysis-env/lib/python3.9/site-packages/structuremap/processing.py:286: FutureWarning: errors='ignore' is deprecated and will raise in a future version. Use to_numeric without passing `errors` and catch exceptions explicitly instead\n",
      "  df = df.apply(pd.to_numeric, errors='ignore')\n",
      "100%|██████████| 654/654 [00:31<00:00, 20.95it/s]\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>protein_id</th>\n",
       "      <th>protein_number</th>\n",
       "      <th>AA</th>\n",
       "      <th>position</th>\n",
       "      <th>quality</th>\n",
       "      <th>x_coord_c</th>\n",
       "      <th>x_coord_ca</th>\n",
       "      <th>x_coord_cb</th>\n",
       "      <th>x_coord_n</th>\n",
       "      <th>y_coord_c</th>\n",
       "      <th>...</th>\n",
       "      <th>z_coord_ca</th>\n",
       "      <th>z_coord_cb</th>\n",
       "      <th>z_coord_n</th>\n",
       "      <th>secondary_structure</th>\n",
       "      <th>structure_group</th>\n",
       "      <th>BEND</th>\n",
       "      <th>HELX</th>\n",
       "      <th>STRN</th>\n",
       "      <th>TURN</th>\n",
       "      <th>unstructured</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A2ADF7</td>\n",
       "      <td>1</td>\n",
       "      <td>M</td>\n",
       "      <td>1</td>\n",
       "      <td>34.90</td>\n",
       "      <td>-16.170</td>\n",
       "      <td>-15.807</td>\n",
       "      <td>-16.898</td>\n",
       "      <td>-15.637</td>\n",
       "      <td>-29.957</td>\n",
       "      <td>...</td>\n",
       "      <td>38.096</td>\n",
       "      <td>37.222</td>\n",
       "      <td>39.345</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A2ADF7</td>\n",
       "      <td>1</td>\n",
       "      <td>K</td>\n",
       "      <td>2</td>\n",
       "      <td>27.13</td>\n",
       "      <td>-16.032</td>\n",
       "      <td>-15.462</td>\n",
       "      <td>-14.173</td>\n",
       "      <td>-15.203</td>\n",
       "      <td>-26.991</td>\n",
       "      <td>...</td>\n",
       "      <td>38.724</td>\n",
       "      <td>39.162</td>\n",
       "      <td>38.463</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A2ADF7</td>\n",
       "      <td>1</td>\n",
       "      <td>P</td>\n",
       "      <td>3</td>\n",
       "      <td>32.83</td>\n",
       "      <td>-16.958</td>\n",
       "      <td>-17.790</td>\n",
       "      <td>-19.230</td>\n",
       "      <td>-17.200</td>\n",
       "      <td>-24.520</td>\n",
       "      <td>...</td>\n",
       "      <td>36.289</td>\n",
       "      <td>36.675</td>\n",
       "      <td>37.473</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A2ADF7</td>\n",
       "      <td>1</td>\n",
       "      <td>T</td>\n",
       "      <td>4</td>\n",
       "      <td>33.64</td>\n",
       "      <td>-16.686</td>\n",
       "      <td>-15.841</td>\n",
       "      <td>-15.615</td>\n",
       "      <td>-16.546</td>\n",
       "      <td>-22.219</td>\n",
       "      <td>...</td>\n",
       "      <td>33.884</td>\n",
       "      <td>32.424</td>\n",
       "      <td>34.597</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A2ADF7</td>\n",
       "      <td>1</td>\n",
       "      <td>Q</td>\n",
       "      <td>5</td>\n",
       "      <td>28.86</td>\n",
       "      <td>-16.698</td>\n",
       "      <td>-16.784</td>\n",
       "      <td>-16.026</td>\n",
       "      <td>-16.209</td>\n",
       "      <td>-19.372</td>\n",
       "      <td>...</td>\n",
       "      <td>34.618</td>\n",
       "      <td>35.575</td>\n",
       "      <td>34.658</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196649</th>\n",
       "      <td>S4R2K0</td>\n",
       "      <td>526</td>\n",
       "      <td>M</td>\n",
       "      <td>227</td>\n",
       "      <td>93.01</td>\n",
       "      <td>-4.021</td>\n",
       "      <td>-2.734</td>\n",
       "      <td>-1.530</td>\n",
       "      <td>-2.851</td>\n",
       "      <td>2.349</td>\n",
       "      <td>...</td>\n",
       "      <td>-27.747</td>\n",
       "      <td>-27.920</td>\n",
       "      <td>-26.354</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196650</th>\n",
       "      <td>S4R2K0</td>\n",
       "      <td>526</td>\n",
       "      <td>E</td>\n",
       "      <td>228</td>\n",
       "      <td>93.81</td>\n",
       "      <td>-7.039</td>\n",
       "      <td>-5.845</td>\n",
       "      <td>-6.164</td>\n",
       "      <td>-4.573</td>\n",
       "      <td>1.709</td>\n",
       "      <td>...</td>\n",
       "      <td>-27.713</td>\n",
       "      <td>-26.580</td>\n",
       "      <td>-27.423</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196651</th>\n",
       "      <td>S4R2K0</td>\n",
       "      <td>526</td>\n",
       "      <td>V</td>\n",
       "      <td>229</td>\n",
       "      <td>88.77</td>\n",
       "      <td>-7.847</td>\n",
       "      <td>-7.972</td>\n",
       "      <td>-7.895</td>\n",
       "      <td>-6.944</td>\n",
       "      <td>4.796</td>\n",
       "      <td>...</td>\n",
       "      <td>-27.292</td>\n",
       "      <td>-26.038</td>\n",
       "      <td>-27.216</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196652</th>\n",
       "      <td>S4R2K0</td>\n",
       "      <td>526</td>\n",
       "      <td>N</td>\n",
       "      <td>230</td>\n",
       "      <td>90.23</td>\n",
       "      <td>-6.518</td>\n",
       "      <td>-6.363</td>\n",
       "      <td>-4.940</td>\n",
       "      <td>-6.637</td>\n",
       "      <td>4.882</td>\n",
       "      <td>...</td>\n",
       "      <td>-30.321</td>\n",
       "      <td>-30.226</td>\n",
       "      <td>-29.116</td>\n",
       "      <td>HELX_RH_AL_P</td>\n",
       "      <td>HELX</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>196653</th>\n",
       "      <td>S4R2K0</td>\n",
       "      <td>526</td>\n",
       "      <td>D</td>\n",
       "      <td>231</td>\n",
       "      <td>84.67</td>\n",
       "      <td>-8.280</td>\n",
       "      <td>-6.767</td>\n",
       "      <td>-6.194</td>\n",
       "      <td>-6.463</td>\n",
       "      <td>2.565</td>\n",
       "      <td>...</td>\n",
       "      <td>-32.605</td>\n",
       "      <td>-32.255</td>\n",
       "      <td>-31.516</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>unstructured</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>196654 rows × 24 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       protein_id  protein_number AA  position  quality  x_coord_c  \\\n",
       "0          A2ADF7               1  M         1    34.90    -16.170   \n",
       "1          A2ADF7               1  K         2    27.13    -16.032   \n",
       "2          A2ADF7               1  P         3    32.83    -16.958   \n",
       "3          A2ADF7               1  T         4    33.64    -16.686   \n",
       "4          A2ADF7               1  Q         5    28.86    -16.698   \n",
       "...           ...             ... ..       ...      ...        ...   \n",
       "196649     S4R2K0             526  M       227    93.01     -4.021   \n",
       "196650     S4R2K0             526  E       228    93.81     -7.039   \n",
       "196651     S4R2K0             526  V       229    88.77     -7.847   \n",
       "196652     S4R2K0             526  N       230    90.23     -6.518   \n",
       "196653     S4R2K0             526  D       231    84.67     -8.280   \n",
       "\n",
       "        x_coord_ca  x_coord_cb  x_coord_n  y_coord_c  ...  z_coord_ca  \\\n",
       "0          -15.807     -16.898    -15.637    -29.957  ...      38.096   \n",
       "1          -15.462     -14.173    -15.203    -26.991  ...      38.724   \n",
       "2          -17.790     -19.230    -17.200    -24.520  ...      36.289   \n",
       "3          -15.841     -15.615    -16.546    -22.219  ...      33.884   \n",
       "4          -16.784     -16.026    -16.209    -19.372  ...      34.618   \n",
       "...            ...         ...        ...        ...  ...         ...   \n",
       "196649      -2.734      -1.530     -2.851      2.349  ...     -27.747   \n",
       "196650      -5.845      -6.164     -4.573      1.709  ...     -27.713   \n",
       "196651      -7.972      -7.895     -6.944      4.796  ...     -27.292   \n",
       "196652      -6.363      -4.940     -6.637      4.882  ...     -30.321   \n",
       "196653      -6.767      -6.194     -6.463      2.565  ...     -32.605   \n",
       "\n",
       "        z_coord_cb  z_coord_n  secondary_structure  structure_group  BEND  \\\n",
       "0           37.222     39.345         unstructured     unstructured     0   \n",
       "1           39.162     38.463         unstructured     unstructured     0   \n",
       "2           36.675     37.473         unstructured     unstructured     0   \n",
       "3           32.424     34.597         unstructured     unstructured     0   \n",
       "4           35.575     34.658         unstructured     unstructured     0   \n",
       "...            ...        ...                  ...              ...   ...   \n",
       "196649     -27.920    -26.354         HELX_RH_AL_P             HELX     0   \n",
       "196650     -26.580    -27.423         HELX_RH_AL_P             HELX     0   \n",
       "196651     -26.038    -27.216         HELX_RH_AL_P             HELX     0   \n",
       "196652     -30.226    -29.116         HELX_RH_AL_P             HELX     0   \n",
       "196653     -32.255    -31.516         unstructured     unstructured     0   \n",
       "\n",
       "        HELX STRN TURN  unstructured  \n",
       "0          0    0    0             1  \n",
       "1          0    0    0             1  \n",
       "2          0    0    0             1  \n",
       "3          0    0    0             1  \n",
       "4          0    0    0             1  \n",
       "...      ...  ...  ...           ...  \n",
       "196649     1    0    0             0  \n",
       "196650     1    0    0             0  \n",
       "196651     1    0    0             0  \n",
       "196652     1    0    0             0  \n",
       "196653     0    0    0             1  \n",
       "\n",
       "[196654 rows x 24 columns]"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# format alphafold data into dataframe\n",
    "alphafold_annotation = format_alphafold_data(\n",
    "    directory=cif_dir, \n",
    "    protein_ids=uniprotIDs_fullproteome)\n",
    "alphafold_annotation"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "methionine-analysis-env",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
